Index: src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- src/plugins/compilergcc/compilergcc.cpp	(revision 2411)
+++ src/plugins/compilergcc/compilergcc.cpp	(arbetskopia)
@@ -61,6 +61,9 @@
     #include "compilerDMD.h"
 	#include "compilerGNUARM.h"
 #endif
+#ifdef __WXMAC__
+    #include "compilerGDC.h"
+#endif
 #include "compilerICC.h"
 #include "compilerSDCC.h"
 
@@ -244,6 +247,9 @@
 	CompilerFactory::RegisterCompiler(new CompilerDMD);
 	CompilerFactory::RegisterCompiler(new CompilerGNUARM);
 #endif
+#ifdef __WXMAC__
+	CompilerFactory::RegisterCompiler(new CompilerGDC);
+#endif
 
 	// register (if any) user-copies of built-in compilers
 	CompilerFactory::RegisterUserCompilers();
