Index: src/sdk/editkeywordsdlg.h
===================================================================
--- src/sdk/editkeywordsdlg.h	(revision 2371)
+++ src/sdk/editkeywordsdlg.h	(working copy)
@@ -15,9 +15,10 @@
     protected:
         void OnSetChange(wxSpinEvent& event);
 
+        wxSpinCtrl* spnSet;
+    public:
         EditorColorSet* m_pTheme;
         HighlightLanguage m_Lang;
-        wxSpinCtrl* spnSet;
         wxTextCtrl* txtKeywords;
         int m_LastSet;
     private:
Index: src/sdk/editorconfigurationdlg.cpp
===================================================================
--- src/sdk/editorconfigurationdlg.cpp	(revision 2371)
+++ src/sdk/editorconfigurationdlg.cpp	(working copy)
@@ -635,6 +635,8 @@
 	    EditKeywordsDlg dlg(0, m_Theme, m_Lang);
         PlaceWindow(&dlg);
 	    dlg.ShowModal();
+        // Update the last modified keywords set            //pecan 2006/4/26
+        dlg.m_pTheme->SetKeywords(dlg.m_Lang, dlg.m_LastSet, dlg.txtKeywords->GetValue());
 	}
 }
  