Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 2438)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -213,7 +213,7 @@
 
     CreateSearchLog();
     LoadAutoComplete();
-    m_zoom = Manager::Get()->GetConfigManager(_T("editor"))->ReadInt(_T("zoom"));
+    m_zoom = Manager::Get()->GetConfigManager(_T("editor"))->ReadInt(_T("/zoom"));
 }
 
 // class destructor
@@ -246,7 +246,7 @@
         delete m_pData;
         m_pData = NULL;
     }
-    Manager::Get()->GetConfigManager(_T("editor"))->Write(_T("zoom"), m_zoom);
+    Manager::Get()->GetConfigManager(_T("editor"))->Write(_T("/zoom"), m_zoom);
 }
 
 void EditorManager::CreateMenu(wxMenuBar* menuBar)
