Index: src/sdk/templatemanager.cpp
===================================================================
--- src/sdk/templatemanager.cpp	(revision 2440)
+++ src/sdk/templatemanager.cpp	(working copy)
@@ -282,7 +282,7 @@
                     continue;
                 wxCopyFile(baseDir + sep + fsf.source, dst);
                 for (int i = 0; i < prj->GetBuildTargetsCount(); ++i)
-                    prj->AddFile(i, fsf.destination);
+                    prj->AddFile(i, dst);
             }
         }
 
