Index: D:/codeblocks/src/sdk/toolsmanager.cpp
===================================================================
--- D:/codeblocks/src/sdk/toolsmanager.cpp	(revision 2483)
+++ D:/codeblocks/src/sdk/toolsmanager.cpp	(working copy)
@@ -307,7 +307,7 @@
 
 		// prepend a 0-padded 2-digit number to keep ordering
 		wxString tmp;
-		tmp.Printf(_("tool%2.2d"), count++);
+		tmp.Printf(_T("tool%2.2d"), count++);
 
 		elem << _T("/") << tmp  << _T("/");
 		cfg->Write(elem + _T("name"), tool->name);
There is a little bug with CB using the localization file. This patch can fix it.