Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 2490)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -776,16 +776,16 @@
     }
 }
 
-void cbEditor::SetLanguage( HighlightLanguage lang )
+void cbEditor::SetLanguage(HighlightLanguage lang)
 {
+    if (m_pControl->GetReadOnly())
+        return;
+
     if (m_pTheme)
-    {
         m_lang = m_pTheme->Apply(this, lang);
-    }
+
     else
-    {
         m_lang = HL_AUTO;
-    }
 }
 
 bool cbEditor::Open(bool detectEncoding)
