--- src/sdk/scriptingcall.h   2006-05-24 12:53:22.424842528 -0700 
+++ src/sdk/scriptingcall.h   2006-05-24 12:58:17.075048888 -0700 
@@ -30,7 +30,7 @@ 
 template <typename OP> inline void ExecuteSetArg(asIScriptContext* ctx, int arg, OP op) 
 { 
     // default template treats it as 'void*' 
-    ctx->SetArgObject(arg, op); 
+    ctx->SetArgObject(arg, (void*)op); 
 } 
 template <> inline void ExecuteSetArg(asIScriptContext* ctx, int arg, DummyOperand op){ /* do nothing */ } 
 template <> inline void ExecuteSetArg(asIScriptContext* ctx, int arg, asQWORD op){ ctx->SetArgQWord(arg, op); }