Index: src/plugins/compilergcc/compilererrors.cpp
===================================================================
--- src/plugins/compilergcc/compilererrors.cpp	(revision 2530)
+++ src/plugins/compilergcc/compilererrors.cpp	(working copy)
@@ -188,9 +188,9 @@
     if (ed)
     {
         ed->Activate();
+        ed->UnfoldBlockFromLine(error.line - 1);
         ed->GotoLine(error.line - 1);
         ed->SetErrorLine(error.line - 1);
-        ed->UnfoldBlockFromLine(error.line - 1);
     }
 }
 
