Index: src/plugins/debuggergdb/gdb_driver.cpp
===================================================================
--- src/plugins/debuggergdb/gdb_driver.cpp	(revision 2539)
+++ src/plugins/debuggergdb/gdb_driver.cpp	(working copy)
@@ -481,7 +482,8 @@
         }
 
         // signal
-        else if (lines[i].StartsWith(_T("Program received signal SIGSEGV")))
+        else if (lines[i].StartsWith(_T("Program received signal SIG")) &&
+        !( lines[i].StartsWith(_T("Program received signal SIGINT")) || lines[i].StartsWith(_T("Program received signal SIGTRAP"))) )
         {
             Log(lines[i]);
             m_pDBG->BringAppToFront();
