Index: configure.in
===================================================================
--- configure.in	(Revision 2539)
+++ configure.in	(Arbeitskopie)
@@ -59,7 +59,7 @@
 CODEBLOCKS_ENABLE_SETTINGS
 
 AM_OPTIONS_WXCONFIG
-AM_PATH_WXCONFIG(2.4.2, wxWin=1)
+AM_PATH_WXCONFIG(2.6.0, wxWin=1)
         if test "$wxWin" != 1; then
         AC_MSG_ERROR([
                 wxWidgets must be installed on your system.
@@ -68,7 +68,7 @@
                 where wxWidgets libraries are installed (returned by
                 'wx-config --libs' or 'wx-config --static --libs' command)
                 is in LD_LIBRARY_PATH or equivalent variable and
-                wxWindows version is 2.4.2 or above.
+                wxWindows version is 2.6.0 or above.
                 ])
         fi
 
