Index: trunk/src/sdk/compiler.cpp
===================================================================
--- trunk/src/sdk/compiler.cpp	(revision 2539)
+++ trunk/src/sdk/compiler.cpp	(working copy)
@@ -125,9 +125,9 @@
     }
 
     // make sure it's not starting with a number.
-    // if it is, prepend an underscore
+    // if it is, prepend "cb"
     if (wxIsdigit(newID.GetChar(0)))
-        newID.Prepend(_T('_'));
+        newID.Prepend(_T("cb"));
 
     if (newID.IsEmpty())
     {
