Index: src/plugins/debuggergdb/debuggertree.cpp
===================================================================
--- src/plugins/debuggergdb/debuggertree.cpp	(revision 2539)
+++ src/plugins/debuggergdb/debuggertree.cpp	(working copy)
@@ -795,7 +795,7 @@
         FixupVarNameForChange(itemtext);
         if (!itemtext.IsEmpty())
             var = itemtext;
-        while ((parent = m_pTree->GetItemParent(item)) && parent.IsOk())
+        while ((parent = m_pTree->GetItemParent(item)) && parent.IsOk() && parent != m_pTree->GetRootItem())
         {
             item = parent;
             wxString itemtext = m_pTree->GetItemText(item);
