Index: src/sdk/selecttargetdlg.cpp
===================================================================
--- src/sdk/selecttargetdlg.cpp	(revision 2602)
+++ src/sdk/selecttargetdlg.cpp	(working copy)
@@ -61,6 +61,7 @@
 	list->SetSelection(m_pProject->GetDefaultExecuteTargetIndex());
 
 	UpdateSelected();
+    XRCCTRL(*this, "wxID_OK", wxButton)->MoveBeforeInTabOrder (XRCCTRL(*this, "lstItems", wxListBox));
 }
 
 SelectTargetDlg::~SelectTargetDlg()
