Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 2602)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -448,7 +448,7 @@
     if (ftd)
     {
 	    // if it is a project...
-    	if (ftd->GetKind() == FileTreeData::ftdkProject)
+    	if ( (ftd->GetKind() == FileTreeData::ftdkProject) || (ftd->GetKind() == FileTreeData::ftdkVirtualGroup) )
     	{
 			if (ftd->GetProject() != m_pActiveProject)
 				menu.Append(idMenuSetActiveProject, _("Activate project"));
