diff -Naur codeblocks-1.0rc2/src/sdk/editormanager.h codeblocks-1.0rc2-fix-extra-qualification/src/sdk/editormanager.h
--- codeblocks-1.0rc2/src/sdk/editormanager.h	2005-10-25 10:59:02.000000000 +0300
+++ codeblocks-1.0rc2-fix-extra-qualification/src/sdk/editormanager.h	2006-06-24 10:20:56.000000000 +0300
@@ -126,7 +126,7 @@
 
         /** Builds Opened Files tree in the Projects tab
           */
-        wxTreeCtrl *EditorManager::GetTree();
+        wxTreeCtrl *GetTree();
         wxTreeItemId FindTreeFile(const wxString& filename);
         wxString GetTreeItemFilename(wxTreeItemId item);
         void BuildOpenedFilesTree(wxWindow* parent);
diff -Naur codeblocks-1.0rc2/src/sdk/managedthread.h codeblocks-1.0rc2-fix-extra-qualification/src/sdk/managedthread.h
--- codeblocks-1.0rc2/src/sdk/managedthread.h	2005-04-07 08:47:26.000000000 +0300
+++ codeblocks-1.0rc2-fix-extra-qualification/src/sdk/managedthread.h	2006-06-24 10:18:41.000000000 +0300
@@ -11,8 +11,8 @@
 class ManagedThread : public wxThread
 {
 public:
-    ManagedThread::ManagedThread(bool* abortflag  = 0L);
-    virtual ManagedThread::~ManagedThread();
+    ManagedThread(bool* abortflag  = 0L);
+    virtual ~ManagedThread();
     static unsigned long count_running();
     static unsigned long count_threads();
     static void abort_all(); // Warning! Once set, can't be reset!
diff -Naur codeblocks-1.0rc2/src/sdk/xtra_classes.h codeblocks-1.0rc2-fix-extra-qualification/src/sdk/xtra_classes.h
--- codeblocks-1.0rc2/src/sdk/xtra_classes.h	2005-08-27 23:27:14.000000000 +0300
+++ codeblocks-1.0rc2-fix-extra-qualification/src/sdk/xtra_classes.h	2006-06-24 10:19:24.000000000 +0300
@@ -18,8 +18,8 @@
 class wxSplitPanel : public wxPanel
 {
 public:
-    wxSplitPanel::wxSplitPanel() { /*nothing to init, really */ }
-    wxSplitPanel::wxSplitPanel(wxWindow* parent, wxWindowID id = -1,
+    wxSplitPanel() { /*nothing to init, really */ }
+    wxSplitPanel(wxWindow* parent, wxWindowID id = -1,
                                const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize,
                                long style = wxTAB_TRAVERSAL, const wxString& name = _T("splitpanel"),const wxString configname = wxEmptyString,int defaultsashposition = 150)
     {
@@ -33,9 +33,9 @@
       * If only one is found, the panel is not split.
       *
       */
-    void wxSplitPanel::RefreshSplitter(int idtop,int idbottom);
+    void RefreshSplitter(int idtop,int idbottom);
 
-    virtual wxSplitPanel::~wxSplitPanel();
+    virtual ~wxSplitPanel();
     wxSplitterWindow* GetSplitter() { return (this) ? m_splitter : 0L; }
     void SetConfigEntryForSplitter(const wxString& splitterconfig){ m_SplitterConfig = splitterconfig; }
 protected:
diff -Naur codeblocks-1.0rc2/src/src/main.cpp codeblocks-1.0rc2-fix-extra-qualification/src/src/main.cpp
--- codeblocks-1.0rc2/src/src/main.cpp	2005-10-25 10:59:02.000000000 +0300
+++ codeblocks-1.0rc2-fix-extra-qualification/src/src/main.cpp	2006-06-24 10:35:45.000000000 +0300
@@ -65,7 +65,7 @@
 class wxMyFileDropTarget : public wxFileDropTarget
 {
 public:
-    wxMyFileDropTarget::wxMyFileDropTarget(MainFrame *frame):m_frame(frame){}
+    wxMyFileDropTarget(MainFrame *frame):m_frame(frame){}
     virtual bool OnDropFiles(wxCoord x, wxCoord y, const wxArrayString& filenames)
     {
         if(!m_frame) return false;
diff -Naur codeblocks-1.0rc2/src/src/wxDockit/include/wx/barholder.h codeblocks-1.0rc2-fix-extra-qualification/src/src/wxDockit/include/wx/barholder.h
--- codeblocks-1.0rc2/src/src/wxDockit/include/wx/barholder.h	2005-09-08 10:55:24.000000000 +0300
+++ codeblocks-1.0rc2-fix-extra-qualification/src/src/wxDockit/include/wx/barholder.h	2006-06-24 10:33:57.000000000 +0300
@@ -39,7 +39,7 @@
     }
 
     // Normal constructor
-    wxBarHolder::wxBarHolder( wxWindow *parent, wxWindowID id = -1, const wxPoint& pos = wxDefaultPosition,
+    wxBarHolder( wxWindow *parent, wxWindowID id = -1, const wxPoint& pos = wxDefaultPosition,
         const wxSize& size = wxDefaultSize, long style = wxTAB_TRAVERSAL, const wxString& name = wxT("barholder") ) {
         Init();
 
