Index: src/sdk/cbproject.cpp
===================================================================
--- src/sdk/cbproject.cpp	(revision 2661)
+++ src/sdk/cbproject.cpp	(working copy)
@@ -533,7 +533,7 @@
                     EditorBase* eb = Manager::Get()->GetEditorManager()->Open(f->file.GetFullPath());
                     if(eb)
                     {
-                        Manager::Get()->GetProjectManager()->SetTopEditor(eb);
+//!                        Manager::Get()->GetProjectManager()->SetTopEditor(eb);
                         eb->Activate();
                     }
                 }
Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 2661)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -195,7 +195,7 @@
 ProjectManager::ProjectManager()
 	: m_pTree(0),
     m_pWorkspace(0),
-	m_pTopEditor(0),
+//!	m_pTopEditor(0),
     m_TreeCategorize(false),
     m_TreeUseFolders(true),
     m_TreeFreezeCounter(0),
@@ -960,7 +960,7 @@
     }
     return m_pWorkspace;
 }
-
+/*!
 void ProjectManager::SetTopEditor(EditorBase* ed)
 {
     m_pTopEditor = ed;
@@ -970,10 +970,10 @@
 {
     return m_pTopEditor;
 }
-
+*/
 bool ProjectManager::LoadWorkspace(const wxString& filename)
 {
-    m_pTopEditor = 0;
+    //! m_pTopEditor = 0;
     if (!CloseWorkspace())
         return false; // didn't close
     m_IsLoadingWorkspace=true;
@@ -992,8 +992,8 @@
         m_IsLoadingWorkspace=false;
         Manager::Get()->GetEditorManager()->RebuildOpenedFilesTree();
         m_pTree->SetItemText(m_TreeRoot, m_pWorkspace->GetTitle());
-        if(m_pTopEditor)
-            m_pTopEditor->Activate();
+        // if(m_pTopEditor)
+        //    m_pTopEditor->Activate();
         Manager::Get()->GetEditorManager()->RefreshOpenedFilesTree(true);
         UnfreezeTree(true);
     }
Index: src/sdk/projectmanager.h
===================================================================
--- src/sdk/projectmanager.h	(revision 2661)
+++ src/sdk/projectmanager.h	(working copy)
@@ -346,9 +346,9 @@
           */
 		wxMenu* GetProjectMenu();
 		/** Sets the Top Editor (the active editor from the last session) */
-		void SetTopEditor(EditorBase* ed);
+//!		void SetTopEditor(EditorBase* ed);
 		/** @return The Top Editor */
-		EditorBase* GetTopEditor() const;
+//!		EditorBase* GetTopEditor() const;
 
 		/** @return The workspace icon index in the image list. */
 		int WorkspaceIconIndex();
@@ -399,7 +399,7 @@
         ProjectsArray* m_pProjects;
         DepsMap m_ProjectDeps;
         cbWorkspace* m_pWorkspace;
-        EditorBase* m_pTopEditor;
+//!        EditorBase* m_pTopEditor;
         bool m_TreeCategorize;
         bool m_TreeUseFolders;
 		FilesGroupsAndMasks* m_pFileGroups;
