Index: codestat.cpp
===================================================================
--- codestat.cpp	(revision 2750)
+++ codestat.cpp	(working copy)
@@ -100,13 +100,14 @@
 		return -1;
 	}
 
+    int dlgReturnCode = 0; //(pecan 2006/7/16)
     dlg = new CodeStatExecDlg(Manager::Get()->GetAppWindow());
 
     // Load the language settings and launch the main function
     LanguageDef languages[NB_FILETYPES_MAX];
     int nb_languages = LoadSettings(languages);
     if(dlg->Execute(languages,nb_languages) != 0)
-        return -1;
-
-    return 0;
+        dlgReturnCode = -1; //(pecan 2006/7/16)
+    dlg->Destroy();         //(pecan 2006/7/16)
+    return dlgReturnCode;   //(pecan 2006/7/16)
 }
