Index: E:/GPL/CodeBlocks/plugins/contrib/byogames/byogames.cbp
===================================================================
--- E:/GPL/CodeBlocks/plugins/contrib/byogames/byogames.cbp	(revision 2842)
+++ E:/GPL/CodeBlocks/plugins/contrib/byogames/byogames.cbp	(working copy)
@@ -34,7 +35,7 @@
 			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
-			<Add directory="$(#WX.lib)\gcc_dll\msw$(WX_SUFFIX)" />
+			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
 			<Add directory="$(#CB.include)" />
 			<Add directory="$(#CB.include)\tinyxml" />
@@ -50,7 +51,7 @@
 		<Linker>
 			<Add library="wxmsw26$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
-			<Add directory="$(#WX.lib)\gcc_dll" />
+			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB)\sdk\tinyxml" />
Index: E:/GPL/CodeBlocks/plugins/contrib/keybinder/keybinder.cbp
===================================================================
--- E:/GPL/CodeBlocks/plugins/contrib/keybinder/keybinder.cbp	(revision 2842)
+++ E:/GPL/CodeBlocks/plugins/contrib/keybinder/keybinder.cbp	(working copy)
@@ -49,7 +49,7 @@
 			<Add library="wxmsw26$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.lib)" />
-			<Add directory="$(#WX.lib)\gcc_dll" />
+			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="cbkeybinder.cpp">
 			<Option compilerVar="CPP" />