Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp   (revision 2889)
+++ src/sdk/editormanager.cpp   (working copy)
@@ -533,7 +533,15 @@
     }
 #ifdef USE_OPENFILES_TREE
     if(can_updateui)
-        AddFiletoTree(ed);
+    {
+       // If this file is already in the tree we have to call
+       // RefreshOpenedFilesTree because AddFiletoTree would not do anything
+       // and the file would not be selected.
+       if (eb)
+                       RefreshOpenedFilesTree();
+               else
+                       AddFiletoTree(ed);
+    }
 #endif
 
     // we 're done
