Index: src/plugins/codecompletion/nativeparser.cpp
===================================================================
--- src/plugins/codecompletion/nativeparser.cpp	(revision 2890)
+++ src/plugins/codecompletion/nativeparser.cpp	(arbetskopia)
@@ -1376,9 +1376,9 @@
             TokenIdxSet::iterator itsearch;
             if (search_scope)
                 itsearch = search_scope->begin();
-            while (itsearch == 0 || itsearch != search_scope->end())
+            while (!search_scope || itsearch != search_scope->end())
             {
-                Token* parent = tree->at(itsearch != 0 ? *itsearch : parentTokenIdx);
+                Token* parent = tree->at(search_scope ? *itsearch : parentTokenIdx);
 #ifdef DEBUG_CC_AI
                 Manager::Get()->GetMessageManager()->DebugLog(_T(" : looking under '%s'"), parent ? parent->m_Name.c_str() : _T("Global namespace"));
 #endif
@@ -1391,7 +1391,7 @@
                         break;
                     parent = tree->at(parent->m_ParentIndex);
                 } while (true);
-                if (itsearch == 0)
+                if (!search_scope)
                     break;
                 ++itsearch;
             }
