--- src/plugins/astyle/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/astyle/Makefile.am	2007-03-02 09:46:22.000000000 +0100
@@ -11,7 +11,7 @@
 
 libastyle_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libastyle_la_LIBADD = ../../sdk/libcodeblocks.la
+libastyle_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libastyle_la_SOURCES = asstreamiterator.cpp \
       astyleconfigdlg.cpp \
--- src/plugins/autosave/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/autosave/Makefile.am	2007-03-02 09:46:22.000000000 +0100
@@ -8,7 +8,7 @@
 
 libautosave_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libautosave_la_LIBADD = ../../sdk/libcodeblocks.la
+libautosave_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libautosave_la_SOURCES = autosave.cpp
 noinst_HEADERS = autosave.h
--- src/plugins/classwizard/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/classwizard/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libclasswizard_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libclasswizard_la_LIBADD = ../../sdk/libcodeblocks.la
+libclasswizard_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libclasswizard_la_SOURCES = classwizard.cpp classwizarddlg.cpp
 noinst_HEADERS = classwizard.h classwizarddlg.h
--- src/plugins/codecompletion/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/codecompletion/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -11,7 +11,7 @@
 
 libcodecompletion_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libcodecompletion_la_LIBADD = ../../sdk/libcodeblocks.la
+libcodecompletion_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libcodecompletion_la_SOURCES = ccdebuginfo.cpp \
 			ccoptionsdlg.cpp \
--- src/plugins/compilergcc/Makefile.am.orig	2007-03-02 09:35:13.000000000 +0100
+++ src/plugins/compilergcc/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -15,7 +15,7 @@
 
 libcompiler_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libcompiler_la_LIBADD = ../../sdk/libcodeblocks.la \
+libcompiler_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la \
 				depslib/src/libdepslib.la
 
 libcompiler_la_SOURCES = advancedcompileroptionsdlg.cpp \
--- src/plugins/contrib/byogames/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/byogames/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -8,7 +8,7 @@
 
 libbyogames_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libbyogames_la_LIBADD = ../../../sdk/libcodeblocks.la
+libbyogames_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libbyogames_la_SOURCES = byocbtris.cpp \
             byogame.cpp \
--- src/plugins/contrib/cb_koders/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/cb_koders/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -8,7 +8,7 @@
 
 libcb_koders_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libcb_koders_la_LIBADD = ../../../sdk/libcodeblocks.la
+libcb_koders_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libcb_koders_la_SOURCES = kodersdialog.cpp \
 			cb_koders.cpp
--- src/plugins/contrib/codesnippets/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/codesnippets/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libcodesnippets_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libcodesnippets_la_LIBADD = ../../../sdk/libcodeblocks.la
+libcodesnippets_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libcodesnippets_la_SOURCES = 	codesnippets.cpp \
 				codesnippetstreectrl.cpp \
--- src/plugins/contrib/codestat/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/codestat/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libcodestat_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libcodestat_la_LIBADD = ../../../sdk/libcodeblocks.la
+libcodestat_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libcodestat_la_SOURCES = codestatconfig.cpp \
 			codestat.cpp \
--- src/plugins/contrib/dragscroll/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/dragscroll/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -8,7 +8,7 @@
 
 libdragscroll_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libdragscroll_la_LIBADD = ../../../sdk/libcodeblocks.la
+libdragscroll_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libdragscroll_la_SOURCES = dragscroll.cpp \
 			dragscrollcfg.cpp
--- src/plugins/contrib/envvars/Makefile.am.orig	2007-03-02 09:35:11.000000000 +0100
+++ src/plugins/contrib/envvars/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -9,7 +9,7 @@
 
 libenvvars_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libenvvars_la_LIBADD = ../../../sdk/libcodeblocks.la
+libenvvars_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libenvvars_la_SOURCES = envvars.cpp
 
--- src/plugins/contrib/help_plugin/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/help_plugin/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -9,7 +9,7 @@
 
 libhelp_plugin_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libhelp_plugin_la_LIBADD = ../../../sdk/libcodeblocks.la
+libhelp_plugin_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libhelp_plugin_la_SOURCES = help_common.cpp \
 			HelpConfigDialog.cpp \
--- src/plugins/contrib/keybinder/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/keybinder/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -8,7 +8,7 @@
 
 libkeybinder_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libkeybinder_la_LIBADD = ../../../sdk/libcodeblocks.la
+libkeybinder_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libkeybinder_la_SOURCES = cbkeybinder.cpp \
 			keybinder.cpp \
--- src/plugins/contrib/lib_finder/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/lib_finder/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -8,7 +8,7 @@
 
 liblib_finder_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-liblib_finder_la_LIBADD = ../../../sdk/libcodeblocks.la
+liblib_finder_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 liblib_finder_la_SOURCES = 	dirlistdlg.cpp \
 				lib_finder.cpp \
--- src/plugins/contrib/profiler/Makefile.am.orig	2007-03-02 09:35:11.000000000 +0100
+++ src/plugins/contrib/profiler/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libProfiler_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libProfiler_la_LIBADD = ../../../sdk/libcodeblocks.la
+libProfiler_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libProfiler_la_SOURCES = cbprofilerconfig.cpp \
 			cbprofiler.cpp \
--- src/plugins/contrib/regex_testbed/Makefile.am.orig	2007-03-02 09:35:11.000000000 +0100
+++ src/plugins/contrib/regex_testbed/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -9,7 +9,7 @@
 
 libRegExTestbed_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libRegExTestbed_la_LIBADD = ../../../sdk/libcodeblocks.la
+libRegExTestbed_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libRegExTestbed_la_SOURCES = regexdlg.cpp RegExTestbed.cpp
 
--- src/plugins/contrib/source_exporter/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/source_exporter/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -12,7 +12,7 @@
 
 libexporter_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libexporter_la_LIBADD = ../../../sdk/libcodeblocks.la \
+libexporter_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la \
 				wxPdfDocument/libwxPdfDocument.la
 
 libexporter_la_SOURCES = BaseExporter.cpp \
--- src/plugins/contrib/symtab/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/symtab/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libSymTab_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libSymTab_la_LIBADD = ../../../sdk/libcodeblocks.la
+libSymTab_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la
 
 libSymTab_la_SOURCES = symtabconfig.cpp \
 			symtab.cpp \
--- src/plugins/contrib/wxSmith/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/wxSmith/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -12,7 +12,7 @@
 
 libwxsmith_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libwxsmith_la_LIBADD = ../../../sdk/libcodeblocks.la \
+libwxsmith_la_LIBADD = $(WX_LIBS) ../../../sdk/libcodeblocks.la \
 				properties/libwxsmith_properties.la \
 				wxwidgets/libwxsmith_wxwidgets.la
 
--- src/plugins/contrib/wxSmith/properties/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/wxSmith/properties/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -7,7 +7,7 @@
 
 libwxsmith_properties_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libwxsmith_properties_la_LIBADD = ../../../../sdk/libcodeblocks.la
+libwxsmith_properties_la_LIBADD = $(WX_LIBS) ../../../../sdk/libcodeblocks.la
 
 libwxsmith_properties_la_SOURCES = \
 ./wxsarraystringeditordlg.cpp \
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -9,7 +9,7 @@
 
 libwxsmith_wxwidgets_defitems_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libwxsmith_wxwidgets_defitems_la_LIBADD = ../../../../../sdk/libcodeblocks.la
+libwxsmith_wxwidgets_defitems_la_LIBADD = $(WX_LIBS) ../../../../../sdk/libcodeblocks.la
 
 libwxsmith_wxwidgets_defitems_la_SOURCES = \
 ./wxsdatepickerctrl.cpp \
--- src/plugins/contrib/wxSmith/wxwidgets/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/wxSmith/wxwidgets/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -15,7 +15,7 @@
 
 libwxsmith_wxwidgets_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libwxsmith_wxwidgets_la_LIBADD = ../../../../sdk/libcodeblocks.la \
+libwxsmith_wxwidgets_la_LIBADD = $(WX_LIBS) ../../../../sdk/libcodeblocks.la \
 				    properties/libwxsmith_wxwidgets_properties.la \
 				    defitems/libwxsmith_wxwidgets_defitems.la
 
--- src/plugins/contrib/wxSmith/wxwidgets/properties/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -9,7 +9,7 @@
 
 libwxsmith_wxwidgets_properties_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libwxsmith_wxwidgets_properties_la_LIBADD = ../../../../../sdk/libcodeblocks.la
+libwxsmith_wxwidgets_properties_la_LIBADD = $(WX_LIBS) ../../../../../sdk/libcodeblocks.la
 
 libwxsmith_wxwidgets_properties_la_SOURCES = \
 ./wxspositionsizeproperty.cpp \
--- src/plugins/debuggergdb/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/debuggergdb/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -13,7 +13,7 @@
 
 libdebugger_la_LDFLAGS = `pkg-config --libs gtk+-2.0` -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libdebugger_la_LIBADD = ../../sdk/libcodeblocks.la
+libdebugger_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libdebugger_la_SOURCES = backtracedlg.cpp \
 			breakpointsdlg.cpp \
--- src/plugins/defaultmimehandler/Makefile.am.orig	2007-03-02 09:35:13.000000000 +0100
+++ src/plugins/defaultmimehandler/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libdefaultmimehandler_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libdefaultmimehandler_la_LIBADD = ../../sdk/libcodeblocks.la
+libdefaultmimehandler_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libdefaultmimehandler_la_SOURCES = defaultmimehandler.cpp \
 			editmimetypesdlg.cpp
--- src/plugins/scriptedwizard/Makefile.am.orig	2007-03-02 09:35:13.000000000 +0100
+++ src/plugins/scriptedwizard/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -12,7 +12,7 @@
 
 libscriptedwizard_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libscriptedwizard_la_LIBADD = ../../sdk/libcodeblocks.la
+libscriptedwizard_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libscriptedwizard_la_SOURCES = buildtargetpanel.cpp \
                                 compilerpanel.cpp \
--- src/plugins/todo/Makefile.am.orig	2007-03-02 09:35:12.000000000 +0100
+++ src/plugins/todo/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -10,7 +10,7 @@
 
 libtodo_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libtodo_la_LIBADD = ../../sdk/libcodeblocks.la
+libtodo_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libtodo_la_SOURCES = addtododlg.cpp \
 			asktypedlg.cpp \
--- src/plugins/xpmanifest/Makefile.am.orig	2007-03-02 09:35:13.000000000 +0100
+++ src/plugins/xpmanifest/Makefile.am	2007-03-02 09:46:23.000000000 +0100
@@ -8,7 +8,7 @@
 
 libxpmanifest_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libxpmanifest_la_LIBADD = ../../sdk/libcodeblocks.la
+libxpmanifest_la_LIBADD = $(WX_LIBS) ../../sdk/libcodeblocks.la
 
 libxpmanifest_la_SOURCES = windowsxplooknfeel.cpp
 noinst_HEADERS = windowsxplooknfeel.h
