Index: sdk/virtualbuildtargetsdlg.cpp
===================================================================
--- sdk/virtualbuildtargetsdlg.cpp	(revision 2900)
+++ sdk/virtualbuildtargetsdlg.cpp	(working copy)
@@ -127,6 +127,9 @@
     if (targetName.IsEmpty())
         return;
 
+    if (lstAliases->GetStringSelection() == targetName)
+        return;
+
     if (lstAliases->FindString(targetName) != wxNOT_FOUND)
     {
         cbMessageBox(_("A virtual build target with this name already exists in this project!"),
