Index: src/plugins/contrib/byogames/byogames.cbp

===================================================================

--- src/plugins/contrib/byogames/byogames.cbp	(revision 2916)

+++ src/plugins/contrib/byogames/byogames.cbp	(working copy)

@@ -14,6 +14,7 @@

 			</Target>
 			<Environment>
 				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
 		<VirtualTargets>
@@ -37,7 +38,7 @@

 			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
-			<Add directory="$(#WX.lib)\gcc_dll\msw$(WX_SUFFIX)" />
+			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
 			<Add directory="$(#CB.include)" />
 			<Add directory="$(#CB.include)\tinyxml" />
@@ -53,7 +54,7 @@

 		<Linker>
 			<Add library="wxmsw26$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
-			<Add directory="$(#WX.lib)\gcc_dll" />
+			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB)\sdk\tinyxml" />
Index: src/plugins/contrib/codesnippets/codesnippets.cbp

===================================================================

--- src/plugins/contrib/codesnippets/codesnippets.cbp	(revision 2916)

+++ src/plugins/contrib/codesnippets/codesnippets.cbp	(working copy)

@@ -51,7 +51,6 @@

 			<Add directory="$(#cb)\sdk" />
 			<Add directory="$(#cb)\sdk\wxscintilla\include" />
 			<Add directory="$(#wx.include)" />
-			<Add directory="$(#wx.lib)\gcc_dll\mswu" />
 			<Add directory="$(#wx.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 		</Compiler>
 		<Linker>
@@ -61,7 +60,6 @@

 			<Add library="codeblocks" />
 			<Add directory="$(#cb)\devel" />
 			<Add directory="$(#cb)\sdk\tinyxml" />
-			<Add directory="$(#wx.lib)\gcc_dll" />
 			<Add directory="$(#wx.lib)\gcc_dll$(WX_CFG)" />
 		</Linker>
 		<ExtraCommands>
Index: src/plugins/contrib/keybinder/keybinder.cbp

===================================================================

--- src/plugins/contrib/keybinder/keybinder.cbp	(revision 2916)

+++ src/plugins/contrib/keybinder/keybinder.cbp	(working copy)

@@ -52,7 +52,7 @@

 			<Add library="wxmsw26$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.lib)" />
-			<Add directory="$(#WX.lib)\gcc_dll" />
+			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="cbkeybinder.cpp">
 			<Option compilerVar="CPP" />
