--- src/plugins/contrib/dragscroll/dragscroll.cpp.orig	2006-08-28 20:58:32.000000000 +0200
+++ src/plugins/contrib/dragscroll/dragscroll.cpp	2006-08-28 22:04:40.000000000 +0200
@@ -1156,5 +1156,18 @@
 }//OnMouseEvent
 #endif //__WXGTK__ scrolling
 // ----------------------------------------------------------------------------
-//   end __WXGTK__ scrolling
+//      MOUSE SCROLLING for __WXMAC__
+// ----------------------------------------------------------------------------
+#ifdef __WXMAC__
+void MyMouseEvents::OnMouseEvent(wxMouseEvent& event)    //Mac
+{
+#warning MyMouseEvents::OnMouseEvent is still TODO
+
+    // pass on the event
+    event.Skip();
+
+}//OnMouseEvent
+#endif //__WXMAC__ scrolling
+// ----------------------------------------------------------------------------
+//   end mouse scrolling
 // ----------------------------------------------------------------------------
--- src/plugins/contrib/keybinder/menuutils.cpp.orig	2006-08-28 20:58:33.000000000 +0200
+++ src/plugins/contrib/keybinder/menuutils.cpp	2006-08-28 22:17:25.000000000 +0200
@@ -217,6 +217,14 @@
 }//RebuildMenuitem
 #endif //#if defined( __WXMSW__ )
 
+#if defined( __WXMAC__ )
+// ----------------------------------------------------------------------------
+void wxMenuCmd::Update(wxMenuItem* pSpecificMenuItem) // for __WXMAC__
+{
+#warning wxMenuCmd::Update is still TODO
+}
+#endif //update for __WXMAC__
+
 // ----------------------------------------------------------------------------
 void wxMenuCmd::Exec(wxObject *origin, wxEvtHandler *client)
 // ----------------------------------------------------------------------------
