--- src/src/appglobals.cpp.orig	2006-08-29 20:04:23.000000000 +0200
+++ src/src/appglobals.cpp	2006-08-29 20:23:24.000000000 +0200
@@ -32,6 +32,8 @@
   const wxString g_AppPlatform = _T("OS/2");
 #elif defined(__WXMAC__)
   const wxString g_AppPlatform = _T("Mac OS X");
+#elif defined(__APPLE__)
+  const wxString g_AppPlatform = _T("Darwin");
 #elif defined(__FreeBSD__)
   const wxString g_AppPlatform = _T("FreeBSD");
 #elif defined(__UNIX__)
