--- src/src/app.cpp.orig	2006-08-29 00:27:54.000000000 +0200
+++ src/src/app.cpp	2006-08-29 00:41:04.000000000 +0200
@@ -665,11 +665,13 @@
     if (!m_Prefix.IsEmpty())
         return m_Prefix;
 
+#ifdef SELFPATH
     // SELFPATH is a macro from prefix.h (binreloc)
     // it returns the absolute filename of us
     // similar to win32 GetModuleFileName()...
     base = wxString(SELFPATH,wxConvUTF8);
     base = wxFileName(base).GetPath();
+#endif
 	if (base.IsEmpty())
 		base = _T(".");
 #endif
