Index: src/plugins/scriptedwizard/resources/wxwidgets/wizard.script
===================================================================
--- src/plugins/scriptedwizard/resources/wxwidgets/wizard.script	(revision 2963)
+++ src/plugins/scriptedwizard/resources/wxwidgets/wizard.script	(working copy)
@@ -116,7 +116,9 @@
         }
 
         if (IsDLL)
-            lib = lib + _T("dll");
+            lib = lib + _T("dll");
+        else
+            lib = lib + _T("lib");
 
         lib = lib + Configuration;
 
@@ -226,8 +228,12 @@
         project.AddIncludeDir(libdir);
         project.AddIncludeDir(WxPath + _T("/contrib/include"));
         project.AddLibDir(LibPath);
-        project.AddLinkLib(LibName);
-        project.AddLinkLib(_T("gdi32"));
+        project.AddLinkLib(LibName);
+        project.AddLinkLib(_T("comctl32"));
+        project.AddLinkLib(_T("gdi32"));
+        project.AddLinkLib(_T("ole32"));
+        project.AddLinkLib(_T("oleaut32"));
+        project.AddLinkLib(_T("uuid"));
         if (GetCompilerFactory().CompilerInheritsFrom(Wizard.GetCompilerID(), _T("msvc*")))
         {
             project.AddCompilerOption(_T("/D__WXMSW__"));
