Index: src/plugins/todo/todolistview.cpp
===================================================================
--- src/plugins/todo/todolistview.cpp	(revision 2964)
+++ src/plugins/todo/todolistview.cpp	(arbetskopia)
@@ -123,6 +123,7 @@
 			GetListControl()->SetItem(idx, 3, item.priorityStr);
 			GetListControl()->SetItem(idx, 4, item.lineStr);
 			GetListControl()->SetItem(idx, 5, item.filename);
+			GetListControl()->SetItemData(idx, i);
 		}
 	}
 	GetListControl()->Thaw();
@@ -375,9 +376,11 @@
 {
     if (event.GetIndex() == -1)
         return;
-    wxString file = m_Items[event.GetIndex()].filename;
-    long int line = m_Items[event.GetIndex()].line;
 
+    unsigned int idx = GetListControl()->GetItemData(event.GetIndex());
+    wxString file = m_Items[idx].filename;
+    long int line = m_Items[idx].line;
+
     if (file.IsEmpty() || line <= 0)
         return;
 
