Index: src/sdk/resources/editor_configuration.xrc
===================================================================
--- src/sdk/resources/editor_configuration.xrc	(revision 2964)
+++ src/sdk/resources/editor_configuration.xrc	(arbetskopia)
@@ -575,7 +575,7 @@
                           </object>
                         </object>
                         <object class="sizeritem">
-                          <flag>wxALIGN_LEFT|wxALIGN_TOP</flag>
+                          <flag>wxALIGN_LEFT|wxALIGN_TOP|wxEXPAND</flag>
                           <object class="wxSlider" name="slCaretPeriod">
                             <value>500</value>
                             <max>500</max>
