Index: src/sdk/printing_types.cpp
===================================================================
--- src/sdk/printing_types.cpp	(revision 2989)
+++ src/sdk/printing_types.cpp	(working copy)
@@ -26,9 +26,15 @@
 
 void DeInitPrinting()
 {
-    delete g_pageSetupData;
-    g_pageSetupData = 0;
+    if (g_pageSetupData)
+    {
+        delete g_pageSetupData;
+        g_pageSetupData = 0;
+    }
 
-    delete g_printData;
-    g_printData = 0;
+    if (g_printData)
+    {
+        delete g_printData;
+        g_printData = 0;
+    }
 }
