Index: src/sdk/pluginmanager.cpp
===================================================================
--- src/sdk/pluginmanager.cpp	(revision 2994)
+++ src/sdk/pluginmanager.cpp	(arbetskopia)
@@ -144,7 +144,7 @@
         wxFileName fname(pluginFilename);
         fname.SetExt(_T("zip"));
         wxString actual = fname.GetFullName();
-        #ifdef __WXGTK__
+        #if defined(__WXGTK__) || defined(__WXMAC__)
         // remove 'lib' prefix from plugin name (if any)
         if (actual.StartsWith(_T("lib")))
             actual.Remove(0, 3);
