Index: src/plugins/compilergcc/compilerDMC.cpp
===================================================================
--- src/plugins/compilergcc/compilerDMC.cpp	(revision 3004)
+++ src/plugins/compilergcc/compilerDMC.cpp	(working copy)
@@ -155,9 +155,9 @@
     // FIXME (hd#1#): should be work on: we need $res_options
     m_Commands[(int)ctCompileObjectCmd] = _T("$compiler -mn -c $options $includes -o$object $file");
     m_Commands[(int)ctCompileResourceCmd] = _T("$rescomp -32 -I$res_includes -o$resource_output $file");
-    m_Commands[(int)ctLinkExeCmd] = _T("$linker /NOLOGO /subsystem:windows $link_options $link_objects, $exe_output, , $libs, , $link_resobjects");
-    m_Commands[(int)ctLinkConsoleExeCmd] = _T("$linker /NOLOGO $link_options $link_objects, $exe_output, , $libs");
-    m_Commands[(int)ctLinkDynamicCmd] = _T("$linker /NOLOGO /subsystem:windows $link_options $link_objects, $exe_output, , $libs, , $link_resobjects");
+    m_Commands[(int)ctLinkExeCmd] = _T("$linker /NOLOGO /subsystem:windows $link_objects, $exe_output, , $libs $link_options, , $link_resobjects");
+    m_Commands[(int)ctLinkConsoleExeCmd] = _T("$linker /NOLOGO $link_objects, $exe_output, , $libs $link_options");
+    m_Commands[(int)ctLinkDynamicCmd] = _T("$linker /NOLOGO /subsystem:windows $link_objects, $exe_output, , $libs $link_options, , $link_resobjects");
     m_Commands[(int)ctLinkStaticCmd] = _T("$lib_linker -c $link_options $static_output $link_objects");
 
     LoadDefaultRegExArray();
