Index: src/plugins/debuggergdb/debugger_defs.h
===================================================================
--- src/plugins/debuggergdb/debugger_defs.h	(revision 3012)
+++ src/plugins/debuggergdb/debugger_defs.h	(working copy)
@@ -149,7 +149,7 @@
     Hex, ///< Variable should be displayed as hexadecimal (e.g. 0xFFFFFFFF).
     Binary, ///< Variable should be displayed as binary (e.g. 00011001).
     Char, ///< Variable should be displayed as a single character (e.g. 'x').
-
+	Float, ///< Variable should be displayed as floating point number (e.g. 14.35)
     // do not remove these
     Last, ///< used for iterations
     Any ///< used for watches searches
Index: src/plugins/debuggergdb/gdb_commands.h
===================================================================
--- src/plugins/debuggergdb/gdb_commands.h	(revision 3012)
+++ src/plugins/debuggergdb/gdb_commands.h	(working copy)
@@ -586,6 +586,7 @@
                     case Hex:           m_Cmd << _T("/x "); break;
                     case Binary:        m_Cmd << _T("/t "); break;
                     case Char:          m_Cmd << _T("/c "); break;
+                    case Float:         m_Cmd << _T("/f "); break;
                     default:            break;
                 }
                 m_Cmd << m_pWatch->keyword;
Index: src/plugins/debuggergdb/resources/edit_watches_dlg.xrc
===================================================================
--- src/plugins/debuggergdb/resources/edit_watches_dlg.xrc	(revision 3012)
+++ src/plugins/debuggergdb/resources/edit_watches_dlg.xrc	(working copy)
@@ -70,6 +70,7 @@
                         <item>Hexadecimal</item>
                         <item>Binary</item>
                         <item>Character</item>
+                        <item>Floating point</item>
                       </content>
                       <selection>0</selection>
                       <dimension>2</dimension>
@@ -95,6 +96,7 @@
                             <object class="wxSpinCtrl" name="spnArrStart">
                               <value>0</value>
                               <min>0</min>
+                              <max>32000</max>
                             </object>
                             <flag>wxGROW</flag>
                           </object>
@@ -110,6 +112,7 @@
                             <object class="wxSpinCtrl" name="spnArrCount">
                               <value>0</value>
                               <min>0</min>
+                              <max>1000</max>
                             </object>
                             <flag>wxGROW</flag>
                           </object>
@@ -192,6 +195,7 @@
                     <item>Hexadecimal</item>
                     <item>Binary</item>
                     <item>Character</item>
+                    <item>Floating point</item>
                   </content>
                   <selection>0</selection>
                   <dimension>2</dimension>
@@ -217,6 +221,7 @@
                         <object class="wxSpinCtrl" name="spnArrStart">
                           <value>0</value>
                           <min>0</min>
+                          <max>32000</max>
                         </object>
                         <flag>wxGROW</flag>
                       </object>
@@ -232,6 +237,7 @@
                         <object class="wxSpinCtrl" name="spnArrCount">
                           <value>0</value>
                           <min>0</min>
+                          <max>1000</max>
                         </object>
                         <flag>wxGROW</flag>
                       </object>
@@ -279,4 +285,4 @@
       </object>
     </object>
   </object>
-</resource>
\ No newline at end of file
+</resource>
