Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 3086)
+++ src/src/main.cpp	(working copy)
@@ -1365,6 +1365,10 @@
             nbstyle = wxFNB_VC71;
             break;
 
+        case 3: // vc8
+            nbstyle = wxFNB_VC8;
+            break;
+
         default:
             nbstyle = 0;
             break;
Index: src/src/resources/env_settings.xrc
===================================================================
--- src/src/resources/env_settings.xrc	(revision 3086)
+++ src/src/resources/env_settings.xrc	(working copy)
@@ -357,7 +357,8 @@
                         <content>
                           <item>Default</item>
                           <item>Fancy gradient</item>
-                          <item>Microsoft (tm) VisualStudio</item>
+                          <item>Microsoft (tm) Visual Studio 7.1</item>
+                          <item>Microsoft (tm) Visual Studio 8</item>
                         </content>
                         <style>wxCB_READONLY</style>
                       </object>
@@ -696,4 +697,4 @@
       </object>
     </object>
   </object>
-</resource>
\ No newline at end of file
+</resource>
