Index: src/sdk/wxscintilla/src/ScintillaWX.cpp
===================================================================
--- src/sdk/wxscintilla/src/ScintillaWX.cpp     (revision 3088)
+++ src/sdk/wxscintilla/src/ScintillaWX.cpp     (working copy)
@@ -518,11 +518,6 @@            
                                
 // This is called by the Editor base class whenever something is selected
 void ScintillaWX::ClaimSelection() {
-#if 0                          
-    // Until wxGTK is able to support using both the primary selection and the
-    // clipboard at the same time I think it causes more problems than it is
-    // worth to implement this method.  Selecting text should not clear the
-    // clipboard.  --Robin
 #ifdef __WXGTK__
     // Put the selected text in the PRIMARY selection
     if (currentPos != anchor) {
@@ -537,7 +532,6 @@
         }
     }
 #endif
-#endif
 }
 
 
