Index: codeblocks/src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- codeblocks/src/plugins/compilergcc/compilergcc.cpp	(revision 3130)
+++ codeblocks/src/plugins/compilergcc/compilergcc.cpp	(working copy)
@@ -3114,7 +3114,9 @@
 
     // first output the standard header blurb
     f.Write(_T("<html>\n"));
+    f.Write(_T("<head>\n"));
     f.Write(_T("<title>") + m_BuildLogTitle + _T("</title>\n"));
+    f.Write(_T("</head>\n"));
     f.Write(_T("<body>\n"));
 
     // use fixed-width font
