Index: src/sdk/globals.cpp
===================================================================
--- src/sdk/globals.cpp	(revision 3180)
+++ src/sdk/globals.cpp	(working copy)
@@ -145,7 +145,9 @@
             ext.IsSameAs(FileFilters::D_EXT) ||
             ext.IsSameAs(FileFilters::F_EXT) ||
             ext.IsSameAs(FileFilters::F77_EXT) ||
-            ext.IsSameAs(FileFilters::F95_EXT)
+            ext.IsSameAs(FileFilters::F90_EXT) ||
+            ext.IsSameAs(FileFilters::F95_EXT) ||
+            ext.IsSameAs(FileFilters::F03_EXT)
        )
         return ftSource;
 
Index: src/sdk/filefilters.h
===================================================================
--- src/sdk/filefilters.h	(revision 3180)
+++ src/sdk/filefilters.h	(working copy)
@@ -68,7 +68,9 @@
     extern const DLLIMPORT wxString D_EXT;
     extern const DLLIMPORT wxString F_EXT;
     extern const DLLIMPORT wxString F77_EXT;
+    extern const DLLIMPORT wxString F90_EXT;
     extern const DLLIMPORT wxString F95_EXT;
+    extern const DLLIMPORT wxString F03_EXT;
     extern const DLLIMPORT wxString CPP_EXT;
     extern const DLLIMPORT wxString C_EXT;
     extern const DLLIMPORT wxString CC_EXT;
@@ -85,7 +87,7 @@
     extern const DLLIMPORT wxString RESOURCE_EXT;
     extern const DLLIMPORT wxString RESOURCEBIN_EXT;
     extern const DLLIMPORT wxString XML_EXT;
-    
+
     // a dot *and* the extension, e.g. ".exe"
     extern const DLLIMPORT wxString WORKSPACE_DOT_EXT;
     extern const DLLIMPORT wxString CODEBLOCKS_DOT_EXT;
@@ -97,7 +99,9 @@
     extern const DLLIMPORT wxString D_DOT_EXT;
     extern const DLLIMPORT wxString F_DOT_EXT;
     extern const DLLIMPORT wxString F77_DOT_EXT;
+    extern const DLLIMPORT wxString F90_DOT_EXT;
     extern const DLLIMPORT wxString F95_DOT_EXT;
+    extern const DLLIMPORT wxString F03_DOT_EXT;
     extern const DLLIMPORT wxString CPP_DOT_EXT;
     extern const DLLIMPORT wxString C_DOT_EXT;
     extern const DLLIMPORT wxString CC_DOT_EXT;
Index: src/sdk/filefilters.cpp
===================================================================
--- src/sdk/filefilters.cpp	(revision 3180)
+++ src/sdk/filefilters.cpp	(working copy)
@@ -138,7 +138,9 @@
 const wxString FileFilters::D_EXT                        = _T("d");
 const wxString FileFilters::F_EXT                        = _T("f");
 const wxString FileFilters::F77_EXT                      = _T("f77");
+const wxString FileFilters::F90_EXT                      = _T("f90");
 const wxString FileFilters::F95_EXT                      = _T("f95");
+const wxString FileFilters::F03_EXT                      = _T("f03");
 const wxString FileFilters::C_EXT                        = _T("c");
 const wxString FileFilters::CC_EXT                       = _T("cc");
 const wxString FileFilters::CXX_EXT                      = _T("cxx");
@@ -172,7 +174,9 @@
 const wxString FileFilters::D_DOT_EXT                        = _T('.') + FileFilters::D_EXT;
 const wxString FileFilters::F_DOT_EXT                        = _T('.') + FileFilters::F_EXT;
 const wxString FileFilters::F77_DOT_EXT                      = _T('.') + FileFilters::F77_EXT;
+const wxString FileFilters::F90_DOT_EXT                      = _T('.') + FileFilters::F90_EXT;
 const wxString FileFilters::F95_DOT_EXT                      = _T('.') + FileFilters::F95_EXT;
+const wxString FileFilters::F03_DOT_EXT                      = _T('.') + FileFilters::F03_EXT;
 const wxString FileFilters::C_DOT_EXT                        = _T('.') + FileFilters::C_EXT;
 const wxString FileFilters::CC_DOT_EXT                       = _T('.') + FileFilters::CC_EXT;
 const wxString FileFilters::CXX_DOT_EXT                      = _T('.') + FileFilters::CXX_EXT;
