--- src/plugins/contrib/keybinder/keybinder.cpp.orig	2006-11-05 23:17:49.000000000 +0100
+++ src/plugins/contrib/keybinder/keybinder.cpp	2006-11-06 00:59:18.000000000 +0100
@@ -197,12 +197,16 @@
 
 	case WXK_NUMPAD_HOME:
 		res << wxT("HOME"); break;
+#if WXK_NUMPAD_PRIOR != WXK_NUMPAD_PAGEUP
 	case WXK_NUMPAD_PRIOR:
 		res << wxT("PgUp"); break;
+#endif
 	case WXK_NUMPAD_PAGEUP:
 		res << wxT("PAGEUP"); break;
+#if WXK_NUMPAD_NEXT != WXK_NUMPAD_PAGEDOWN
 	case WXK_NUMPAD_NEXT:
 		res << wxT("PgDn"); break;
+#endif
 	case WXK_NUMPAD_PAGEDOWN:
 		res << wxT("PAGEDOWN"); break;
 	case WXK_NUMPAD_END:
@@ -344,11 +348,15 @@
         res << wxT("MENU"); break;
 	case WXK_CAPITAL:
         res << wxT("CAPITAL"); break;
+#if WXK_PRIOR != WXK_PAGEUP
 	case WXK_PRIOR:
         res << wxT("PgUp"); break;
-	case WXK_NEXT:
+#endif
+#if WXK_NEXT != WXK_PAGEDOWN
+    case WXK_NEXT:
         res << wxT("PgDn"); break;
-	case WXK_END:
+#endif
+case WXK_END:
         res << wxT("END"); break;
 	case WXK_HOME:
         res << wxT("HOME"); break;
