--- src/plugins/compilergcc//compilergcc.cpp.orig	2006-11-05 23:17:53.000000000 +0100
+++ src/plugins/compilergcc//compilergcc.cpp	2006-11-06 19:17:04.000000000 +0100
@@ -1407,7 +1407,7 @@
 
         // invoke Makefile generation
         if (temporary)
-            m_LastTempMakefile = wxFileName::CreateTempFileName(_T("cbmk"), 0L);
+            m_LastTempMakefile = wxFileName::CreateTempFileName(_T("cbmk"), (wxFile*) 0L);
         else
         {
             m_LastTempMakefile = makefile;
