Index: src/src/wxAUI/manager.h
===================================================================
--- src/src/wxAUI/manager.h	(revision 3186)
+++ src/src/wxAUI/manager.h	(working copy)
@@ -12,6 +12,15 @@
 #ifndef __WXAUI_WXAUI_H
 #define __WXAUI_WXAUI_H
 
+#include <wx/version.h>
+#if wxCHECK_VERSION(2, 7, 0)
+#include <wx/aui/aui.h>
+#define wxFrameManager wxAuiManager
+#define wxDockArt wxAuiDockArt
+#define wxPaneInfo wxAuiPaneInfo
+#define wxPaneInfoArray wxAuiPaneInfoArray
+#else // VERSION(2, 6, 3)
+
 enum wxFrameManagerDock
 {
     wxAUI_DOCK_NONE = 0,
@@ -770,4 +779,6 @@
    wx__DECLARE_EVT0(wxEVT_AUI_PANEBUTTON, wxFrameManagerEventHandler(func))
 
 
+#endif // wxCHECK_VERSION
+
 #endif
Index: src/src/wxAUI/manager.cpp
===================================================================
--- src/src/wxAUI/manager.cpp	(revision 3186)
+++ src/src/wxAUI/manager.cpp	(working copy)
@@ -14,6 +14,7 @@
 #include <wx/image.h>
 #include "manager.h"
 
+#if ! wxCHECK_VERSION(2, 7, 0)
 
 // -- various array and event implementations --
 
@@ -4286,3 +4298,5 @@
         Update();
     }
 }
+
+#endif // wxCHECK_VERSION
