Index: src/sdk/manager.cpp
===================================================================
--- src/sdk/manager.cpp	(revision 3222)
+++ src/sdk/manager.cpp	(working copy)
@@ -326,12 +326,12 @@
             wxMemoryFSHandler::AddFile(file, buf, len);
         }
         wxXmlResource::Get()->Load(memoryFile);
-        delete buf;
+        delete[] buf;
         return true;
     }
     catch (...)
     {
-        delete buf;
+        delete[] buf;
         return false;
     }
 }
