Index: src/sdk/pluginmanager.cpp
===================================================================
--- src/sdk/pluginmanager.cpp	(revision 3240)
+++ src/sdk/pluginmanager.cpp	(working copy)
@@ -753,6 +753,8 @@
 #endif
 
     int count = 0;
+    if(!wxDirExists(path))
+    	return count;
     wxDir dir(path);
 
     if (!dir.IsOpened())
Index: src/sdk/configmanager.cpp
===================================================================
--- src/sdk/configmanager.cpp	(revision 3240)
+++ src/sdk/configmanager.cpp	(working copy)
@@ -1368,8 +1368,8 @@
         ConfigManager::data_path_global = wxStandardPathsBase::Get().GetDataDir();
 
     CreateDirRecursively(ConfigManager::config_folder);
-    CreateDirRecursively(ConfigManager::data_path_user   + _T("/plugins"));
-    CreateDirRecursively(ConfigManager::data_path_global + _T("/plugins"));
+    CreateDirRecursively(ConfigManager::data_path_user   + _T("/plugins/"));
+    CreateDirRecursively(ConfigManager::data_path_global + _T("/plugins/"));
     CreateDir(ConfigManager::data_path_user   + _T("/scripts"));
     CreateDir(ConfigManager::data_path_global + _T("/scripts"));
 
