Index: src/sdk/projectloader.cpp
===================================================================
--- src/sdk/projectloader.cpp   (révision 3240)
+++ src/sdk/projectloader.cpp   (copie de travail)
@@ -32,6 +32,7 @@
 #include "projectloader.h"
 #include "projectloader_hooks.h"
 #include "annoyingdialog.h"
+#include "configmanager.h"

 ProjectLoader::ProjectLoader(cbProject* project)
    : m_pProject(project),