Index: src/sdk/resources/project_options.xrc
===================================================================
--- src/sdk/resources/project_options.xrc	(revision 3287)
+++ src/sdk/resources/project_options.xrc	(working copy)
@@ -426,7 +426,7 @@
                         <label>Build target files:</label>
                         <orient>wxVERTICAL</orient>
                         <object class="sizeritem">
-                          <object class="wxCheckList" name="lstFiles">
+                          <object class="wxCheckListBox" name="lstFiles">
                             <content/>
                           </object>
                           <option>1</option>
Index: src/sdk/resources/projectfile_options.xrc
===================================================================
--- src/sdk/resources/projectfile_options.xrc	(revision 3287)
+++ src/sdk/resources/projectfile_options.xrc	(working copy)
@@ -30,7 +30,7 @@
                   <object class="wxStaticBoxSizer">
                     <label>Belongs in targets:</label>
                     <object class="sizeritem">
-                      <object class="wxCheckList" name="lstTargets">
+                      <object class="wxCheckListBox" name="lstTargets">
                         <content/>
                         <size>-1,150</size>
                       </object>
Index: src/plugins/codecompletion/resources/insert_class_method.xrc
===================================================================
--- src/plugins/codecompletion/resources/insert_class_method.xrc	(revision 3287)
+++ src/plugins/codecompletion/resources/insert_class_method.xrc	(working copy)
@@ -34,7 +34,7 @@
             <object class="wxBoxSizer">
               <orient>wxVERTICAL</orient>
               <object class="sizeritem">
-                <object class="wxCheckList" name="chklstMethods">
+                <object class="wxCheckListBox" name="chklstMethods">
                   <content/>
                 </object>
                 <option>1</option>
Index: src/plugins/compilergcc/resources/compiler_options.xrc
===================================================================
--- src/plugins/compilergcc/resources/compiler_options.xrc	(revision 3287)
+++ src/plugins/compilergcc/resources/compiler_options.xrc	(working copy)
@@ -153,7 +153,7 @@
                                     <option>1</option>
                                     <flag>wxALL|wxEXPAND</flag>
                                     <border>8</border>
-                                    <object class="wxCheckList" name="lstCompilerOptions">
+                                    <object class="wxCheckListBox" name="lstCompilerOptions">
                                       <content/>
                                       <style>wxLB_HSCROLL</style>
                                     </object>
Index: src/src/resources/manage_assocs.xrc
===================================================================
--- src/src/resources/manage_assocs.xrc	(revision 3287)
+++ src/src/resources/manage_assocs.xrc	(working copy)
@@ -14,7 +14,7 @@
         <object class="sizeritem">
             <flag>wxGROW|wxALIGN_CENTER_VERTICAL|wxALL</flag>
             <border>5</border>
-            <object class="wxCheckList" name="checkList">
+            <object class="wxCheckListBox" name="checkList">
                 <size>320,400</size>
             </object>
         </object>
Index: src/src/resources/env_settings.xrc
===================================================================
--- src/src/resources/env_settings.xrc	(revision 3287)
+++ src/src/resources/env_settings.xrc	(working copy)
@@ -593,7 +593,7 @@
                   <option>1</option>
                   <border>8</border>
                   <flag>wxLEFT|wxALIGN_LEFT|wxALIGN_TOP|wxEXPAND</flag>
-                  <object class="wxCheckList" name="chkDialogs">
+                  <object class="wxCheckListBox" name="chkDialogs">
                     <style></style>
                     <content/>
                   </object>
