--- C:/Required/svn-win32-1.3.0/bin/trunk/src/plugins/scriptedwizard/wizpage.cpp	Mon Nov 27 23:26:11 2006
+++ C:/Temp/wizpage.cpp	Tue Nov 28 17:42:52 2006
@@ -452,7 +452,7 @@
     wxArrayString valids = GetArrayFromString(validCompilerIDs, _T(";"), true);
     wxString def = compilerID;
     if (def.IsEmpty())
-        def = CompilerFactory::GetDefaultCompiler()->GetName();
+		def = CompilerFactory::GetDefaultCompilerID();
     int id = 0;
     wxComboBox* cmb = m_pCompilerPanel->GetCompilerCombo();
     cmb->Clear();
@@ -466,7 +466,7 @@
             {
                 cmb->Append(compiler->GetName());
                 if (compiler->GetID().IsSameAs(def))
-                    id = cmb->GetCount();
+                    id = (cmb->GetCount() - 1) < 0 ? 0 : (cmb->GetCount() - 1) ;
                 break;
             }
         }