Index: src/src/resources/env_settings.xrc
===================================================================
--- src/src/resources/env_settings.xrc	(revision 3334)
+++ src/src/resources/env_settings.xrc	(working copy)
@@ -337,7 +337,7 @@
             </object>
           </object>
           <object class="listbookpage">
-            <label>Notebooks appearence</label>
+            <label>Notebooks appearance</label>
             <object class="wxPanel" name="pblNotebooks">
               <object class="wxBoxSizer">
                 <orient>wxVERTICAL</orient>
@@ -665,4 +665,4 @@
       </object>
     </object>
   </object>
-</resource>
\ No newline at end of file
+</resource>
