Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 3336)
+++ src/src/main.cpp	(working copy)
@@ -3336,19 +3336,19 @@
 void MainFrame::OnToggleStatusBar(wxCommandEvent& event)
 {
     wxStatusBar* sb = GetStatusBar();
+
     if (sb)
     {
         sb->Hide();
-        SetStatusBar(NULL);
+
+        SetStatusBar(0);
+
+        sb->Destroy();
     }
+
     else
         DoCreateStatusBar();
 
-// under Windows, the statusbar doesn't disappear immediately...
-#ifdef __WXMSW__
-    SendSizeEvent(); // make sure everything is laid out properly
-    Manager::ProcessPendingEvents(); // Tell wxWindows to process the event we just sent
-#endif // __WXMSW__
     DoUpdateStatusBar();
 }
 
