Index: src/plugins/scriptedwizard/resources/common_functions.script
===================================================================
--- src/plugins/scriptedwizard/resources/common_functions.script	(revision 3373)
+++ src/plugins/scriptedwizard/resources/common_functions.script	(working copy)
@@ -100,7 +100,7 @@
     }
     else if (GetCompilerFactory().CompilerInheritsFrom(compilerID, _T("dmd")))
     {
-        base.AddCompilerOption(_T("-o"));
+        base.AddCompilerOption(_T("-O"));
     }
     else if (GetCompilerFactory().CompilerInheritsFrom(compilerID, _T("icc")))
     {
Index: src/sdk/compilercommandgenerator.cpp
===================================================================
--- src/sdk/compilercommandgenerator.cpp	(revision 3373)
+++ src/sdk/compilercommandgenerator.cpp	(working copy)
@@ -46,14 +46,14 @@
 
     if (!project)
     {
-        m_DefOutput[0] = SetupOutputFilenames(compiler, 0);
-        m_Inc[0] = SetupIncludeDirs(compiler, 0);
-        m_Lib[0] = SetupLibrariesDirs(compiler, 0);
-        m_RC[0] = SetupResourceIncludeDirs(compiler, 0);
-        m_CFlags[0] = SetupCompilerOptions(compiler, 0);
-        m_LDFlags[0] = SetupLinkerOptions(compiler, 0);
-        m_LDAdd[0] = SetupLinkLibraries(compiler, 0);
-        m_RCFlags[0] = SetupResourceCompilerOptions(compiler, 0);
+        m_DefOutput[static_cast<ProjectBuildTarget*>(0)] = SetupOutputFilenames(compiler, 0);
+        m_Inc[static_cast<ProjectBuildTarget*>(0)] = SetupIncludeDirs(compiler, 0);
+        m_Lib[static_cast<ProjectBuildTarget*>(0)] = SetupLibrariesDirs(compiler, 0);
+        m_RC[static_cast<ProjectBuildTarget*>(0)] = SetupResourceIncludeDirs(compiler, 0);
+        m_CFlags[static_cast<ProjectBuildTarget*>(0)] = SetupCompilerOptions(compiler, 0);
+        m_LDFlags[static_cast<ProjectBuildTarget*>(0)] = SetupLinkerOptions(compiler, 0);
+        m_LDAdd[static_cast<ProjectBuildTarget*>(0)] = SetupLinkLibraries(compiler, 0);
+        m_RCFlags[static_cast<ProjectBuildTarget*>(0)] = SetupResourceCompilerOptions(compiler, 0);
         return;
     }
     else
