Index: src/plugins/contrib/profiler/cbprofiler.cbp
===================================================================
--- src/plugins/contrib/profiler/cbprofiler.cbp	(revision 3445)
+++ src/plugins/contrib/profiler/cbprofiler.cbp	(working copy)
@@ -51,7 +51,7 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="codeblocks" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/copystrings/copystrings.cbp
===================================================================
--- src/plugins/contrib/copystrings/copystrings.cbp	(revision 3445)
+++ src/plugins/contrib/copystrings/copystrings.cbp	(working copy)
@@ -53,7 +53,7 @@
 		<Linker>
 			<Add library="codeblocks" />
 			<Add library="wxscintilla" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#CB)\devel" />
Index: src/plugins/contrib/envvars/envvars.cbp
===================================================================
--- src/plugins/contrib/envvars/envvars.cbp	(revision 3445)
+++ src/plugins/contrib/envvars/envvars.cbp	(working copy)
@@ -51,7 +51,7 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="codeblocks" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.LIB)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/regex_testbed/RegExTestbed.cbp
===================================================================
--- src/plugins/contrib/regex_testbed/RegExTestbed.cbp	(revision 3445)
+++ src/plugins/contrib/regex_testbed/RegExTestbed.cbp	(working copy)
@@ -60,7 +60,7 @@
 			<Add directory="$(#WX.include)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 			<Add directory="$(#CB.lib)" />
Index: src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
===================================================================
--- src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(revision 3445)
+++ src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(working copy)
@@ -68,7 +68,7 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="codeblocks" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="bzip2" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 		</Linker>
Index: src/plugins/contrib/source_exporter/Exporter.cbp
===================================================================
--- src/plugins/contrib/source_exporter/Exporter.cbp	(revision 3445)
+++ src/plugins/contrib/source_exporter/Exporter.cbp	(working copy)
@@ -71,7 +71,7 @@
 			<Add directory="$(#WX.include)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
 			<Add library="wxscintilla" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/cb_koders/cb_koders.cbp
===================================================================
--- src/plugins/contrib/cb_koders/cb_koders.cbp	(revision 3445)
+++ src/plugins/contrib/cb_koders/cb_koders.cbp	(working copy)
@@ -56,7 +56,7 @@
 			<Add directory="$(#WX.include)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
 			<Add library="wxscintilla" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/dragscroll/DragScroll.cbp
===================================================================
--- src/plugins/contrib/dragscroll/DragScroll.cbp	(revision 3445)
+++ src/plugins/contrib/dragscroll/DragScroll.cbp	(working copy)
@@ -51,7 +51,7 @@
 		<Linker>
 			<Add library="codeblocks" />
 			<Add library="wxscintilla" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/lib_finder/lib_finder.cbp
===================================================================
--- src/plugins/contrib/lib_finder/lib_finder.cbp	(revision 3445)
+++ src/plugins/contrib/lib_finder/lib_finder.cbp	(working copy)
@@ -56,7 +56,7 @@
 			<Add directory="$(#WX.include)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add library="txml" />
 			<Add library="codeblocks" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/symtab/symtab.cbp
===================================================================
--- src/plugins/contrib/symtab/symtab.cbp	(revision 3445)
+++ src/plugins/contrib/symtab/symtab.cbp	(working copy)
@@ -56,7 +56,7 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="codeblocks" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/codesnippets/codesnippets.cbp
===================================================================
--- src/plugins/contrib/codesnippets/codesnippets.cbp	(revision 3445)
+++ src/plugins/contrib/codesnippets/codesnippets.cbp	(working copy)
@@ -55,7 +55,7 @@
 			<Add directory="$(#wx.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 		</Compiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add library="wxscintilla" />
 			<Add library="txml" />
 			<Add library="codeblocks" />
Index: src/plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith.cbp	(revision 3445)
+++ src/plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -20,7 +20,7 @@
 					<Add library="txml" />
 					<Add library="codeblocks" />
 					<Add library="libpropgrid.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="libwxscintilla.a" />
 				</Linker>
 			</Target>
Index: src/plugins/contrib/byogames/byogames.cbp
===================================================================
--- src/plugins/contrib/byogames/byogames.cbp	(revision 3445)
+++ src/plugins/contrib/byogames/byogames.cbp	(working copy)
@@ -53,7 +53,7 @@
 			<Add directory="$(#WX.include)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 			<Add directory="$(#CB.lib)" />
Index: src/plugins/contrib/codestat/codestat.cbp
===================================================================
--- src/plugins/contrib/codestat/codestat.cbp	(revision 3445)
+++ src/plugins/contrib/codestat/codestat.cbp	(working copy)
@@ -52,7 +52,7 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="libcodeblocks.a" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.LIB)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/help_plugin/help-plugin.cbp
===================================================================
--- src/plugins/contrib/help_plugin/help-plugin.cbp	(revision 3445)
+++ src/plugins/contrib/help_plugin/help-plugin.cbp	(working copy)
@@ -47,7 +47,7 @@
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="libcodeblocks.a" />
 					<Add library="libwxscintilla.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add directory="$(#CB)\devel" />
 					<Add directory="$(#CB.lib)" />
 					<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/plugins/contrib/keybinder/keybinder.cbp
===================================================================
--- src/plugins/contrib/keybinder/keybinder.cbp	(revision 3445)
+++ src/plugins/contrib/keybinder/keybinder.cbp	(working copy)
@@ -50,7 +50,7 @@
 		</Compiler>
 		<Linker>
 			<Add library="codeblocks" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#CB)\devel" />
 			<Add directory="$(#CB.lib)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
Index: src/tools/cb_share_config/cb_share_config.cbp
===================================================================
--- src/tools/cb_share_config/cb_share_config.cbp	(revision 3445)
+++ src/tools/cb_share_config/cb_share_config.cbp	(working copy)
@@ -47,7 +47,7 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="txml" />
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 			<Add directory="..\..\sdk\tinyxml" />
 		</Linker>
Index: src/CodeBlocks.cbp
===================================================================
--- src/CodeBlocks.cbp	(revision 3445)
+++ src/CodeBlocks.cbp	(working copy)
@@ -99,7 +99,7 @@
 					<Add directory="sdk\wxscintilla\include" />
 				</ResourceCompiler>
 				<Linker>
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="gdi32" />
 				</Linker>
 			</Target>
@@ -231,7 +231,7 @@
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="libcodeblocks.a" />
 					<Add library="libwxscintilla.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add directory="devel" />
 				</Linker>
 			</Target>
@@ -267,7 +267,7 @@
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="libcodeblocks.a" />
 					<Add library="libwxscintilla.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="depslib" />
 					<Add directory="devel" />
 					<Add directory="plugins\compilergcc\depslib" />
@@ -293,7 +293,7 @@
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="libcodeblocks.a" />
 					<Add library="libwxscintilla.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add directory="devel" />
 				</Linker>
 			</Target>
@@ -316,7 +316,7 @@
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="libcodeblocks.a" />
 					<Add library="libwxscintilla.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add directory="devel" />
 				</Linker>
 			</Target>
@@ -338,7 +338,7 @@
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="libcodeblocks.a" />
 					<Add library="libwxscintilla.a" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add directory="devel" />
 				</Linker>
 			</Target>
@@ -359,7 +359,7 @@
 					<Add option="-Wl,--enable-auto-image-base" />
 					<Add option="-Wl,--add-stdcall-alias" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw26$(WX_SUFFIX)" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add directory="devel" />
 				</Linker>
 			</Target>
@@ -475,7 +475,7 @@
 			<Add directory="$(#WX.include)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add library="wxmsw26$(WX_SUFFIX)" />
+			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="sdk\tinyxml" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)" />
 		</Linker>
