Index: src/sdk/pipedprocess.cpp
===================================================================
--- src/sdk/pipedprocess.cpp	(revision 3173)
+++ src/sdk/pipedprocess.cpp	(working copy)
@@ -75,7 +75,11 @@
                     return wxEOT;
                 if (m_allowMBconversion)
                 {
+#if wxCHECK_VERSION(2, 8, 0)
+                    int retlen = (int) m_conv->MB2WC(wbuf, m_lastBytes, 2); // returns -1 for failure
+#else
                     int retlen = (int) m_conv.MB2WC(wbuf, m_lastBytes, 2); // returns -1 for failure
+#endif
                     if(retlen >= 0) // res == 0 could happen for '\0' char
                         return wbuf[0];
                 }
