Index: src/sdk/compiler.cpp
===================================================================
--- src/sdk/compiler.cpp	(revision 3388)
+++ src/sdk/compiler.cpp	(working copy)
@@ -243,6 +243,8 @@
     cfg->Write(tmp + _T("/switches/linkerNeedsLibPrefix"), m_Switches.linkerNeedsLibPrefix);
     cfg->Write(tmp + _T("/switches/linkerNeedsLibExtension"), m_Switches.linkerNeedsLibExtension);
     cfg->Write(tmp + _T("/switches/forceFwdSlashes"), m_Switches.forceFwdSlashes);
+    cfg->Write(tmp + _T("/switches/supportsPCH"), m_Switches.supportsPCH);
+    cfg->Write(tmp + _T("/switches/pchExtension"), m_Switches.PCHExtension);
 
     // regexes
     cfg->DeleteSubPath(tmp + _T("/regex"));
@@ -350,6 +352,8 @@
     m_Switches.linkerNeedsLibPrefix = cfg->ReadBool(tmp + _T("/switches/linkerNeedsLibPrefix"), m_Switches.linkerNeedsLibPrefix);
     m_Switches.linkerNeedsLibExtension = cfg->ReadBool(tmp + _T("/switches/linkerNeedsLibExtension"), m_Switches.linkerNeedsLibExtension);
     m_Switches.forceFwdSlashes = cfg->ReadBool(tmp + _T("/switches/forceFwdSlashes"), m_Switches.forceFwdSlashes);
+    m_Switches.supportsPCH = cfg->ReadBool(tmp + _T("/switches/supportsPCH"), m_Switches.supportsPCH);
+    m_Switches.PCHExtension = cfg->Read(tmp + _T("/switches/pchExtension"), m_Switches.PCHExtension);
 
     // regexes
     wxString group;
