Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 3412)
+++ src/src/main.cpp	(working copy)
@@ -1144,6 +1144,8 @@
     }
     if (select)
         DoSelectLayout(name);
+
+    m_LastLayoutName = name;
 }
 
 bool MainFrame::DoCheckCurrentLayoutForChanges(bool canCancel)
@@ -3010,8 +3012,9 @@
         {
             m_LayoutViews[gDefaultLayout] = gDefaultLayoutData;
             LoadViewLayout(gDefaultLayout);
-            return;
         }
+
+        return;
     }
 
     if (cbMessageBox(wxString::Format(_("Are you really sure you want to delete the layout '%s'?"), m_LastLayoutName.c_str()),
