Index: src/plugins/contrib/wxSmith/defwidgets/wxslistbook.cpp
===================================================================
--- src/plugins/contrib/wxSmith/defwidgets/wxslistbook.cpp	(revision 3438)
+++ src/plugins/contrib/wxSmith/defwidgets/wxslistbook.cpp	(working copy)
@@ -204,7 +204,11 @@
 	if ( GetPreview() && event.LeftDown() )
 	{
 	    wxListbook* Preview = (wxListbook*)GetPreview();
+#if wxCHECK_VERSION(2, 8, 0)
+        int Hit = wxStaticCast(Preview, wxBookCtrlBase)->HitTest(wxPoint(event.GetX(),event.GetY()));
+#else
 	    int Hit = Preview->HitTest(wxPoint(event.GetX(),event.GetY()));
+#endif
         if ( Hit != wxNOT_FOUND )
         {
             CurrentSelection = GetChild(Hit);
