Index: src/src/resources/main_menu.xrc
===================================================================
--- src/src/resources/main_menu.xrc	(revision 3482)
+++ src/src/resources/main_menu.xrc	(working copy)
@@ -54,7 +54,7 @@
         </object>
       </object>
       <object class="wxMenu" name="idFileReopenFile">
-        <label>&amp;Recent files</label>
+        <label>R&amp;ecent files</label>
         <help>Open a recent file</help>
         <object class="wxMenuItem" name="idFileOpenRecentFileClearHistory">
           <label>&amp;Clear history</label>
@@ -101,7 +101,7 @@
         <help>Save the active file under a different name</help>
       </object>
       <object class="wxMenuItem" name="idFileSaveAllFiles">
-        <label>&amp;Save a&amp;ll files</label>
+        <label>Save a&amp;ll files</label>
         <accel>Ctrl-Shift-S</accel>
         <help>Save all modified files</help>
       </object>
