Index: src/sdk/resources/edit_tool.xrc
===================================================================
--- src/sdk/resources/edit_tool.xrc	(Revision 3541)
+++ src/sdk/resources/edit_tool.xrc	(Arbeitskopie)
@@ -107,6 +107,10 @@
 $${PROJECT_DIR}
 $${ACTIVE_EDITOR_FILENAME}
 $${ALL_PROJECT_FILES}
+$${TARGET_OUTPUT_DIR}
+$${TARGET_NAME}
+$${TARGET_OUTPUT_BASENAME}
+$${TARGET_OUTPUT_FILE}
 $${MAKEFILE}</label>
             </object>
             <option>1</option>
