Index: sdk/editormanager.cpp
===================================================================
--- sdk/editormanager.cpp	(revision 3557)
+++ sdk/editormanager.cpp	(arbetskopia)
@@ -1376,6 +1376,11 @@
         if (!data->directionDown)       // up
             data->end = 0;
 
+        // selected text, if user has deslected since last, then change scope
+        if (data->scope == 1 &&
+            control->GetSelectionStart()==control->GetSelectionEnd())
+                data->scope = 0;
+                
         if (data->scope == 1) // selected text
         {
             if(data->NewSearch)
