Index: CodeBlocks.cbp
===================================================================
--- CodeBlocks.cbp	(revision 3601)
+++ CodeBlocks.cbp	(working copy)
@@ -447,7 +447,7 @@
 				</Linker>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+               <Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -471,7 +471,6 @@
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="include\wxscintilla\include" />
Index: plugins/contrib/byogames/byogames.cbp
===================================================================
--- plugins/contrib/byogames/byogames.cbp	(revision 3601)
+++ plugins/contrib/byogames/byogames.cbp	(working copy)
@@ -12,7 +12,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -34,7 +34,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
Index: plugins/contrib/cb_koders/cb_koders.cbp
===================================================================
--- plugins/contrib/cb_koders/cb_koders.cbp	(revision 3601)
+++ plugins/contrib/cb_koders/cb_koders.cbp	(working copy)
@@ -16,7 +16,7 @@
 				<Option projectResourceIncludeDirsRelation="0" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -38,7 +38,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
Index: plugins/contrib/codesnippets/codesnippets.cbp
===================================================================
--- plugins/contrib/codesnippets/codesnippets.cbp	(revision 3601)
+++ plugins/contrib/codesnippets/codesnippets.cbp	(working copy)
@@ -24,7 +24,7 @@
 				<Option compiler="gcc" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -46,7 +46,6 @@
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#cb.include)" />
			<Add directory="$(#cb.include)\\wxscintilla\include" />
 			<Add directory="$(#wx.include)" />
Index: plugins/contrib/codestat/codestat.cbp
===================================================================
--- plugins/contrib/codestat/codestat.cbp	(revision 3601)
+++ plugins/contrib/codestat/codestat.cbp	(working copy)
@@ -14,7 +14,7 @@
 				<Option projectCompilerOptionsRelation="0" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -36,7 +36,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB.INCLUDE)" />
 			<Add directory="$(#CB.INCLUDE)\wxscintilla\include" />
 			<Add directory="$(#WX.INCLUDE)" />
Index: plugins/contrib/copystrings/copystrings.cbp
===================================================================
--- plugins/contrib/copystrings/copystrings.cbp	(revision 3601)
+++ plugins/contrib/copystrings/copystrings.cbp	(working copy)
@@ -11,7 +11,7 @@
 				<Option compiler="gcc" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -34,7 +34,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
Index: plugins/contrib/devpak_plugin/DevPakPlugin.cbp
===================================================================
--- plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(revision 3601)
+++ plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(working copy)
@@ -35,7 +35,6 @@
 					<Add option="-DWXUSINGDLL" />
 					<Add option="-DcbDEBUG" />
 					<Add option="-DTIXML_USE_STL" />
-					<Add option="-DwxUSE_UNICODE" />
 					<Add directory="..\..\..\include" />
 					<Add directory="..\..\..\include\wxscintilla\include" />
 				</Compiler>
@@ -49,7 +48,7 @@
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
Index: plugins/contrib/dragscroll/DragScroll.cbp
===================================================================
--- plugins/contrib/dragscroll/DragScroll.cbp	(revision 3601)
+++ plugins/contrib/dragscroll/DragScroll.cbp	(working copy)
@@ -13,7 +13,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -37,7 +37,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
Index: plugins/contrib/envvars/envvars.cbp
===================================================================
--- plugins/contrib/envvars/envvars.cbp	(revision 3601)
+++ plugins/contrib/envvars/envvars.cbp	(working copy)
@@ -13,7 +13,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -35,7 +35,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB.INCLUDE)" />
 			<Add directory="$(#CB.INCLUDE)\wxscintilla\include" />
 			<Add directory="$(#WX.INCLUDE)" />
Index: plugins/contrib/help_plugin/help-plugin.cbp
===================================================================
--- plugins/contrib/help_plugin/help-plugin.cbp	(revision 3601)
+++ plugins/contrib/help_plugin/help-plugin.cbp	(working copy)
@@ -28,7 +28,6 @@
 					<Add option="-DWXUSINGDLL" />
 					<Add option="-DcbDEBUG" />
 					<Add option="-DTIXML_USE_STL" />
-					<Add option="-DwxUSE_UNICODE" />
 					<Add directory="$(#WX.include)" />
 					<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 					<Add directory="$(#WX)\contrib\include" />
@@ -54,7 +53,7 @@
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
Index: plugins/contrib/keybinder/keybinder.cbp
===================================================================
--- plugins/contrib/keybinder/keybinder.cbp	(revision 3601)
+++ plugins/contrib/keybinder/keybinder.cbp	(working copy)
@@ -26,11 +26,10 @@
 					<Add option="-DWXUSINGDLL" />
 					<Add option="-DcbDEBUG" />
 					<Add option="-DTIXML_USE_STL" />
-					<Add option="-DwxUSE_UNICODE" />
 				</Compiler>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
Index: plugins/contrib/lib_finder/lib_finder.cbp
===================================================================
--- plugins/contrib/lib_finder/lib_finder.cbp	(revision 3601)
+++ plugins/contrib/lib_finder/lib_finder.cbp	(working copy)
@@ -16,7 +16,7 @@
 				<Option projectResourceIncludeDirsRelation="0" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -39,7 +39,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
Index: plugins/contrib/profiler/cbprofiler.cbp
===================================================================
--- plugins/contrib/profiler/cbprofiler.cbp	(revision 3601)
+++ plugins/contrib/profiler/cbprofiler.cbp	(working copy)
@@ -13,7 +13,7 @@
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -35,7 +35,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB.include)" />
 			<Add directory="$(#CB.include)\wxscintilla\include" />
 			<Add directory="$(#WX.include)" />
Index: plugins/contrib/regex_testbed/RegExTestbed.cbp
===================================================================
--- plugins/contrib/regex_testbed/RegExTestbed.cbp	(revision 3601)
+++ plugins/contrib/regex_testbed/RegExTestbed.cbp	(working copy)
@@ -22,7 +22,7 @@
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -42,7 +42,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add option="-DWX_PRECOMP" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DSUPPORT_UTF8" />
 			<Add directory="$(#WX.include)" />
Index: plugins/contrib/source_exporter/Exporter.cbp
===================================================================
--- plugins/contrib/source_exporter/Exporter.cbp	(revision 3601)
+++ plugins/contrib/source_exporter/Exporter.cbp	(working copy)
@@ -31,7 +31,7 @@
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -53,7 +53,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#CB.include)" />
Index: plugins/contrib/symtab/symtab.cbp
===================================================================
--- plugins/contrib/symtab/symtab.cbp	(revision 3601)
+++ plugins/contrib/symtab/symtab.cbp	(working copy)
@@ -19,7 +19,7 @@
 				</Compiler>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -41,7 +41,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DTIXML_USE_STL" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#CB.include)" />
 			<Add directory="$(#CB.include)\wxscintilla\include" />
 			<Add directory="$(#WX.include)" />
Index: plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- plugins/contrib/wxSmith/wxSmith.cbp	(revision 3601)
+++ plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -20,7 +20,7 @@
 				</Linker>
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -44,7 +44,6 @@
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
 			<Add option="-DBUILDING_PLUGIN" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#CB.include)" />
 			<Add directory="$(#CB.include)\wxscintilla\include" />
Index: tools/cb_share_config/cb_share_config.cbp
===================================================================
--- tools/cb_share_config/cb_share_config.cbp	(revision 3601)
+++ tools/cb_share_config/cb_share_config.cbp	(working copy)
@@ -13,7 +13,7 @@
 				<Option compiler="gcc" />
 			</Target>
 			<Environment>
-				<Variable name="WX_SUFFIX" value="u" />
+				<Variable name="WX_SUFFIX" value="" />
 				<Variable name="WX_CFG" value="" />
 			</Environment>
 		</Build>
@@ -31,7 +31,6 @@
 			<Add option="-D__GNUWIN32__" />
 			<Add option="-D__WXMSW__" />
 			<Add option="-DHAVE_W32API_H" />
-			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DTIXML_USE_STL" />
 			<Add directory="$(#WX.include)" />
