Index: src/plugins/codecompletion/classbrowserbuilderthread.cpp
===================================================================
--- src/plugins/codecompletion/classbrowserbuilderthread.cpp	(revision 3614)
+++ src/plugins/codecompletion/classbrowserbuilderthread.cpp	(working copy)
@@ -104,10 +104,12 @@

         if (TestDestroy() || Manager::IsAppShuttingDown())
         {
+#ifdef __WXGTK__
 			if(!::wxIsMainThread())
 			{
 				::wxMutexGuiLeave();
 			}
+#endif // __WXGTK__
             break;
         }

