Index: src/sdk/resources/project_deps.xrc
===================================================================
--- src/sdk/resources/project_deps.xrc	(revision 3627)
+++ src/sdk/resources/project_deps.xrc	(working copy)
@@ -28,7 +28,7 @@
         <border>8</border>
       </object>
       <object class="sizeritem">
-        <object class="wxCheckList" name="lstDeps">
+        <object class="wxCheckListBox" name="lstDeps">
           <content/>
         </object>
         <option>1</option>
