Index: src/sdk/templatemanager.cpp
===================================================================
--- src/sdk/templatemanager.cpp	(revision 3618)
+++ src/sdk/templatemanager.cpp	(working copy)
@@ -182,6 +182,16 @@
 //        Manager::Get()->GetMessageManager()->DebugLog("dst=%s, dstname=%s", dst.c_str(), dstname.GetFullPath().c_str());
         if (!CreateDirRecursively(dst))
             Manager::Get()->GetMessageManager()->DebugLog(_T("Failed creating directory for %s"), dst.c_str());
+        if (wxFileExists(dst))
+        {
+            if(cbMessageBox(dst + _(" already exists.\nDo you want to replace it?"), _("File exists"), wxICON_EXCLAMATION | wxYES_NO | wxNO_DEFAULT) != wxID_YES)
+            {
+                if (FileTypeOf(dst) == ftCodeBlocksProject)
+                    project_filename = dst;
+                ++count;
+                continue;
+            }
+        }
         if (wxCopyFile(src, dst, true))
         {
             if (FileTypeOf(dst) == ftCodeBlocksProject)
