Index: src/plugins/contrib/devpak_plugin/update.xrc
===================================================================
--- src/plugins/contrib/devpak_plugin/update.xrc	(revision 5097)
+++ src/plugins/contrib/devpak_plugin/update.xrc	(working copy)
@@ -166,6 +166,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
\ No newline at end of file
Index: src/plugins/codecompletion/resources/insert_class_method.xrc
===================================================================
--- src/plugins/codecompletion/resources/insert_class_method.xrc	(revision 5097)
+++ src/plugins/codecompletion/resources/insert_class_method.xrc	(working copy)
@@ -135,6 +135,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
Index: src/plugins/codecompletion/resources/selectincludefile.xrc
===================================================================
--- src/plugins/codecompletion/resources/selectincludefile.xrc	(revision 5097)
+++ src/plugins/codecompletion/resources/selectincludefile.xrc	(working copy)
@@ -2,7 +2,7 @@
     <object class="wxDialog" name="SelectIncludeFile">
         <title>Candidate include files</title>
         <centered>1</centered>
-        <style>wxCAPTION|wxTHICK_FRAME|wxSYSTEM_MENU|wxRESIZE_BOX|wxCLOSE_BOX</style>
+        <style>wxCAPTION|wxRESIZE_BORDER|wxSYSTEM_MENU|wxMAXIMIZE_BOX|wxCLOSE_BOX</style>
         <object class="wxBoxSizer">
             <orient>wxVERTICAL</orient>
             <object class="sizeritem">
Index: src/sdk/resources/generic_multi_select.xrc
===================================================================
--- src/sdk/resources/generic_multi_select.xrc	(revision 5097)
+++ src/sdk/resources/generic_multi_select.xrc	(working copy)
@@ -92,6 +92,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
\ No newline at end of file
Index: src/sdk/resources/incremental_select_list.xrc
===================================================================
--- src/sdk/resources/incremental_select_list.xrc	(revision 5097)
+++ src/sdk/resources/incremental_select_list.xrc	(working copy)
@@ -31,6 +31,6 @@
         <minsize>320,160</minsize>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
\ No newline at end of file
Index: src/sdk/resources/generic_multiline_notes.xrc
===================================================================
--- src/sdk/resources/generic_multiline_notes.xrc	(revision 5097)
+++ src/sdk/resources/generic_multiline_notes.xrc	(working copy)
@@ -3,7 +3,7 @@
   <object class="wxDialog" name="dlgGenericMultiLineNotes">
     <title>Notes</title>
     <centered>1</centered>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
     <object class="wxBoxSizer">
       <orient>wxVERTICAL</orient>
       <object class="sizeritem">
Index: src/sdk/resources/auto_detect_compilers.xrc
===================================================================
--- src/sdk/resources/auto_detect_compilers.xrc	(revision 5097)
+++ src/sdk/resources/auto_detect_compilers.xrc	(working copy)
@@ -68,6 +68,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
\ No newline at end of file
Index: src/sdk/resources/edit_array_order.xrc
===================================================================
--- src/sdk/resources/edit_array_order.xrc	(revision 5097)
+++ src/sdk/resources/edit_array_order.xrc	(working copy)
@@ -3,7 +3,7 @@
   <object class="wxDialog" name="dlgEditArrayOrder">
     <title>Edit order</title>
     <centered>1</centered>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
     <object class="wxBoxSizer">
       <orient>wxVERTICAL</orient>
       <object class="sizeritem">
Index: src/sdk/resources/edit_tool.xrc
===================================================================
--- src/sdk/resources/edit_tool.xrc	(revision 5097)
+++ src/sdk/resources/edit_tool.xrc	(working copy)
@@ -3,7 +3,7 @@
   <object class="wxDialog" name="dlgEditTool">
     <title>Edit tool</title>
     <centered>1</centered>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
     <object class="wxBoxSizer">
       <orient>wxVERTICAL</orient>
       <object class="sizeritem">
Index: src/sdk/resources/project_manager_file_types.xrc
===================================================================
--- src/sdk/resources/project_manager_file_types.xrc	(revision 5097)
+++ src/sdk/resources/project_manager_file_types.xrc	(working copy)
@@ -113,6 +113,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
\ No newline at end of file
Index: src/sdk/resources/plugins_configuration.xrc
===================================================================
--- src/sdk/resources/plugins_configuration.xrc	(revision 5097)
+++ src/sdk/resources/plugins_configuration.xrc	(working copy)
@@ -2,7 +2,7 @@
     <object class="wxDialog" name="dlgConfigurePlugins">
         <title>Manage plugins</title>
         <centered>1</centered>
-        <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+        <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
         <object class="wxBoxSizer">
             <orient>wxVERTICAL</orient>
             <object class="sizeritem">
Index: src/src/resources/app_compiler_settings.xrc
===================================================================
--- src/src/resources/app_compiler_settings.xrc	(revision 5097)
+++ src/src/resources/app_compiler_settings.xrc	(working copy)
@@ -149,6 +149,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX|wxMAXIMIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
Index: src/src/resources/dlg_about_plugin.xrc
===================================================================
--- src/src/resources/dlg_about_plugin.xrc	(revision 5097)
+++ src/src/resources/dlg_about_plugin.xrc	(working copy)
@@ -180,6 +180,6 @@
         <border>8</border>
       </object>
     </object>
-    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+    <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
   </object>
 </resource>
