Index: src/src/appglobals.cpp
===================================================================
--- src/src/appglobals.cpp	(revision 3652)
+++ src/src/appglobals.cpp	(working copy)
@@ -73,9 +73,17 @@
 #elif defined(__WXMSW__)
 	int Major = 0;
 	int family = wxGetOsVersion(&Major, NULL);
+#if wxCHECK_VERSION(2, 8, 0)
+	if(family == wxOS_WINDOWS_9X)
+#else
 	if(family == wxWIN95)
+#endif
         return osWindows9598ME;
+#if wxCHECK_VERSION(2, 8, 0)
+    if(family == wxOS_WINDOWS_NT)
+#else
     if(family == wxWINDOWS_NT)
+#endif
     {
         if(Major > 4)
             return osWindowsXP;
