Index: src/tools/cb_share_config/mainframe.cpp
===================================================================
--- src/tools/cb_share_config/mainframe.cpp	(revision 4032)
+++ src/tools/cb_share_config/mainframe.cpp	(working copy)
@@ -417,7 +417,11 @@
     wxT("*.conf"),                                   // default extension
     wxT("Code::Blocks configuration files (*.conf)|*.conf|"
         "All files (*.*)|*.*"),                      // wildcards
-    wxOPEN | wxFILE_MUST_EXIST                       // flags
+#if wxCHECK_VERSION(2, 8, 0)
+    wxFD_OPEN | wxFD_FILE_MUST_EXIST
+#else
+    wxOPEN | wxFILE_MUST_EXIST
+#endif
 #if (WXWIN_COMPATIBILITY_2_4)
     | wxHIDE_READONLY
 #endif
