Index: src/plugins/classwizard/classwizarddlg.cpp
===================================================================
--- src/plugins/classwizard/classwizarddlg.cpp	(revision 3652)
+++ src/plugins/classwizard/classwizarddlg.cpp	(working copy)
@@ -83,7 +83,7 @@
     if (prj)
   	    XRCCTRL(*this, "txtBaseDir", wxTextCtrl)->SetValue(prj->GetBasePath());
     else
-  	    XRCCTRL(*this, "txtBaseDir", wxTextCtrl)->SetValue(wxGetWorkingDirectory());
+  	    XRCCTRL(*this, "txtBaseDir", wxTextCtrl)->SetValue(::wxGetCwd());
 }
 
 
