Index: compilergcc.cpp
===================================================================
--- compilergcc.cpp	(revision 3757)
+++ compilergcc.cpp	(working copy)
@@ -1749,7 +1749,7 @@
 
     if (!cmd.Replace(_T("$SCRIPT"), script))
         // if they didn't specify $SCRIPT, append:
-        cmd << _T(" ") << command;
+        cmd << command;
 
     Manager::Get()->GetMessageManager()->Log(m_PageIndex, _("Checking for existence: %s"), f.GetFullPath().c_str());
     if (!wxFileExists(f.GetFullPath()))