--- /src/plugins/compilergcc/compileroptionsdlg.cpp	2007-03-29 03:36:39.000000000 -0700
+++ /src/plugins/compilergcc/compileroptionsdlg.cpp	2007-03-29 03:37:23.000000000 -0700
@@ -254,13 +254,12 @@
     {
         // project settings

-        nb->DeletePage(7); // remove "Other" page
+        nb->RemovePage(7); // remove "Other" page
         nb->DeletePage(4); // remove "Programs" page

         // remove "Compiler" buttons
         wxWindow* win = XRCCTRL(*this, "btnAddCompiler", wxButton);
         wxSizer* sizer2 = win->GetContainingSizer();
-        sizer2->Clear(true);
         sizer2->RecalcSizes();
         sizer2->Layout();

