Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 3805)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -2438,6 +2438,14 @@
 
         if(name != new_name)
         {
+            if(wxFileExists(path + new_name))
+            {
+                cbMessageBox(_("Could not rename file ") + path + new_name +
+                             _(",\nbecause file with the name you specified already exists.") +
+                             _("\nPlease specify a different file name."), _("Error renaming file"), wxICON_ERROR);
+                return;
+            }
+
             if(!wxRenameFile(path + name, path + new_name))
             {
                 wxBell();
