Index: src/sdk/toolsmanager.cpp
===================================================================
--- src/sdk/toolsmanager.cpp	(revision 3808)
+++ src/sdk/toolsmanager.cpp	(working copy)
@@ -132,9 +132,11 @@
             cmdline << baseDir << wxT("/" CONSOLE_RUNNER " ");
     }
 
-    if (!cmdline.Replace(_T("$SCRIPT"), cmd << _T(" ") << params))
+    cmd << _T(" ") << params;
+
+    if (!cmdline.Replace(_T("$SCRIPT"), cmd))
         // if they didn't specify $SCRIPT, append:
-        cmdline << cmd << _T(" ") << params;
+        cmdline << cmd;
 
     if(!(Manager::Get()->GetMacrosManager()))
         return false; // We cannot afford the Macros Manager to fail here!
