Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 3811)
+++ src/src/main.cpp	(working copy)
@@ -3232,6 +3232,7 @@
     mbar->Enable(idEditAutoComplete, ed);
     mbar->Enable(idEditUncommentSelected, ed);
     mbar->Enable(idEditToggleCommentSelected, ed);
+    mbar->Enable(idEditSpecialCommandsCase, eb && hasSel);
 
     if (ed)
     {
