Index: src/sdk/globals.cpp
===================================================================
--- src/sdk/globals.cpp	(revision 3816)
+++ src/sdk/globals.cpp	(working copy)
@@ -894,6 +894,10 @@
     }
 }
 
+void PlaceWindow(wxWindow *w, cbPlaceDialogMode mode, bool enforce)
+{
+}
+
 #endif //platform-specific placement code
 
 DirAccessCheck cbDirAccessCheck(const wxString& dir)
Index: src/plugins/contrib/codesnippets/snippetsconfig.cpp
===================================================================
--- src/plugins/contrib/codesnippets/snippetsconfig.cpp	(revision 3816)
+++ src/plugins/contrib/codesnippets/snippetsconfig.cpp	(working copy)
@@ -136,7 +136,7 @@
     cfgFile.Read( wxT("WindowPosition"),  &winPos, wxEmptyString) ;
     if ( not winPos.IsEmpty() )
     {
-        wxWX2MBbuf buf = cbU2C(winPos);
+        const wxWX2MBbuf buf = cbU2C(winPos);
         std::string cstring( buf );
         std::stringstream istream(cstring);
         istream >> windowXpos ;
