Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 3855)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -1491,7 +1491,7 @@
             } else
                 pos=-1;
         }
-        if (pos != -1)
+        if (pos != -1 && data->start!=data->end)
         {
             control->GotoPos(pos);
             control->EnsureVisible(control->LineFromPosition(pos));
@@ -1844,7 +1844,7 @@
                     pos=-1;
             }

-            if (pos == -1)
+            if (pos == -1 || data->start==data->end)
                 break;

             if (confirm)
@@ -2050,7 +2050,7 @@
             } else
                 pos=-1;
         }
-        if (pos != -1)
+        if (pos != -1 && data->start!=data->end)
         {
             control->GotoPos(pos);
             control->EnsureVisible(control->LineFromPosition(pos));
