Index: src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp
===================================================================
--- src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp	(revision 3869)
+++ src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp	(working copy)
@@ -97,14 +97,14 @@
    #endif
 
     wxWindow* pw = this;
-    LOGIT( _T("this[%s]Title[%s]"),pw->GetName().c_str(),pw->GetTitle().c_str() );
+    LOGIT( _T("this[%s]Title[%s]"),pw->GetName().c_str(),pw->GetLabel().c_str() );
     if (pw && pw->GetParent()) //
     {   pw = pw->GetParent();
-        LOGIT( _T("parent1[%s]Title[%s]"),pw->GetName().c_str(),pw->GetTitle().c_str() );
+        LOGIT( _T("parent1[%s]Title[%s]"),pw->GetName().c_str(),pw->GetLabel().c_str() );
     }
     if (pw && pw->GetParent())  //This is the SnippetWindow parent
     {   pw = pw->GetParent();
-        LOGIT( _T("parent2[%s]Title[%s]"),pw->GetName().c_str(),pw->GetTitle().c_str() );
+        LOGIT( _T("parent2[%s]Title[%s]"),pw->GetName().c_str(),pw->GetLabel().c_str() );
     }
 
 //    m_pSnippetWindowParent = pw;
Index: src/plugins/contrib/codesnippets/codesnippetstreectrl.h
===================================================================
--- src/plugins/contrib/codesnippets/codesnippetstreectrl.h	(revision 3869)
+++ src/plugins/contrib/codesnippets/codesnippetstreectrl.h	(working copy)
@@ -40,7 +40,7 @@
 #endif
 
 WX_DEFINE_ARRAY(wxDialog*, DlgPtrArray);
-WX_DEFINE_ARRAY(int, DlgRetcodeArray);
+WX_DEFINE_ARRAY_INT(int, DlgRetcodeArray);
 
 // ----------------------------------------------------------------------------
 class CodeSnippetsTreeCtrl : public wxTreeCtrl
