Index: src/plugins/contrib/keybinder/keybinder.cbp
===================================================================
--- src/plugins/contrib/keybinder/keybinder.cbp	(revision 3955)
+++ src/plugins/contrib/keybinder/keybinder.cbp	(working copy)
@@ -43,6 +43,7 @@
 			<Add option="-Wall" />
 			<Add option="-g" />
 			<Add directory="$(#CB.include)" />
+			<Add directory="$(#CB.include)\wxscintilla\include" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX)\contrib\include" />
