Index: src/plugins/contrib/codesnippets/memorymappedfile.h
===================================================================
--- src/plugins/contrib/codesnippets/memorymappedfile.h	(revision 3955)
+++ src/plugins/contrib/codesnippets/memorymappedfile.h	(working copy)
@@ -60,7 +60,8 @@
 #endif
 
 #ifdef __WXMSW__
-#include <windows.h>
+// includes windows.h but wraps it to prevent errors per wxWidgets special note tn0019.txt
+#include <wx/msw/wrapwin.h> 
 #else
 #include <unistd.h>
 #include <sys/mman.h>
