Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 3960)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -1279,7 +1279,7 @@
     wxFileDialog* dlg = new wxFileDialog(Manager::Get()->GetAppWindow(),
                                          _("Save file"),
                                          Path,
-                                         fname.GetName(),
+                                         fname.GetFullName(),
                                          Filters,
                                          wxSAVE | wxOVERWRITE_PROMPT);
     dlg->SetFilterIndex(StoredIndex);
