Index: src/sdk/finddlg.cpp
===================================================================
--- src/sdk/finddlg.cpp	(revision 3968)
+++ src/sdk/finddlg.cpp	(working copy)
@@ -88,7 +88,7 @@
 	XRCCTRL(*this, "rbDirection", wxRadioBox)->SetSelection(cfg->ReadInt(CONF_GROUP _T("/direction"), 1));
 	XRCCTRL(*this, "rbDirection", wxRadioBox)->Enable(!XRCCTRL(*this, "chkRegEx1", wxCheckBox)->GetValue()); // if regex, only forward searches
 	XRCCTRL(*this, "rbOrigin", wxRadioBox)->SetSelection(cfg->ReadInt(CONF_GROUP _T("/origin"), 0));
-	XRCCTRL(*this, "rbScope1", wxRadioBox)->SetSelection(hasSelection);
+	XRCCTRL(*this, "rbScope1", wxRadioBox)->SetSelection(0);
 	XRCCTRL(*this, "rbScope1", wxRadioBox)->Enable(hasSelection);
 
 	// find in files options
Index: src/sdk/replacedlg.cpp
===================================================================
--- src/sdk/replacedlg.cpp	(revision 3968)
+++ src/sdk/replacedlg.cpp	(working copy)
@@ -71,7 +71,7 @@
 	XRCCTRL(*this, "rbDirection", wxRadioBox)->SetSelection(cfg->ReadInt(CONF_GROUP _T("/direction"), 1));
 	XRCCTRL(*this, "rbDirection", wxRadioBox)->Enable(!XRCCTRL(*this, "chkRegEx1", wxCheckBox)->GetValue()); // if regex, only forward searches
 	XRCCTRL(*this, "rbOrigin", wxRadioBox)->SetSelection(cfg->ReadInt(CONF_GROUP _T("/origin"), 0));
-	XRCCTRL(*this, "rbScope1", wxRadioBox)->SetSelection(hasSelection);
+	XRCCTRL(*this, "rbScope1", wxRadioBox)->SetSelection(0);
 	XRCCTRL(*this, "rbScope1", wxRadioBox)->Enable(hasSelection);
 	// special key, uses same config for both find & replace options
 	XRCCTRL(*this, "chkAutoWrapSearch", wxCheckBox)->SetValue(cfg->ReadBool(_T("/find_options/auto_wrap_search"), true));
