Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 3993)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -2305,8 +2305,9 @@
 
         // make the filename relative
         wxString filename = filesList[i];
-        if (filename.StartsWith(data->searchPath))
-            filename.Remove(0, data->searchPath.Length());
+        wxFileName fname(filename);
+        if(fname.MakeRelativeTo(data->searchPath))
+            filename=fname.GetFullPath();
 
         // log it
         LogSearch(filename, line + 1, control->GetLine(line));
