Index: src/plugins/contrib/wxSmith/wxsmith/wxsBitmapIconEditorDlg.wxs
===================================================================
--- src/plugins/contrib/wxSmith/wxsmith/wxsBitmapIconEditorDlg.wxs	(revision 4051)
+++ src/plugins/contrib/wxSmith/wxsmith/wxsBitmapIconEditorDlg.wxs	(working copy)
@@ -1,198 +1,186 @@
+<?xml version="1.0" encoding="utf-8" ?>
 <wxsmith>
     <object class="wxDialog" name="wxsBitmapIconEditorDlg">
         <title>Image editor</title>
-        <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX</style>
+        <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER</style>
         <object class="wxBoxSizer" variable="BoxSizer1" member="yes">
             <orient>wxVERTICAL</orient>
             <object class="sizeritem">
-                <option>1</option>
-                <border>5</border>
-                <flag>wxALIGN_CENTER|wxEXPAND</flag>
                 <object class="wxBoxSizer" variable="BoxSizer5" member="yes">
-                    <orient>wxHORIZONTAL</orient>
                     <object class="sizeritem">
-                        <border>5</border>
-                        <flag>wxALL|wxALIGN_CENTER|wxEXPAND</flag>
                         <object class="wxStaticBoxSizer" variable="StaticBoxSizer1" member="yes">
-                            <orient>wxHORIZONTAL</orient>
                             <label>Image options</label>
                             <object class="sizeritem">
-                                <option>1</option>
-                                <border>5</border>
-                                <flag>wxALL|wxALIGN_CENTER|wxEXPAND</flag>
                                 <object class="wxFlexGridSizer" variable="FlexGridSizer1" member="yes">
-                                    <rows>0</rows>
                                     <cols>1</cols>
-                                    <vgap>0</vgap>
-                                    <hgap>0</hgap>
                                     <growablecols>1</growablecols>
                                     <object class="sizeritem">
-                                        <option>1</option>
-                                        <border>5</border>
-                                        <flag>wxLEFT|wxRIGHT|wxTOP|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL|wxEXPAND</flag>
                                         <object class="wxRadioButton" name="ID_RADIOBUTTON1" variable="NoImage" member="yes">
                                             <label>No image</label>
-                                            <value>1</value>
-                                            <handler entry="EVT_RADIOBUTTON" function="OnUpdatePreview" />
+                                            <handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
                                         </object>
+                                        <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+                                        <border>5</border>
+                                        <option>1</option>
                                     </object>
                                     <object class="spacer">
-                                        <size>1,1</size>
-                                        <option>1</option>
+                                        <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
                                         <border>5</border>
-                                        <flag>wxALL|wxALIGN_CENTER</flag>
+                                        <option>1</option>
+                                        <size>1,1</size>
                                     </object>
                                     <object class="sizeritem">
-                                        <option>1</option>
-                                        <border>5</border>
-                                        <flag>wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
                                         <object class="wxRadioButton" name="ID_RADIOBUTTON2" variable="ImageFile" member="yes">
                                             <label>Image From File:</label>
-                                            <handler entry="EVT_RADIOBUTTON" function="OnUpdatePreview" />
+                                            <handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
                                         </object>
+                                        <flag>wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+                                        <border>5</border>
+                                        <option>1</option>
                                     </object>
                                     <object class="sizeritem">
-                                        <option>1</option>
-                                        <border>5</border>
-                                        <flag>wxALIGN_CENTER|wxEXPAND</flag>
                                         <object class="wxBoxSizer" variable="BoxSizer2" member="yes">
-                                            <orient>wxHORIZONTAL</orient>
                                             <object class="spacer">
-                                                <size>20,1</size>
+                                                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
-                                                <flag>wxALL|wxALIGN_CENTER</flag>
+                                                <size>20,1</size>
                                             </object>
                                             <object class="sizeritem">
-                                                <option>1</option>
-                                                <border>5</border>
-                                                <flag>wxLEFT|wxTOP|wxBOTTOM|wxALIGN_CENTER</flag>
                                                 <object class="wxTextCtrl" name="ID_TEXTCTRL1" variable="FileName" member="yes">
-                                                    <value />
-                                                    <handler entry="EVT_TEXT" function="OnFileNameText" />
+                                                    <handler function="OnFileNameText" entry="EVT_TEXT" />
                                                 </object>
+                                                <flag>wxTOP|wxBOTTOM|wxLEFT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <border>5</border>
+                                                <option>1</option>
                                             </object>
                                             <object class="sizeritem">
-                                                <border>5</border>
-                                                <flag>wxRIGHT|wxTOP|wxBOTTOM|wxALIGN_CENTER</flag>
                                                 <object class="wxButton" name="ID_BUTTON3" variable="Button3" member="yes">
                                                     <label>...</label>
-                                                    <handler entry="EVT_BUTTON" function="OnButton3Click" />
                                                     <style>wxBU_EXACTFIT</style>
+                                                    <handler function="OnButton3Click" entry="EVT_BUTTON" />
                                                 </object>
+                                                <flag>wxTOP|wxBOTTOM|wxRIGHT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <border>5</border>
                                             </object>
                                         </object>
+                                        <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                        <border>5</border>
+                                        <option>1</option>
                                     </object>
                                     <object class="sizeritem">
-                                        <option>1</option>
-                                        <border>5</border>
-                                        <flag>wxLEFT|wxRIGHT|wxBOTTOM|wxALIGN_LEFT|wxALIGN_TOP</flag>
                                         <object class="wxRadioButton" name="ID_RADIOBUTTON3" variable="ImageArt" member="yes">
                                             <label>Image from wxArtProvider:</label>
-                                            <handler entry="EVT_RADIOBUTTON" function="OnUpdatePreview" />
+                                            <handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
                                         </object>
+                                        <flag>wxBOTTOM|wxLEFT|wxRIGHT|wxALIGN_LEFT|wxALIGN_TOP</flag>
+                                        <border>5</border>
+                                        <option>1</option>
                                     </object>
                                     <object class="sizeritem">
-                                        <option>1</option>
-                                        <border>5</border>
-                                        <flag>wxALIGN_CENTER</flag>
                                         <object class="wxBoxSizer" variable="BoxSizer4" member="yes">
-                                            <orient>wxHORIZONTAL</orient>
                                             <object class="spacer">
-                                                <size>20,1</size>
+                                                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
-                                                <flag>wxALL|wxALIGN_CENTER</flag>
+                                                <size>20,1</size>
                                             </object>
                                             <object class="sizeritem">
-                                                <option>1</option>
-                                                <border>5</border>
-                                                <flag>wxALIGN_CENTER</flag>
                                                 <object class="wxFlexGridSizer" variable="FlexGridSizer2" member="yes">
-                                                    <rows>0</rows>
                                                     <cols>2</cols>
-                                                    <vgap>0</vgap>
-                                                    <hgap>0</hgap>
                                                     <object class="sizeritem">
-                                                        <option>1</option>
-                                                        <border>5</border>
-                                                        <flag>wxLEFT|wxTOP|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
                                                         <object class="wxStaticText" name="ID_STATICTEXT1" variable="StaticText1" member="yes">
                                                             <label>Art Id:</label>
                                                         </object>
+                                                        <flag>wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+                                                        <border>5</border>
+                                                        <option>1</option>
                                                     </object>
                                                     <object class="sizeritem">
-                                                        <option>1</option>
-                                                        <border>5</border>
-                                                        <flag>wxLEFT|wxRIGHT|wxTOP|wxALIGN_CENTER|wxEXPAND</flag>
                                                         <object class="wxComboBox" name="ID_COMBOBOX2" variable="ArtId" member="yes">
-                                                            <handler entry="EVT_COMBOBOX" function="OnArtIdSelect" />
+                                                            <style>wxCB_SORT</style>
+                                                            <handler function="OnArtIdSelect" entry="EVT_COMBOBOX" />
                                                         </object>
+                                                        <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                        <border>5</border>
+                                                        <option>1</option>
                                                     </object>
                                                     <object class="sizeritem">
-                                                        <option>1</option>
-                                                        <border>5</border>
-                                                        <flag>wxLEFT|wxTOP|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
                                                         <object class="wxStaticText" name="ID_STATICTEXT2" variable="StaticText2" member="yes">
                                                             <label>Art Client:</label>
                                                         </object>
+                                                        <flag>wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+                                                        <border>5</border>
+                                                        <option>1</option>
                                                     </object>
                                                     <object class="sizeritem">
-                                                        <option>1</option>
-                                                        <border>5</border>
-                                                        <flag>wxLEFT|wxRIGHT|wxTOP|wxALIGN_CENTER|wxEXPAND</flag>
                                                         <object class="wxComboBox" name="ID_COMBOBOX1" variable="ArtClient" member="yes">
-                                                            <handler entry="EVT_COMBOBOX" function="OnArtIdSelect" />
+                                                            <style>wxCB_SORT</style>
+                                                            <handler function="OnArtIdSelect" entry="EVT_COMBOBOX" />
                                                         </object>
+                                                        <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                        <border>5</border>
+                                                        <option>1</option>
                                                     </object>
                                                 </object>
+                                                <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <border>5</border>
+                                                <option>1</option>
                                             </object>
                                         </object>
+                                        <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                        <border>5</border>
+                                        <option>1</option>
                                     </object>
                                 </object>
+                                <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                <border>5</border>
+                                <option>1</option>
                             </object>
                         </object>
+                        <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <border>5</border>
                     </object>
                     <object class="sizeritem">
-                        <option>1</option>
-                        <border>5</border>
-                        <flag>wxALL|wxALIGN_CENTER|wxEXPAND</flag>
                         <object class="wxStaticBoxSizer" variable="StaticBoxSizer2" member="yes">
-                            <orient>wxHORIZONTAL</orient>
                             <label>Preview</label>
                             <object class="sizeritem">
-                                <option>1</option>
-                                <border>5</border>
-                                <flag>wxALL|wxALIGN_CENTER|wxEXPAND</flag>
                                 <object class="wxStaticBitmap" name="ID_STATICBITMAP1" variable="Preview" member="yes">
-                                    <bitmap />
                                     <size>200,200</size>
                                 </object>
+                                <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                <border>5</border>
+                                <option>1</option>
                             </object>
                         </object>
+                        <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <border>5</border>
+                        <option>1</option>
                     </object>
                 </object>
+                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <border>5</border>
+                <option>1</option>
             </object>
             <object class="sizeritem">
-                <border>5</border>
-                <flag>wxALL|wxALIGN_CENTER</flag>
                 <object class="wxBoxSizer" variable="BoxSizer3" member="yes">
-                    <orient>wxHORIZONTAL</orient>
                     <object class="sizeritem">
-                        <border>5</border>
-                        <flag>wxLEFT|wxTOP|wxBOTTOM|wxALIGN_CENTER</flag>
                         <object class="wxButton" name="ID_BUTTON1" variable="Button1" member="yes">
                             <label>OK</label>
                             <default>1</default>
-                            <handler entry="EVT_BUTTON" function="OnButton1Click" />
+                            <handler function="OnButton1Click" entry="EVT_BUTTON" />
                         </object>
+                        <flag>wxTOP|wxBOTTOM|wxLEFT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <border>5</border>
                     </object>
                     <object class="sizeritem">
-                        <border>5</border>
-                        <flag>wxALL|wxALIGN_CENTER</flag>
                         <object class="wxButton" name="wxID_CANCEL" variable="Button2" member="yes">
                             <label>Cancel</label>
                         </object>
+                        <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <border>5</border>
                     </object>
                 </object>
+                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <border>5</border>
             </object>
         </object>
     </object>
Index: src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.cpp	(revision 4051)
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.cpp	(working copy)
@@ -55,13 +55,20 @@
     _T("wxART_GO_TO_PARENT"),
     _T("wxART_GO_HOME"),
     _T("wxART_FILE_OPEN"),
+    _T("wxART_FILE_SAVE"),
+    _T("wxART_FILE_SAVE_AS"),
     _T("wxART_PRINT"),
     _T("wxART_HELP"),
     _T("wxART_TIP"),
     _T("wxART_REPORT_VIEW"),
     _T("wxART_LIST_VIEW"),
     _T("wxART_NEW_DIR"),
+    _T("wxART_HARDDISK"),
+    _T("wxART_FLOPPY"),
+    _T("wxART_CDROM"),
+    _T("wxART_REMOVABLE"),
     _T("wxART_FOLDER"),
+    _T("wxART_FOLDER_OPEN"),
     _T("wxART_GO_DIR_UP"),
     _T("wxART_EXECUTABLE_FILE"),
     _T("wxART_NORMAL_FILE"),
@@ -72,6 +79,16 @@
     _T("wxART_WARNING"),
     _T("wxART_INFORMATION"),
     _T("wxART_MISSING_IMAGE"),
+    _T("wxART_COPY"),
+    _T("wxART_CUT"),
+    _T("wxART_PASTE"),
+    _T("wxART_DELETE"),
+    _T("wxART_NEW"),
+    _T("wxART_UNDO"),
+    _T("wxART_REDO"),
+    _T("wxART_QUIT"),
+    _T("wxART_FIND"),
+    _T("wxART_FIND_AND_REPLACE"),
     NULL
 };
 
@@ -115,7 +132,7 @@
     Data(_Data)
 {
 	//(*Initialize(wxsBitmapIconEditorDlg)
-    Create(parent,id,_("Image editor"),wxDefaultPosition,wxDefaultSize,wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX,_T("id"));
+	Create(parent,id,_("Image editor"),wxDefaultPosition,wxDefaultSize,wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER,_T("wxDialog"));
 	BoxSizer1 = new wxBoxSizer(wxVERTICAL);
 	BoxSizer5 = new wxBoxSizer(wxHORIZONTAL);
 	StaticBoxSizer1 = new wxStaticBoxSizer(wxHORIZONTAL,this,_("Image options"));
@@ -140,11 +157,11 @@
 	FlexGridSizer2 = new wxFlexGridSizer(0,2,0,0);
 	StaticText1 = new wxStaticText(this,ID_STATICTEXT1,_("Art Id:"),wxDefaultPosition,wxDefaultSize,0,_T("ID_STATICTEXT1"));
 	FlexGridSizer2->Add(StaticText1,1,wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL,5);
-	ArtId = new wxComboBox(this,ID_COMBOBOX2,wxEmptyString,wxDefaultPosition,wxDefaultSize,0,NULL,0,wxDefaultValidator,_T("ID_COMBOBOX2"));
+	ArtId = new wxComboBox(this,ID_COMBOBOX2,wxEmptyString,wxDefaultPosition,wxDefaultSize,0,NULL,wxCB_SORT,wxDefaultValidator,_T("ID_COMBOBOX2"));
 	FlexGridSizer2->Add(ArtId,1,wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL,5);
 	StaticText2 = new wxStaticText(this,ID_STATICTEXT2,_("Art Client:"),wxDefaultPosition,wxDefaultSize,0,_T("ID_STATICTEXT2"));
 	FlexGridSizer2->Add(StaticText2,1,wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL,5);
-	ArtClient = new wxComboBox(this,ID_COMBOBOX1,wxEmptyString,wxDefaultPosition,wxDefaultSize,0,NULL,0,wxDefaultValidator,_T("ID_COMBOBOX1"));
+	ArtClient = new wxComboBox(this,ID_COMBOBOX1,wxEmptyString,wxDefaultPosition,wxDefaultSize,0,NULL,wxCB_SORT,wxDefaultValidator,_T("ID_COMBOBOX1"));
 	FlexGridSizer2->Add(ArtClient,1,wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL,5);
 	BoxSizer4->Add(FlexGridSizer2,1,wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL,5);
 	FlexGridSizer1->Add(BoxSizer4,1,wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL,5);
