Index: src/sdk/wxFlatNotebook/src/wxFlatNotebook/wxFlatNotebook.cpp
===================================================================
--- src/sdk/wxFlatNotebook/src/wxFlatNotebook/wxFlatNotebook.cpp	(revision 4051)
+++ src/sdk/wxFlatNotebook/src/wxFlatNotebook/wxFlatNotebook.cpp	(working copy)
@@ -253,6 +253,7 @@
 		event.SetSelection( (int)page );
 		event.SetOldSelection( oldSelection );
 		event.SetEventObject( this );
+		GetEventHandler()->ProcessEvent(event);
 
 		if( !event.IsAllowed() )
 		{
