--- compilergcc.cpp.org 2007-06-04 20:37:19.000000000 +0800
+++ compilergcc.cpp     2007-06-04 20:47:53.000000000 +0800
@@ -1709,7 +1709,7 @@
                 //                 -- Csh Programming Considered Harmful
                 command << DEFAULT_CONSOLE_SHELL << strSPACE;
             }
-            command << _T("'") << LIBRARY_ENVVAR _T("=.:$") LIBRARY_ENVVAR << _T(" ");
+            command << _T(" '") << LIBRARY_ENVVAR _T("=.:$") LIBRARY_ENVVAR << _T(" ");
         }
         // should console runner be used?
         if (target->GetUseConsoleRunner())
