Index: src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp	(revision 4078)
+++ src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp	(working copy)
@@ -1481,7 +1481,7 @@
     }
     for ( int i=0; i<GetToolsCount(); i++ )
     {
-        m_IdMap[m_Tools[i]] = m_Tools[i]->GetLastTreeItemId();
+        StoreTreeIdsReq(m_Tools[i]);
     }
 }
 
