Index: plugins/contrib/wxSmith/wxwidgets/wxsnewwindowdlg.cpp
===================================================================
--- plugins/contrib/wxSmith/wxwidgets/wxsnewwindowdlg.cpp	(revision 4078)
+++ plugins/contrib/wxSmith/wxwidgets/wxsnewwindowdlg.cpp	(working copy)
@@ -252,6 +252,9 @@
         return;
     }
 
+    // Disable button to prevent more clicks while resource is created
+    FindWindowById(wxID_OK)->Enable(false);
+
     // Adding new files to project
     wxArrayInt Targets;
     Manager::Get()->GetProjectManager()->AddFileToProject(Hdr,cbProj,Targets);
