Index: src/include/prep.h
===================================================================
--- src/include/prep.h	(revision 4790)
+++ src/include/prep.h	(working copy)
@@ -5,6 +5,7 @@
 #include <wx/version.h>
 #endif
 
+#include <wx/filedlg.h>
 
 
 /* -----------------------------------------------
@@ -254,6 +254,17 @@
     #endif
 }
 
+#if wxCHECK_VERSION(2, 8, 0) && (WXWIN_COMPATIBILITY_2_6 == 0)
+enum
+{
+    wxOPEN              = wxFD_OPEN,
+    wxSAVE              = wxFD_SAVE,
+    wxOVERWRITE_PROMPT  = wxFD_OVERWRITE_PROMPT,
+    wxFILE_MUST_EXIST   = wxFD_FILE_MUST_EXIST,
+    wxMULTIPLE          = wxFD_MULTIPLE,
+    wxCHANGE_DIR        = wxFD_CHANGE_DIR
+};
+#endif
 
 
 /*  ---------------------------------------------------------------------------------------------------------
