Index: src/include/cbproject.h
===================================================================
--- src/include/cbproject.h	(revision 4103)
+++ src/include/cbproject.h	(working copy)
@@ -36,7 +36,8 @@
             ftdkFolder,
             ftdkFile,
             ftdkVirtualGroup, // wilcard matching
-            ftdkVirtualFolder
+            ftdkVirtualFolder,
+            ftdkMultiSelect   // several tree items selected
         };
 
         FileTreeData(cbProject* project, FileTreeDataKind kind = ftdkUndefined)
Index: src/include/globals.h
===================================================================
--- src/include/globals.h	(revision 4103)
+++ src/include/globals.h	(working copy)
@@ -32,6 +32,7 @@
     mtEditorManager,
     mtMessageManager,
     mtOpenFilesList,                //pecan 2006/03/22
+    mtFileManager,
     mtUnknown
 };

