Index: src/sdk/projectfileoptionsdlg.cpp
===================================================================
--- src/sdk/projectfileoptionsdlg.cpp	(revision 4191)
+++ src/sdk/projectfileoptionsdlg.cpp	(working copy)
@@ -125,7 +125,7 @@
 				long int &comment_lines, long int &empty_lines, long int &total_lines)
 {
 	wxTextFile file;
-	if (file.Open(filename.GetFullPath()))
+	if (file.Open(filename.GetFullPath(), wxConvFile))
 	{
 		bool multi_line_comment = false;
 		total_lines += file.GetLineCount();
