Index: src/plugins/codecompletion/classbrowserbuilderthread.cpp
===================================================================
--- src/plugins/codecompletion/classbrowserbuilderthread.cpp	(revision 4196)
+++ src/plugins/codecompletion/classbrowserbuilderthread.cpp	(working copy)
@@ -185,8 +185,10 @@
     m_pTreeTop->Freeze();
     m_pTreeBottom->Freeze();
 
-	RemoveInvalidNodes(m_pTreeTop, root);
-	RemoveInvalidNodes(m_pTreeBottom, m_pTreeBottom->GetRootItem());

+	m_pTreeTop->DeleteChildren(m_pTreeTop->GetRootItem());
+	m_pTreeBottom->DeleteChildren(m_pTreeBottom->GetRootItem());
+	// RemoveInvalidNodes(m_pTreeTop, root);
+	// RemoveInvalidNodes(m_pTreeBottom, m_pTreeBottom->GetRootItem());
 
     if (TestDestroy() || Manager::IsAppShuttingDown())
     {
