Index: src/sdk/compilerfactory.cpp
===================================================================
--- src/sdk/compilerfactory.cpp	(revision 4395)
+++ src/sdk/compilerfactory.cpp	(working copy)
@@ -20,6 +20,8 @@
     #include "compiler.h"
 #endif
 
+#include <wx/choicdlg.h>
+
 #include "autodetectcompilers.h"
 
 // statics
Index: src/sdk/manager.cpp
===================================================================
--- src/sdk/manager.cpp	(revision 4395)
+++ src/sdk/manager.cpp	(working copy)
@@ -51,6 +51,7 @@
     #include "xtra_res.h" // our new ToolBarAddOn handler
 #endif
 
+#include <wx/app.h>
 #include <wx/toolbar.h>
 #include <wx/fs_mem.h>
 
Index: src/sdk/uservarmanager.cpp
===================================================================
--- src/sdk/uservarmanager.cpp	(revision 4395)
+++ src/sdk/uservarmanager.cpp	(working copy)
@@ -24,6 +24,7 @@
     #include "cbexception.h"
     #include "infowindow.h"
 
+    #include <wx/button.h>
     #include <wx/dialog.h>
     #include <wx/intl.h>
     #include <wx/xrc/xmlres.h>
@@ -31,6 +32,7 @@
 #endif
 
 #include "annoyingdialog.h"
+#include <wx/textdlg.h>
 #include <wx/choice.h>
 #include <ctype.h>
 
Index: src/plugins/contrib/wxSmith/wxwidgets/wxsgridpanel.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/wxsgridpanel.cpp	(revision 4395)
+++ src/plugins/contrib/wxSmith/wxwidgets/wxsgridpanel.cpp	(working copy)
@@ -24,6 +24,7 @@
 #include <wx/dcclient.h> // wxPaintDC
 #include "wxsgridpanel.h"
 
+#include <wx/dcclient.h>
 #include <configmanager.h>
 #include <manager.h>
 
Index: src/plugins/debuggergdb/debuggertree.cpp
===================================================================
--- src/plugins/debuggergdb/debuggertree.cpp	(revision 4395)
+++ src/plugins/debuggergdb/debuggertree.cpp	(working copy)
@@ -8,6 +8,7 @@
 #include <wx/msgdlg.h>
 #include <wx/app.h>
 #include <wx/tipwin.h>
+#include <wx/settings.h>
 #include <globals.h>
 #include "debuggergdb.h"
 #include "debuggerdriver.h"
Index: src/plugins/compilergcc/compileroptionsdlg.cpp
===================================================================
--- src/plugins/compilergcc/compileroptionsdlg.cpp	(revision 4395)
+++ src/plugins/compilergcc/compileroptionsdlg.cpp	(working copy)
@@ -29,6 +29,7 @@
 #ifndef CB_PRECOMP
     #include <wx/arrstr.h>
     #include <wx/button.h>
+    #include <wx/checkbox.h>
     #include <wx/checklst.h>
     #include <wx/choice.h>
     #include <wx/event.h>
@@ -47,7 +48,11 @@
     #include <wx/spinctrl.h>
 	#include <wx/notebook.h>
 #endif
+#include <wx/choicdlg.h>
 #include <wx/filedlg.h>
+#include <wx/sizer.h>
+#include <wx/stattext.h>
+#include <wx/textdlg.h>
 #include "cbexception.h"
 #include "annoyingdialog.h"
 #include "compileroptionsdlg.h"
Index: src/src/dlgabout.cpp
===================================================================
--- src/src/dlgabout.cpp	(revision 4395)
+++ src/src/dlgabout.cpp	(working copy)
@@ -28,6 +28,7 @@
 #ifdef __WXMAC__
 #include <wx/font.h>
 #endif //__WXMAC__
+#include <wx/image.h>
 #include <wx/intl.h>
 #include <wx/stattext.h>
 #include <wx/string.h>
@@ -37,6 +38,7 @@
 #include "configmanager.h"
 #endif
 #include <wx/bitmap.h>
+#include <wx/dcmemory.h>
 #include <wx/statbmp.h>
 #include "appglobals.h"
 #include "dlgabout.h" // class's header file
Index: src/src/environmentsettingsdlg.cpp
===================================================================
--- src/src/environmentsettingsdlg.cpp	(revision 4395)
+++ src/src/environmentsettingsdlg.cpp	(working copy)
@@ -11,6 +11,7 @@
 */
 
 #include <sdk.h>
+#include <wx/button.h>
 #include <wx/xrc/xmlres.h>
 #include <manager.h>
 #include <configmanager.h>
@@ -26,6 +27,8 @@
 #include <wx/colordlg.h>
 #include <wx/msgdlg.h>
 #include <wx/imaglist.h>
+#include <wx/settings.h>
+#include <wx/stattext.h>
 #include "appglobals.h"
 #include "globals.h"
 #include "associations.h"
Index: src/src/compilersettingsdlg.cpp
===================================================================
--- src/src/compilersettingsdlg.cpp	(revision 4395)
+++ src/src/compilersettingsdlg.cpp	(working copy)
@@ -22,6 +22,8 @@
 #include <wx/checklst.h>
 #include <wx/filename.h>
 #include <wx/imaglist.h>
+#include <wx/sizer.h>
+#include <wx/stattext.h>
 
 #include "configurationpanel.h"
 #include "compilersettingsdlg.h"
Index: src/src/infopane.cpp
===================================================================
--- src/src/infopane.cpp	(revision 4395)
+++ src/src/infopane.cpp	(working copy)
@@ -3,6 +3,7 @@
 #ifndef CB_PRECOMP
     #include <wx/event.h>
     #include <wx/menu.h>
+    #include <cbexception.h>
 #endif
 
 #include "infopane.h"
