Index: src/src/app.cpp
===================================================================
--- src/src/app.cpp	(revision 4478)
+++ src/src/app.cpp	(working copy)
@@ -526,6 +526,9 @@
 
         if (!m_Script.IsEmpty())
         {
+            CodeBlocksEvent event(cbEVT_APP_STARTUP_DONE);
+            Manager::Get()->ProcessEvent(event);
+
             s_Loading = false;
             LoaderBase* loader = Manager::Get()->GetFileManager()->Load(m_Script);
 
