Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 4491)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -2702,10 +2702,6 @@
 		}
 	}
 
-    #ifdef USE_OPENFILES_TREE
-    Manager::Get()->GetEditorManager()->RebuildOpenedFilesTree();
-    #endif
-
     // sort out any global user vars that need to be defined now (in a batch) :)
     // but only if not loading workspace (else LoadWorkspace() will handle this)
     if (!m_IsLoadingWorkspace)
Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 4491)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -2894,6 +2894,7 @@
         if(GetActiveEditor()==ed)
             tree->SelectItem(item);
     }
+    tree->SortChildren(m_pData->m_TreeOpenedFiles);
     tree->Expand(m_pData->m_TreeOpenedFiles);
     tree->Thaw();
     m_pData->InvalidateTree();
