Index: src/sdk/wxFlatNotebook/src/wxFlatNotebook/wxFlatNotebook.cpp
===================================================================
--- src/sdk/wxFlatNotebook/src/wxFlatNotebook/wxFlatNotebook.cpp        (revision 4623)
+++ src/sdk/wxFlatNotebook/src/wxFlatNotebook/wxFlatNotebook.cpp        (working copy)
@@ -721,6 +721,7 @@
 , m_pDropTarget(NULL)
 , m_nLeftClickZone(wxFNB_NOWHERE)
 , m_customizeOptions(wxFNB_CUSTOM_ALL)
+, m_nFrom(0)
 {
        m_pRightClickMenu = NULL;
        m_nXButtonStatus = wxFNB_BTN_NONE;
@@ -1196,7 +1197,7 @@

                        wxRect tabRect = wxRect(pgInfo.GetPosition().x, pgInfo.GetPosition().y,
                                pgInfo.GetSize().x, pgInfo.GetSize().y);
-
+
                        if(InsideRect(tabRect, pt))
                        {
                                // We have a match
@@ -1301,7 +1302,7 @@
                //PopPageHistory(m_iActivePage);
        }

-       m_pagesInfoVec.RemoveAt(page);
+       m_pagesInfoVec.RemoveAt(page);

        if(m_iActivePage == wxNOT_FOUND && m_pagesInfoVec.Count() > 0){
                m_iActivePage = 0;
Index: src/sdk/logmanager.cpp
===================================================================
--- src/sdk/logmanager.cpp      (revision 4623)
+++ src/sdk/logmanager.cpp      (working copy)
@@ -9,7 +9,8 @@
 #include "loggers.h"

 LogSlot::LogSlot()
-       : log(0)
+       : log(0),
+       icon(0)
 {
 }
