Index: editorcolourset.cpp
===================================================================
--- editorcolourset.cpp	(revision 4632)
+++ editorcolourset.cpp	(working copy)
@@ -128,7 +128,7 @@
             ok = dir.GetNext(&filename);
             ++count;
         }
-        Manager::Get()->GetLogManager()->Log(_("Found %d lexers"), count);
+        Manager::Get()->GetLogManager()->Log(F(_("Found %d lexers"), count));
         count = 0;
     }
 
