Index: src/sdk/resources/global_uservars.xrc
===================================================================
--- src/sdk/resources/global_uservars.xrc	(revision 4649)
+++ src/sdk/resources/global_uservars.xrc	(working copy)
@@ -59,7 +59,6 @@
             <flag>wxALL|wxALIGN_CENTER</flag>
             <object class="wxButton" name="deleteSet">
               <label>Delete</label>
-              <style>wxNO_BORDER</style>
             </object>
           </object>
           <object class="sizeritem">
@@ -522,4 +521,4 @@
     </object>
     <centered>1</centered>
   </object>
-</resource>
\ No newline at end of file
+</resource>
