Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 4683)
+++ src/src/main.cpp	(working copy)
@@ -3380,7 +3380,7 @@
                                         this );
     long int line = 0;
     strLine.ToLong(&line);
-    if ( line > 1 && line <= max )
+    if ( line >= 1 && line <= max )
     {
         ed->UnfoldBlockFromLine(line - 1);
         ed->GotoLine(line - 1);
