Index: src/src/compilersettingsdlg.cpp
===================================================================
--- src/src/compilersettingsdlg.cpp	(revision 4685)
+++ src/src/compilersettingsdlg.cpp	(working copy)
@@ -76,6 +76,7 @@
     }
     wxCheckListBox* clb = XRCCTRL(*this, "chkBBPlugins", wxCheckListBox);
     clb->Clear();
+    clb->SetMinSize(wxSize(-1, 150));
     const PluginElementsArray& plugins = Manager::Get()->GetPluginManager()->GetPlugins();
     for (size_t i = 0; i < plugins.GetCount(); ++i)
     {
Index: src/src/resources/app_compiler_settings.xrc
===================================================================
--- src/src/resources/app_compiler_settings.xrc	(revision 4685)
+++ src/src/resources/app_compiler_settings.xrc	(working copy)
@@ -32,21 +32,25 @@
             <object class="wxPanel" name="tabBatchBuilds">
               <object class="wxBoxSizer">
                 <orient>wxVERTICAL</orient>
-                <object class="sizeritem">
-                  <object class="wxStaticText" name="">
-                    <label>Extra command-line arguments for batch-builds through shell extensions:</label>
-                  </object>
-                  <flag>wxLEFT|wxEXPAND</flag>
-                  <border>8</border>
-                </object>
-                <object class="sizeritem">
-                  <object class="wxTextCtrl" name="txtBatchBuildsCmdLine"/>
-                  <flag>wxLEFT|wxEXPAND</flag>
-                  <border>8</border>
-                </object>
-                <object class="sizeritem">
-                  <object class="wxStaticText" name="">
-                    <label>If you right-click a Code::Blocks project/workspace in explorer, you will see
+                  <object class="sizeritem">
+                    <option>0</option>
+                      <object class="wxBoxSizer">
+                        <orient>wxVERTICAL</orient>
+                        <object class="sizeritem">
+                          <object class="wxStaticText" name="">
+                            <label>Extra command-line arguments for batch-builds through shell extensions:</label>
+                          </object>
+                          <flag>wxLEFT|wxEXPAND</flag>
+                          <border>8</border>
+                        </object>
+                        <object class="sizeritem">
+                          <object class="wxTextCtrl" name="txtBatchBuildsCmdLine"/>
+                          <flag>wxLEFT|wxEXPAND</flag>
+                          <border>8</border>
+                        </object>
+                        <object class="sizeritem">
+                          <object class="wxStaticText" name="">
+                            <label>If you right-click a Code::Blocks project/workspace in explorer, you will see
 two extensions: &quot;Build&quot; and &quot;Rebuild (clean)&quot; (windows only).
 The arguments configured here, are used by those two extensions.
 You could specify, for example, a profile/personality to use for batch builds.
@@ -61,16 +65,20 @@
 --target: work only on &lt;target&gt; for &quot;Build&quot; or &quot;Rebuild (clean)&quot;
 --no-batch-window-close: do not auto-close log window when batch build is done
 --batch-build-notify: show message when batch build is done</label>
+                          </object>
+                          <flag>wxLEFT|wxEXPAND</flag>
+                          <border>8</border>
+                        </object>
+                      </object>
                   </object>
-                  <flag>wxLEFT|wxEXPAND</flag>
-                  <border>8</border>
-                </object>
                 <object class="sizeritem">
+                  <option>1</option>
                   <object class="wxStaticBoxSizer">
                     <label>Plugins to load in batch-build mode</label>
                     <orient>wxVERTICAL</orient>
                     <object class="sizeritem">
                       <object class="wxCheckListBox" name="chkBBPlugins">
+                        <style>wxLB_NEEDED_SB</style>
                         <content/>
                       </object>
                       <option>1</option>
@@ -89,18 +97,24 @@
                   <flag>wxTOP|wxBOTTOM|wxLEFT|wxGROW</flag>
                   <border>8</border>
                 </object>
-                <object class="spacer">
-                  <size>0,0</size>
-                  <option>1</option>
-                  <flag>wxGROW</flag>
-                </object>
                 <object class="sizeritem">
-                  <object class="wxStaticText">
-                    <label>For more info on profiles/personalities, read the Code::Blocks WiKi at
+                  <option>0</option>
+                  <object class="wxBoxSizer">
+                    <orient>wxVERTICAL</orient>
+                    <object class="spacer">
+                      <size>0,0</size>
+                      <option>1</option>
+                      <flag>wxGROW</flag>
+                    </object>
+                    <object class="sizeritem">
+                      <object class="wxStaticText">
+                        <label>For more info on profiles/personalities, read the Code::Blocks WiKi at
 http://wiki.codeblocks.org/index.php?title=Personalities</label>
+                      </object>
+                      <flag>wxLEFT|wxEXPAND</flag>
+                      <border>8</border>
+                    </object>
                   </object>
-                  <flag>wxLEFT|wxEXPAND</flag>
-                  <border>8</border>
                 </object>
               </object>
             </object>
@@ -137,4 +151,4 @@
     </object>
     <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxRESIZE_BOX|wxMAXIMIZE_BOX</style>
   </object>
-</resource>
\ No newline at end of file
+</resource>
