Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 4717)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -743,7 +743,8 @@
 cbProject* ProjectManager::LoadProject(const wxString& filename, bool activateIt)
 {
     cbProject* result = 0;
-    if (!BeginLoadingProject())
+
+    if (!wxFileExists(filename) || !BeginLoadingProject())
     {
     	return 0;
     }
@@ -1390,9 +1391,9 @@
             Manager::Get()->GetPluginManager()->NotifyPlugins(event);
         }
     }
-    
+
     project->EndAddFiles();
-    
+
     return targets.GetCount();
 }
 
