Index: projectmanager.cpp
===================================================================
--- projectmanager.cpp	(revision 4734)
+++ projectmanager.cpp	(working copy)
@@ -2072,7 +2072,7 @@
                 Manager::Get()->GetPluginManager()->NotifyPlugins(evt);
             }
             prj->CalculateCommonTopLevelPath();
-            prj->EndAddFiles();
+            prj->EndRemoveFiles();
             RebuildTree();
         }
     }
