--- codeblocks-1.0svn.orig/src/src/app.cpp	2007-12-07 20:03:08.000000000 +0100
+++ codeblocks-1.0svn.work/src/src/app.cpp	2007-12-11 00:21:48.000000000 +0100
@@ -900,6 +900,10 @@
                             m_HasProject = true;
                             s_DelayedFilesToOpen.Add(parser.GetParam(param));
                         }
+                        else if (ft == ftSource || ft == ftHeader || ft == ftResource)
+                        {
+                            s_DelayedFilesToOpen.Add(parser.GetParam(param));
+                        }
                         else if (ft == ftCodeBlocksWorkspace)
                         {
                             // only one workspace can be opened
