--- codeblocks-1.0svn.orig/src/plugins/debuggergdb/debuggergdb.cpp	2007-12-07 20:03:07.000000000 +0100
+++ codeblocks-1.0svn.work/src/plugins/debuggergdb/debuggergdb.cpp	2007-12-08 23:55:25.000000000 +0100
@@ -2463,10 +2463,13 @@
     // allow others to catch this
     event.Skip();
 
-    // remove all search dirs sotred for this project so we don't have conflicts
+    // remove all search dirs stored for this project so we don't have conflicts
     // if a newly opened project happens to use the same memory address
     GetSearchDirs(event.GetProject()).clear();
 
+    // clear the RemoteDebuggingMap for this project
+    m_RemoteDebugging.clear();
+
     // remove all breakpoints belonging to the closed project
     m_State.RemoveAllProjectBreakpoints(event.GetProject());
     if (m_pBreakpointsWindow)
