Index: update.bat
===================================================================
--- update.bat	(revision 4760)
+++ update.bat	(working copy)
@@ -16,6 +16,7 @@
 if not exist %CB_OUTPUT_RESDIR%\images\settings md %CB_OUTPUT_RESDIR%\images\settings\
 if not exist %CB_OUTPUT_RESDIR%\images\16x16 md %CB_OUTPUT_RESDIR%\images\16x16\
 if not exist %CB_OUTPUT_RESDIR%\images\codecompletion md %CB_OUTPUT_RESDIR%\images\codecompletion\
+if not exist %CB_OUTPUT_RESDIR%\images\wxsmith md %CB_OUTPUT_RESDIR%\images\wxsmith\
 if not exist %CB_OUTPUT_RESDIR%\plugins md %CB_OUTPUT_RESDIR%\plugins\
 if not exist %CB_OUTPUT_RESDIR%\templates md %CB_OUTPUT_RESDIR%\templates\
 if not exist %CB_OUTPUT_RESDIR%\templates\wizard md %CB_OUTPUT_RESDIR%\templates\wizard\
@@ -69,6 +70,7 @@
 copy /y setup\exchndl.dll devel > nul
 echo Copying files
 copy /y %CB_DEVEL_RESDIR%\*.zip %CB_OUTPUT_RESDIR% > nul
+copy /y %CB_DEVEL_RESDIR%\images\wxsmith\*.png %CB_OUTPUT_RESDIR%\images\wxsmith > nul
 copy /y sdk\resources\lexers\lexer_* %CB_DEVEL_RESDIR%\lexers > nul
 copy /y sdk\resources\lexers\lexer_* %CB_OUTPUT_RESDIR%\lexers > nul
 copy /y src\resources\images\*.png %CB_DEVEL_RESDIR%\images > nul
