Index: globals.cpp
===================================================================
--- globals.cpp	(revision 4764)
+++ globals.cpp	(working copy)
@@ -59,7 +59,7 @@
 #ifndef __WXMAC__
 const wxString DEFAULT_CONSOLE_TERM  = _T("xterm -T $TITLE -e");
 #else
-const wxString DEFAULT_CONSOLE_TERM  = _T("osascript -e 'tell app \"Terminal\" to do script \"$SCRIPT\"'");
+const wxString DEFAULT_CONSOLE_TERM  = _T("osascript -e 'tell app \"Terminal\"' -e 'activate' -e 'do script \"$SCRIPT\"' -e 'end tell'");
 #endif
 const wxString DEFAULT_CONSOLE_SHELL = _T("/bin/sh -c");