Index: src/sdk/infowindow.cpp
===================================================================
--- src/sdk/infowindow.cpp	(revision 4831)
+++ src/sdk/infowindow.cpp	(working copy)
@@ -19,8 +19,8 @@
 EVT_RIGHT_DOWN(InfoWindow::OnClick)
 END_EVENT_TABLE()
 
-const wxColour titleBackground(96,96,96); // dark grey
-const wxColour textBackground(255,255,160); // yellowish
+const wxColour titleBackground = wxSystemSettings::GetColour(wxSYS_COLOUR_ACTIVEBORDER);
+const wxColour textBackground = wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW);
 
 
 const char *iBitmap[] = {
Index: src/sdk/resources/editor_configuration.xrc
===================================================================
--- src/sdk/resources/editor_configuration.xrc	(revision 4831)
+++ src/sdk/resources/editor_configuration.xrc	(working copy)
@@ -39,7 +39,6 @@
 												<object class="sizeritem">
 													<object class="wxStaticText" name="lblEditorFont">
 														<label>lblEditorFont</label>
-														<bg>#FFFFFF</bg>
 													</object>
 													<flag>wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
 													<border>5</border>
