Index: src/plugins/debuggergdb/debuggergdb.cpp
===================================================================
--- src/plugins/debuggergdb/debuggergdb.cpp	(revision 4831)
+++ src/plugins/debuggergdb/debuggergdb.cpp	(working copy)
@@ -1948,9 +1948,9 @@
         mbar->Enable(idMenuContinue, m_pProcess && en && stopped);
         mbar->Enable(idMenuNext, m_pProcess && en && stopped);
         mbar->Enable(idMenuNextInstr, m_pProcess && en && stopped);
-        mbar->Enable(idMenuStep, en && stopped);
+        mbar->Enable(idMenuStep, m_pProcess && en && stopped);
         mbar->Enable(idMenuStepOut, m_pProcess && en && stopped);
-        mbar->Enable(idMenuRunToCursor, en && ed && stopped);
+        mbar->Enable(idMenuRunToCursor, m_pProcess && en && ed && stopped);
         mbar->Enable(idMenuToggleBreakpoint, en && ed && stopped);
         mbar->Enable(idMenuRemoveAllBreakpoints, en && ed && stopped);
         mbar->Enable(idMenuSendCommandToGDB, m_pProcess && stopped);
@@ -1977,10 +1977,10 @@
     #ifdef implement_debugger_toolbar
     wxToolBar* tbar = m_pTbar;//Manager::Get()->GetAppWindow()->GetToolBar();
     tbar->EnableTool(idMenuDebug, (!m_pProcess || stopped) && en);
-    tbar->EnableTool(idMenuRunToCursor, en && ed && stopped);
+    tbar->EnableTool(idMenuRunToCursor, m_pProcess && en && ed && stopped);
     tbar->EnableTool(idMenuNext, m_pProcess && en && stopped);
     tbar->EnableTool(idMenuNextInstr, m_pProcess && en && stopped);
-    tbar->EnableTool(idMenuStep, en && stopped);
+    tbar->EnableTool(idMenuStep, m_pProcess && en && stopped);
     tbar->EnableTool(idMenuStepOut, m_pProcess && en && stopped);
     tbar->EnableTool(idMenuStop, m_pProcess && en);
     tbar->EnableTool(idDebuggerToolInfo, m_pProcess && en); 