Index: wizard.script
===================================================================
--- wizard.script	(revision 4985)
+++ wizard.script	(working copy)
@@ -43,11 +43,14 @@
             return false;

 

         // verify include dependencies

-        local dir_nomacro_inc = GetCompilerIncludeDir(dir, IrrlichtPathDefault, IrrlichtPathDefaultInc);

+        local dir_nomacro_inc = GetCompilerIncludeDir(dir, IrrlichtPathDefault, IrrlichtPathDefaultInc);
+

         if (dir_nomacro_inc.IsEmpty())

-            return false;

+            return false;
+

         if (PLATFORM != PLATFORM_MSW)

-            dir_nomacro_inc = dir_nomacro_inc + _T("/irrlicht");

+     //       dir_nomacro_inc = dir_nomacro_inc;
+

         if (!VerifyFile(dir_nomacro_inc, _T("irrlicht.h"), _T("Irrlicht's include"))) return false;

 

         // verify library dependencies

@@ -95,7 +98,7 @@
         else

         {

             if (PLATFORM != PLATFORM_MSW)

-                IrrlichtPathDefaultInc = IrrlichtPathDefaultInc + _T("/irrlicht");

+                IrrlichtPathDefaultInc = _T("/irrlicht") + IrrlichtPathDefaultInc ;

         }

 

         // try to resolve the library directory as macro
