--- codeblocks-1.0svn.orig/src/plugins/debuggergdb/examinememorydlg.cpp	2008-03-01 09:41:45.000000000 +0100
+++ codeblocks-1.0svn.work/src/plugins/debuggergdb/examinememorydlg.cpp	2008-05-20 09:41:32.000000000 +0200
@@ -115,7 +115,7 @@
 
         long a;
         addr.ToLong(&a, 16);
-        m_pText->AppendText(wxString::Format(_T("0x%x: %s"), m_LastRowStartingAddress, m_LineText));
+        m_pText->AppendText(wxString::Format(_T("0x%x: %.67s"), m_LastRowStartingAddress, m_LineText));
         for (int i = 0; i < 67; ++i)
             m_LineText[i] = _T(' ');
         // update starting address for next row
