Index: src/plugins/codecompletion/parser/tokenizer.h
===================================================================
--- src/plugins/codecompletion/parser/tokenizer.h	(revision 5081)
+++ src/plugins/codecompletion/parser/tokenizer.h	(working copy)
@@ -131,7 +131,10 @@
 
     wxChar CurrentChar() const
     {
-        return m_Buffer.GetChar(m_TokenIndex);
+        if(m_TokenIndex < m_BufferLen)
+            return m_Buffer.GetChar(m_TokenIndex);
+        else
+            return 0;
     };
 
     wxChar CurrentCharMoveNext()
