--- codeblocks-1.0svn.orig/src/src/app.cpp	2008-05-01 07:11:56.000000000 +0200
+++ codeblocks-1.0svn.work/src/src/app.cpp	2008-05-25 12:24:12.000000000 +0200
@@ -780,9 +783,13 @@
 
     if (m_pBatchBuildDialog && m_BatchWindowAutoClose)
     {
-        m_pBatchBuildDialog->EndModal(wxID_OK);
-        m_pBatchBuildDialog->Destroy();
-        m_pBatchBuildDialog = 0;
+        if(m_pBatchBuildDialog->IsModal())
+            m_pBatchBuildDialog->EndModal(wxID_OK);
+        else
+        {
+            m_pBatchBuildDialog->Destroy();
+            m_pBatchBuildDialog = 0;
+        }
     }
 }
 
