Index: sdk/projectfile.cpp
===================================================================
--- sdk/projectfile.cpp	(revision 5106)
+++ sdk/projectfile.cpp	(working copy)
@@ -59,17 +59,17 @@
 	relativeFilename.IsEmpty() || relativeFilename.Append(wxFILE_SEP_PATH);
 	relativeFilename.Append(new_name);
 
-	UpdateFileDetails();
 	if (project)
 	{
 		project->ProjectFileRenamed(this);
 		project->CalculateCommonTopLevelPath();
 		project->SetModified(true);
 	}
+	UpdateFileDetails();
 }
