Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 5106)
+++ src/src/main.cpp	(working copy)
@@ -2577,6 +2577,9 @@
 
 void MainFrame::OnApplicationClose(wxCloseEvent& event)
 {
+    if (m_InitiatedShutdown)
+        return;
+
     CodeBlocksEvent evt(cbEVT_APP_START_SHUTDOWN);
     Manager::Get()->ProcessEvent(evt);
     Manager::Yield();