--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxschoice.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxschoice.h	2008-07-09 17:22:44.000000000 +0200
@@ -39,7 +39,7 @@
         virtual void OnEnumWidgetProperties(long Flags);

         wxArrayString ArrayChoices;
-        int DefaultSelection;
+        long DefaultSelection;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxscombobox.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxscombobox.h	2008-07-09 17:23:15.000000000 +0200
@@ -40,7 +40,7 @@


         wxArrayString ArrayChoices;
-        int DefaultSelection;
+        long DefaultSelection;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsgauge.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsgauge.h	2008-07-09 17:24:02.000000000 +0200
@@ -38,10 +38,10 @@
         virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
         virtual void OnEnumWidgetProperties(long Flags);

-        int Range;
-        int Value;
-        int Shadow;
-        int Bezel;
+        long Range;
+        long Value;
+        long Shadow;
+        long Bezel;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsgenericdirctrl.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsgenericdirctrl.h	2008-07-09 17:24:22.000000000 +0200
@@ -40,7 +40,7 @@

         wxString DefaultFolder;
     	wxString Filter;
-    	int DefaultFilter;
+    	long DefaultFilter;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxslistbox.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxslistbox.h	2008-07-09 17:24:54.000000000 +0200
@@ -40,7 +40,7 @@


         wxArrayString ArrayChoices;
-        int DefaultSelection;
+        long DefaultSelection;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsradiobox.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsradiobox.h	2008-07-09 17:18:51.000000000 +0200
@@ -40,8 +40,8 @@

         wxString Label;
         wxArrayString ArrayChoices;
-        int DefaultSelection;
-        int Dimension;
+        long DefaultSelection;
+        long Dimension;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsscrollbar.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsscrollbar.h	2008-07-09 17:26:09.000000000 +0200
@@ -38,10 +38,10 @@
         virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
         virtual void OnEnumWidgetProperties(long Flags);

-        int Value;
-    	int ThumbSize;
-    	int Range;
-    	int PageSize;
+        long Value;
+    	long ThumbSize;
+    	long Range;
+    	long PageSize;

 };

--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsslider.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsslider.h	2008-07-09 17:26:36.000000000 +0200
@@ -38,16 +38,16 @@
         virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
         virtual void OnEnumWidgetProperties(long Flags);

-        int Value;
-        int Min;
-        int Max;
-        int TickFrequency;
-        int PageSize;
-        int LineSize;
-        int ThumbLength;
-        int Tick;
-        int SelMin;
-        int SelMax;
+        long Value;
+        long Min;
+        long Max;
+        long TickFrequency;
+        long PageSize;
+        long LineSize;
+        long ThumbLength;
+        long Tick;
+        long SelMin;
+        long SelMax;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsspinbutton.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsspinbutton.h	2008-07-09 17:27:00.000000000 +0200
@@ -38,9 +38,9 @@
         virtual wxObject* OnBuildPreview(wxWindow* Parent,long Flags);
         virtual void OnEnumWidgetProperties(long Flags);

-        int Value;
-        int Min;
-        int Max;
+        long Value;
+        long Min;
+        long Max;
 };

 #endif
--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsspinctrl.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsspinctrl.h	2008-07-09 17:27:19.000000000 +0200
@@ -39,8 +39,8 @@
         virtual void OnEnumWidgetProperties(long Flags);

         wxString Value;
-        int Min;
-        int Max;
+        long Min;
+        long Max;

 };

--- codeblocks-1.0svn.orig/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstextctrl.h	2008-07-01 02:08:55.000000000 +0200
+++ codeblocks-1.0svn.work/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstextctrl.h	2008-07-09 17:29:14.000000000 +0200
@@ -39,7 +39,7 @@
         virtual void OnEnumWidgetProperties(long Flags);

         wxString Text;
-        int MaxLength;
+        long MaxLength;
 };

 #endif
