Index: debian/changelog
===================================================================
--- debian/changelog	(revision 5127)
+++ debian/changelog	(working copy)
@@ -1,4 +1,4 @@
-codeblocks (1.0svn5081) dapper; urgency=low
+codeblocks (1.0svn5127) dapper; urgency=low
 
   * Nightly linux build
 
Index: src/plugins/contrib/Valgrind/Valgrind.cpp
===================================================================
--- src/plugins/contrib/Valgrind/Valgrind.cpp	(revision 5127)
+++ src/plugins/contrib/Valgrind/Valgrind.cpp	(working copy)
@@ -13,6 +13,7 @@
 #include <wx/fs_zip.h>
 #include <wx/intl.h>
 #include <wx/string.h>
+#include <wx/menu.h>
 #include <wx/xrc/xmlres.h>
 #include "cbproject.h"
 #include "manager.h"
