Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 5170)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -1170,6 +1170,9 @@
     if (!Open(detectEncoding))
         return false;
 
+    // Re-establish margin styles, width,  etc
+    SetEditorStyleAfterFileOpen();
+
     // return (if possible) to old pos
     if (m_pControl)
         m_pControl->GotoPos(pos); 