Index: src/plugins/contrib/profiler/cbprofiler.cbp
===================================================================
--- src/plugins/contrib/profiler/cbprofiler.cbp	(revision 5202)
+++ src/plugins/contrib/profiler/cbprofiler.cbp	(working copy)
@@ -27,7 +27,6 @@
 			<Add option="-Winvalid-pch" />
 			<Add option="-mthreads" />
 			<Add option="-fexceptions" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
Index: src/plugins/contrib/envvars/envvars.cbp
===================================================================
--- src/plugins/contrib/envvars/envvars.cbp	(revision 5202)
+++ src/plugins/contrib/envvars/envvars.cbp	(working copy)
@@ -43,7 +43,6 @@
 			<Add option="-Winvalid-pch" />
 			<Add option="-mthreads" />
 			<Add option="-fexceptions" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DHAVE_W32API_H" />
 			<Add option="-D__WXMSW__" />
Index: src/plugins/contrib/wxSmithContribItems/wxSmithContribItems.cbp
===================================================================
--- src/plugins/contrib/wxSmithContribItems/wxSmithContribItems.cbp	(revision 5202)
+++ src/plugins/contrib/wxSmithContribItems/wxSmithContribItems.cbp	(working copy)
@@ -29,7 +29,6 @@
 			<Add option="-fmessage-length=0" />
 			<Add option="-fexceptions" />
 			<Add option="-Winvalid-pch" />
-			<Add option="-include sdk.h" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DHAVE_W32API_H" />
 			<Add option="-D__WXMSW__" />
Index: src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
===================================================================
--- src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(revision 5202)
+++ src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp	(working copy)
@@ -28,7 +28,6 @@
 					<Add option="-Winvalid-pch" />
 					<Add option="-mthreads" />
 					<Add option="-fexceptions" />
-					<Add option='-include &quot;sdk.h&quot;' />
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add option="-DCB_PRECOMP" />
 					<Add option="-DWX_PRECOMP" />
Index: src/plugins/contrib/source_exporter/Exporter.cbp
===================================================================
--- src/plugins/contrib/source_exporter/Exporter.cbp	(revision 5202)
+++ src/plugins/contrib/source_exporter/Exporter.cbp	(working copy)
@@ -45,7 +45,6 @@
 			<Add option="-g" />
 			<Add option="-Winvalid-pch" />
 			<Add option="-mthreads" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-fexceptions" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DCB_PRECOMP" />
Index: src/plugins/contrib/dragscroll/DragScroll.cbp
===================================================================
--- src/plugins/contrib/dragscroll/DragScroll.cbp	(revision 5202)
+++ src/plugins/contrib/dragscroll/DragScroll.cbp	(working copy)
@@ -29,7 +29,6 @@
 			<Add option="-mthreads" />
 			<Add option="-fmessage-length=0" />
 			<Add option="-fexceptions" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
Index: src/plugins/contrib/symtab/symtab.cbp
===================================================================
--- src/plugins/contrib/symtab/symtab.cbp	(revision 5202)
+++ src/plugins/contrib/symtab/symtab.cbp	(working copy)
@@ -31,7 +31,6 @@
 			<Add option="-Winvalid-pch" />
 			<Add option="-mthreads" />
 			<Add option="-fexceptions" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
Index: src/plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith.cbp	(revision 5202)
+++ src/plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -53,7 +53,6 @@
 			<Add option="-fmessage-length=0" />
 			<Add option="-fexceptions" />
 			<Add option="-Winvalid-pch" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-DHAVE_W32API_H" />
 			<Add option="-D__WXMSW__" />
 			<Add option="-DWXUSINGDLL" />
Index: src/plugins/contrib/help_plugin/help-plugin.cbp
===================================================================
--- src/plugins/contrib/help_plugin/help-plugin.cbp	(revision 5202)
+++ src/plugins/contrib/help_plugin/help-plugin.cbp	(working copy)
@@ -36,7 +36,6 @@
 					<Add option="-mthreads" />
 					<Add option="-fmessage-length=0" />
 					<Add option="-fexceptions" />
-					<Add option='-include &quot;sdk.h&quot;' />
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add option="-DCB_PRECOMP" />
 					<Add option="-DWX_PRECOMP" />
Index: src/plugins/contrib/codestat/codestat.cbp
===================================================================
--- src/plugins/contrib/codestat/codestat.cbp	(revision 5202)
+++ src/plugins/contrib/codestat/codestat.cbp	(working copy)
@@ -28,7 +28,6 @@
 			<Add option="-Winvalid-pch" />
 			<Add option="-mthreads" />
 			<Add option="-fexceptions" />
-			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DWX_PRECOMP" />
Index: src/plugins/contrib/keybinder/keybinder.cbp
===================================================================
--- src/plugins/contrib/keybinder/keybinder.cbp	(revision 5202)
+++ src/plugins/contrib/keybinder/keybinder.cbp	(working copy)
@@ -18,7 +18,6 @@
 					<Add option="-mthreads" />
 					<Add option="-fmessage-length=0" />
 					<Add option="-fexceptions" />
-					<Add option='-include &quot;sdk.h&quot;' />
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add option="-DCB_PRECOMP" />
 					<Add option="-DWX_PRECOMP" />
Index: src/CodeBlocks.cbp
===================================================================
--- src/CodeBlocks.cbp	(revision 5202)
+++ src/CodeBlocks.cbp	(working copy)
@@ -369,7 +369,6 @@
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Compiler>
-					<Add option="-include sdk.h" />
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
 				</Compiler>
@@ -387,7 +386,6 @@
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Compiler>
-					<Add option="-include sdk.h" />
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
 					<Add directory="include\scripting\include" />
Index: src/include/editorcolourset.h
===================================================================
--- src/include/editorcolourset.h	(revision 5203)
+++ src/include/editorcolourset.h	(working copy)
@@ -11,10 +11,8 @@
 #include <wx/intl.h>
 #include <wx/wxscintilla.h> // wxSCI_KEYWORDSET_MAX
 #include "settings.h"
+#include "globals.h"        // HighlightLanguage
 
-#ifndef CB_PRECOMP
-    #include "globals.h" // HighlightLanguage
-#endif
 
 // forward decls
 class cbEditor;
