Index: src/plugins/contrib/ThreadSearch/SearchInPanel.cpp
===================================================================
--- src/plugins/contrib/ThreadSearch/SearchInPanel.cpp	(revision 5207)
+++ src/plugins/contrib/ThreadSearch/SearchInPanel.cpp	(working copy)
@@ -157,11 +157,11 @@
 {
     // begin wxGlade: SearchInPanel::do_layout
     wxBoxSizer* SizerTop = new wxBoxSizer(wxHORIZONTAL);
-    SizerTop->Add(m_pChkSearchOpenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchTargetFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchProjectFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchWorkspaceFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchDir, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pChkSearchOpenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchTargetFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchProjectFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchWorkspaceFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchDir, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     SetAutoLayout(true);
     SetSizer(SizerTop);
     SizerTop->Fit(this);
Index: src/plugins/contrib/ThreadSearch/ThreadSearchView.cpp
===================================================================
--- src/plugins/contrib/ThreadSearch/ThreadSearchView.cpp	(revision 5207)
+++ src/plugins/contrib/ThreadSearch/ThreadSearchView.cpp	(working copy)
@@ -291,25 +291,25 @@
     wxBoxSizer* m_pSizerSearchPreview = new wxBoxSizer(wxHORIZONTAL);
     m_pSizerSearchDirItems = new wxStaticBoxSizer(m_pSizerSearchDirItems_staticbox, wxHORIZONTAL);
     m_pSizerSearchItems = new wxBoxSizer(wxHORIZONTAL);
-    m_pSizerSearchItems->Add(m_pCboSearchExpr, 2, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    m_pSizerSearchItems->Add(m_pBtnSearch, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    m_pSizerSearchItems->Add(m_pBtnOptions, 0, wxALL|wxADJUST_MINSIZE, 4);
-    m_pSizerSearchItems->Add(30, 20, 0, wxADJUST_MINSIZE, 0);
-    m_pSizerSearchItems->Add(m_pStaTxtSearchIn, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    m_pSizerSearchItems->Add(m_pPnlSearchIn, 0, wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 0);
-    m_pSizerSearchItems->Add(30, 20, 0, wxADJUST_MINSIZE, 0);
-    m_pSizerSearchItems->Add(m_pBtnShowDirItems, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    m_pSizerSearchItems->Add(m_pCboSearchExpr, 2, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    m_pSizerSearchItems->Add(m_pBtnSearch, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    m_pSizerSearchItems->Add(m_pBtnOptions, 0, wxALL, 4);
+    m_pSizerSearchItems->Add(30, 20, 0, 0, 0);
+    m_pSizerSearchItems->Add(m_pStaTxtSearchIn, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    m_pSizerSearchItems->Add(m_pPnlSearchIn, 0, wxALIGN_CENTER_VERTICAL, 0);
+    m_pSizerSearchItems->Add(30, 20, 0, 0, 0);
+    m_pSizerSearchItems->Add(m_pBtnShowDirItems, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     m_pSizerTop->Add(m_pSizerSearchItems, 0, wxEXPAND|wxALIGN_CENTER_VERTICAL, 0);
     m_pSizerSearchDirItems->Add(m_pPnlDirParams, 1, wxALIGN_CENTER_VERTICAL, 0);
     m_pSizerTop->Add(m_pSizerSearchDirItems, 0, wxBOTTOM|wxEXPAND, 4);
-    m_pSizerSearchPreview->Add(m_pSearchPreview, 1, wxEXPAND|wxADJUST_MINSIZE, 0);
+    m_pSizerSearchPreview->Add(m_pSearchPreview, 1, wxEXPAND, 0);
     m_pPnlPreview->SetAutoLayout(true);
     m_pPnlPreview->SetSizer(m_pSizerSearchPreview);
     m_pSizerListLog->Add(m_pLogger->GetWindow(), 1, wxEXPAND|wxFIXED_MINSIZE, 0);
     m_pPnlListLog->SetAutoLayout(true);
     m_pPnlListLog->SetSizer(m_pSizerListLog);
     m_pSplitter->SplitVertically(m_pPnlPreview, m_pPnlListLog);
-    m_pSizerSplitter->Add(m_pSplitter, 1, wxEXPAND|wxADJUST_MINSIZE, 0);
+    m_pSizerSplitter->Add(m_pSplitter, 1, wxEXPAND, 0);
     m_pSizerTop->Add(m_pSizerSplitter, 1, wxEXPAND, 0);
     SetAutoLayout(true);
     SetSizer(m_pSizerTop);
Index: src/plugins/contrib/ThreadSearch/DirectoryParamsPanel.cpp
===================================================================
--- src/plugins/contrib/ThreadSearch/DirectoryParamsPanel.cpp	(revision 5207)
+++ src/plugins/contrib/ThreadSearch/DirectoryParamsPanel.cpp	(working copy)
@@ -101,13 +101,13 @@
 {
     // begin wxGlade: DirectoryParamsPanel::do_layout
     wxBoxSizer* SizerTop = new wxBoxSizer(wxHORIZONTAL);
-    SizerTop->Add(m_pTxtSearchDirPath, 2, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pBtnSelectDir, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchDirRecursively, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchDirHiddenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pTxtMask, 1, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pTxtSearchDirPath, 2, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pBtnSelectDir, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchDirRecursively, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchDirHiddenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pTxtMask, 1, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     wxStaticText* m_pStatTxtMask = new wxStaticText(this, -1, wxT("mask"));
-    SizerTop->Add(m_pStatTxtMask, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pStatTxtMask, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     SetAutoLayout(true);
     SetSizer(SizerTop);
     SizerTop->Fit(this);
Index: src/plugins/contrib/codesnippets/Search/SearchInPanel.cpp
===================================================================
--- src/plugins/contrib/codesnippets/Search/SearchInPanel.cpp	(revision 5207)
+++ src/plugins/contrib/codesnippets/Search/SearchInPanel.cpp	(working copy)
@@ -108,10 +108,10 @@
 {
     // begin wxGlade: SearchInPanel::do_layout
     wxBoxSizer* SizerTop = new wxBoxSizer(wxHORIZONTAL);
-    SizerTop->Add(m_pChkSearchOpenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchSnippetFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pChkSearchOpenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchSnippetFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     //-SizerTop->Add(m_pChkSearchWorkspaceFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchDir, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pChkSearchDir, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     SetAutoLayout(true);
     SetSizer(SizerTop);
     SizerTop->Fit(this);
Index: src/plugins/contrib/codesnippets/Search/ThreadSearchView.cpp
===================================================================
--- src/plugins/contrib/codesnippets/Search/ThreadSearchView.cpp	(revision 5207)
+++ src/plugins/contrib/codesnippets/Search/ThreadSearchView.cpp	(working copy)
@@ -325,25 +325,25 @@
     wxBoxSizer* m_pSizerSearchPreview = new wxBoxSizer(wxHORIZONTAL);
     m_pSizerSearchDirItems = new wxStaticBoxSizer(m_pSizerSearchDirItems_staticbox, wxHORIZONTAL);
     m_pSizerSearchItems = new wxBoxSizer(wxHORIZONTAL);
-    m_pSizerSearchItems->Add(m_pCboSearchExpr, 2, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    m_pSizerSearchItems->Add(m_pBtnSearch, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    m_pSizerSearchItems->Add(30, 20, 0, wxADJUST_MINSIZE, 0);
-    m_pSizerSearchItems->Add(m_pBtnOptions, 0, wxALL|wxADJUST_MINSIZE, 4);
-    //-m_pSizerSearchItems->Add(m_pStaTxtSearchIn, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    //- m_pSizerSearchItems->Add(m_pPnlSearchIn, 0, wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 0);
+    m_pSizerSearchItems->Add(m_pCboSearchExpr, 2, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    m_pSizerSearchItems->Add(m_pBtnSearch, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    m_pSizerSearchItems->Add(30, 20, 0, 0, 0);
+    m_pSizerSearchItems->Add(m_pBtnOptions, 0, wxALL, 4);
+    //-m_pSizerSearchItems->Add(m_pStaTxtSearchIn, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    //- m_pSizerSearchItems->Add(m_pPnlSearchIn, 0, wxALIGN_CENTER_VERTICAL, 0);
     //-m_pSizerSearchItems->Add(30, 20, 0, wxADJUST_MINSIZE, 0);
-    m_pSizerSearchItems->Add(m_pBtnShowDirItems, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    m_pSizerSearchItems->Add(m_pBtnShowDirItems, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     m_pSizerTop->Add(m_pSizerSearchItems, 0, wxEXPAND|wxALIGN_CENTER_VERTICAL, 0);
     m_pSizerSearchDirItems->Add(m_pPnlDirParams, 1, wxALIGN_CENTER_VERTICAL, 0);
     m_pSizerTop->Add(m_pSizerSearchDirItems, 0, wxBOTTOM|wxEXPAND, 4);
-    m_pSizerSearchPreview->Add(m_pSearchPreview, 1, wxEXPAND|wxADJUST_MINSIZE, 0);
+    m_pSizerSearchPreview->Add(m_pSearchPreview, 1, wxEXPAND, 0);
     m_pPnlPreview->SetAutoLayout(true);
     m_pPnlPreview->SetSizer(m_pSizerSearchPreview);
     m_pSizerListLog->Add(m_pLogger->GetWindow(), 1, wxEXPAND|wxFIXED_MINSIZE, 0);
     m_pPnlListLog->SetAutoLayout(true);
     m_pPnlListLog->SetSizer(m_pSizerListLog);
     m_pSplitter->SplitVertically(m_pPnlPreview, m_pPnlListLog);
-    m_pSizerSplitter->Add(m_pSplitter, 1, wxEXPAND|wxADJUST_MINSIZE, 0);
+    m_pSizerSplitter->Add(m_pSplitter, 1, wxEXPAND, 0);
     m_pSizerTop->Add(m_pSizerSplitter, 1, wxEXPAND, 0);
     SetAutoLayout(true);
     SetSizer(m_pSizerTop);
Index: src/plugins/contrib/codesnippets/Search/ThreadSearchConfPanel.cpp
===================================================================
--- src/plugins/contrib/codesnippets/Search/ThreadSearchConfPanel.cpp	(revision 5207)
+++ src/plugins/contrib/codesnippets/Search/ThreadSearchConfPanel.cpp	(working copy)
@@ -285,33 +285,33 @@
     SizerSearchIn->Add(m_pPnlSearchIn, 0, wxALL|wxEXPAND, 2);
     SizerSearchIn->Add(m_pPnlDirParams, 0, wxALL|wxEXPAND|wxALIGN_CENTER_VERTICAL, 2);
     SizerTop->Add(SizerSearchIn, 0, wxALL|wxEXPAND, 4);
-    SizerOptions->Add(m_pChkWholeWord, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerOptions->Add(m_pChkStartWord, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerOptions->Add(m_pChkMatchCase, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerOptions->Add(m_pChkRegExp, 0, wxALL|wxADJUST_MINSIZE, 4);
+    SizerOptions->Add(m_pChkWholeWord, 0, wxALL, 4);
+    SizerOptions->Add(m_pChkStartWord, 0, wxALL, 4);
+    SizerOptions->Add(m_pChkMatchCase, 0, wxALL, 4);
+    SizerOptions->Add(m_pChkRegExp, 0, wxALL, 4);
     SizerTop->Add(SizerOptions, 0, wxALL|wxEXPAND, 4);
-    SizerThreadSearchOptions->Add(m_pChkThreadSearchEnable, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchOptions->Add(m_pChkUseDefaultOptionsForThreadSearch, 0, wxALL|wxADJUST_MINSIZE, 4);
+    SizerThreadSearchOptions->Add(m_pChkThreadSearchEnable, 0, wxALL, 4);
+    SizerThreadSearchOptions->Add(m_pChkUseDefaultOptionsForThreadSearch, 0, wxALL, 4);
     //(pecan 2008/5/22) not supported yet
     //wxStaticText* m_pStaDefaultOptions = new wxStaticText(this, wxID_ANY, wxT("       ('Whole word' = true, 'Start word' = false, 'Match case' = true, 'Regular expression' = false)"));
     wxStaticText* m_pStaDefaultOptions = new wxStaticText(this, wxID_ANY, wxT(" Place holder for yet to be supported context menu option."));
-    SizerThreadSearchOptions->Add(m_pStaDefaultOptions, 0, wxADJUST_MINSIZE, 0);
+    SizerThreadSearchOptions->Add(m_pStaDefaultOptions, 0, 0, 0);
     SizerTop->Add(SizerThreadSearchOptions, 0, wxALL|wxEXPAND, 4);
-    //--moved--belo--SizerThreadSearchLayoutGlobal->Add(m_pChkShowThreadSearchToolBar, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchLayoutGlobal->Add(m_pChkShowThreadSearchWidgets, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchLayoutGlobal->Add(m_pChkShowCodePreview, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchLayoutGlobal->Add(m_pChkShowThreadSearchToolBar, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchGridLayout->Add(SizerThreadSearchLayoutGlobal, 1, wxALL|wxEXPAND|wxADJUST_MINSIZE, 4);
-    SizerListControlOptions->Add(m_pChkDisplayLogHeaders, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerListControlOptions->Add(m_pChkDrawLogLines, 0, wxALL|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchGridLayout->Add(SizerListControlOptions, 1, wxALL|wxEXPAND|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchGridLayout->Add(m_pRadPanelManagement, 0, wxALL|wxEXPAND|wxADJUST_MINSIZE, 4);
-    SizerThreadSearchGridLayout->Add(m_pRadLoggerType, 0, wxALL|wxEXPAND|wxADJUST_MINSIZE, 4);
+    //--moved--belo--SizerThreadSearchLayoutGlobal->Add(m_pChkShowThreadSearchToolBar, 0, wxALL, 4);
+    SizerThreadSearchLayoutGlobal->Add(m_pChkShowThreadSearchWidgets, 0, wxALL, 4);
+    SizerThreadSearchLayoutGlobal->Add(m_pChkShowCodePreview, 0, wxALL, 4);
+    SizerThreadSearchLayoutGlobal->Add(m_pChkShowThreadSearchToolBar, 0, wxALL, 4);
+    SizerThreadSearchGridLayout->Add(SizerThreadSearchLayoutGlobal, 1, wxALL|wxEXPAND, 4);
+    SizerListControlOptions->Add(m_pChkDisplayLogHeaders, 0, wxALL, 4);
+    SizerListControlOptions->Add(m_pChkDrawLogLines, 0, wxALL, 4);
+    SizerThreadSearchGridLayout->Add(SizerListControlOptions, 1, wxALL|wxEXPAND, 4);
+    SizerThreadSearchGridLayout->Add(m_pRadPanelManagement, 0, wxALL|wxEXPAND, 4);
+    SizerThreadSearchGridLayout->Add(m_pRadLoggerType, 0, wxALL|wxEXPAND, 4);
     SizerThreadSearchGridLayout->Add(m_pRadSplitterWndMode, 0, wxALL|wxEXPAND, 4);
     SizerThreadSearchGridLayout->Add(m_pRadSortBy, 0, wxALL|wxEXPAND, 4);
     SizerThreadSearchGridLayout->AddGrowableCol(0);
     SizerThreadSearchGridLayout->AddGrowableCol(1);
-    SizerThreadSearchLayout->Add(SizerThreadSearchGridLayout, 1, wxALL|wxEXPAND|wxADJUST_MINSIZE, 0);
+    SizerThreadSearchLayout->Add(SizerThreadSearchGridLayout, 1, wxALL|wxEXPAND, 0);
     SizerTop->Add(SizerThreadSearchLayout, 0, wxALL|wxEXPAND, 4);
     SetAutoLayout(true);
     SetSizer(SizerTop);
Index: src/plugins/contrib/codesnippets/Search/DirectoryParamsPanel.cpp
===================================================================
--- src/plugins/contrib/codesnippets/Search/DirectoryParamsPanel.cpp	(revision 5207)
+++ src/plugins/contrib/codesnippets/Search/DirectoryParamsPanel.cpp	(working copy)
@@ -101,13 +101,13 @@
 {
     // begin wxGlade: DirectoryParamsPanel::do_layout
     wxBoxSizer* SizerTop = new wxBoxSizer(wxHORIZONTAL);
-    SizerTop->Add(m_pTxtSearchDirPath, 2, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pBtnSelectDir, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchDirRecursively, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pChkSearchDirHiddenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
-    SizerTop->Add(m_pTxtMask, 1, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pTxtSearchDirPath, 2, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pBtnSelectDir, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchDirRecursively, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pChkSearchDirHiddenFiles, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
+    SizerTop->Add(m_pTxtMask, 1, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     wxStaticText* m_pStatTxtMask = new wxStaticText(this, -1, wxT("mask"));
-    SizerTop->Add(m_pStatTxtMask, 0, wxALL|wxALIGN_CENTER_VERTICAL|wxADJUST_MINSIZE, 4);
+    SizerTop->Add(m_pStatTxtMask, 0, wxALL|wxALIGN_CENTER_VERTICAL, 4);
     SetAutoLayout(true);
     SetSizer(SizerTop);
     SizerTop->Fit(this);
