From 67b8f4a7b1805ded8db30c56b6e2bf3dda04413d Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Mon, 19 Jan 2009 16:01:05 +0100
Subject: [PATCH] desktop file cleanup

---
 src/mime/codeblocks.desktop |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/mime/codeblocks.desktop b/src/mime/codeblocks.desktop
index 4392a66..5449cf7 100644
--- a/src/mime/codeblocks.desktop
+++ b/src/mime/codeblocks.desktop
@@ -1,13 +1,12 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
 Name=Code::Blocks IDE
 Comment=Configurable and extensible IDE
 Exec=codeblocks %F
-Icon=codeblocks.png
+Icon=codeblocks
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=Application;Development;
+Categories=Development;GTK;
 StartupNotify=true
 MimeType=application/x-codeblocks;application/x-codeblocks-workspace;
-- 
1.6.0.6

From 1db6aa55bdf33952ad84a4d2cc7c9ee22e614ae1 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 28 Feb 2009 12:57:19 +0100
Subject: [PATCH] add IDE category

---
 src/mime/codeblocks.desktop |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/mime/codeblocks.desktop b/src/mime/codeblocks.desktop
index 5449cf7..cbbf285 100644
--- a/src/mime/codeblocks.desktop
+++ b/src/mime/codeblocks.desktop
@@ -7,6 +7,6 @@ Icon=codeblocks
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=Development;GTK;
+Categories=Development;GTK;IDE;
 StartupNotify=true
 MimeType=application/x-codeblocks;application/x-codeblocks-workspace;
-- 
1.6.0.6

