Index: compilergcc.cpp
===================================================================
--- compilergcc.cpp	(revision 5679)
+++ compilergcc.cpp	(working copy)
@@ -3715,6 +3715,7 @@
             #else
             LogWarningOrError(cltNormal, 0, wxEmptyString, wxEmptyString, wxString::Format(_("=== Build finished: %s ==="), msg.c_str()));
             #endif
+            m_pListLog->GetListControl()->SetColumnWidth(2, wxLIST_AUTOSIZE);
             SaveBuildLog();
         }
         else
Index: compilermessages.h
===================================================================
--- compilermessages.h	(revision 5679)
+++ compilermessages.h	(working copy)
@@ -21,6 +21,7 @@
         virtual void FocusError(int nr);
 
         virtual wxWindow* CreateControl(wxWindow* parent);
+        wxListCtrl* GetListControl(){ return control ; }
     private:
         void OnClick(wxCommandEvent& event);
         void OnDoubleClick(wxCommandEvent& event);