Index: parserthread.cpp
===================================================================
--- parserthread.cpp	(revision 5696)
+++ parserthread.cpp	(working copy)
@@ -1228,7 +1228,7 @@
             {
                 static size_t num = 0;
                 wxString unnamedTmp;
-                unnamedTmp.Printf(_T("Unnamed-%s-%d"),
+                unnamedTmp.Printf(_T("Unnamed%s%d"),
                                   ct == ctClass ? _T("Class") :
                                   ct == ctUnion ? _T("Union") :
                                                   _T("Struct"), num++);
@@ -1460,7 +1472,7 @@
         if (m_ParsingTypedef)
         {
             static size_t num = 0;
-            token.Printf(_T("Unnamed-Enum-%d"), num++);
+            token.Printf(_T("UnnamedEnum%d"), num++);
             m_LastUnnamedTokenName = token;
         }
         else