Index: src/sdk/encodingdetector.cpp
===================================================================
--- src/sdk/encodingdetector.cpp	(revision 5716)
+++ src/sdk/encodingdetector.cpp	(working copy)
@@ -189,7 +189,7 @@
             wxCSConv conv(m_Encoding);

             wideBuff = conv.cMB2WC((char*)buffer, size + 4 - m_BOMSizeInBytes, &outlen);

         }

-        delete tmp;

+        delete[] tmp;

     }

 

     m_ConvStr = wxString(wideBuff);
