Index: src/src/app.cpp
===================================================================
--- src/src/app.cpp	(revision 5719)
+++ src/src/app.cpp	(working copy)
@@ -581,6 +581,15 @@
                 {
                     connection->Execute(_T("[Raise]"));
                 }
+
+                connection->Disconnect();
+                delete connection;
+            }
+
+            delete client;
+
+            if (connection)
+            {
                 // return false to end the application
                 return false;
             }
@@ -642,6 +651,7 @@
             if(loader->GetData())
                 Manager::Get()->GetScriptingManager()->LoadBuffer(cbC2U(loader->GetData()));
 
+            delete loader;
             frame->Close();
             return true;
         }