Index: src/sdk/manager.cpp
===================================================================
--- src/sdk/manager.cpp	(revision 5716)
+++ src/sdk/manager.cpp	(working copy)
@@ -81,6 +81,16 @@
 		}
 	}
 
+	for (LogEventSinksMap::iterator mit = m_LogEventSinks.begin(); mit != m_LogEventSinks.end(); ++mit)
+	{
+		LogEventSinksArray::iterator it = mit->second.begin();
+		while (mit->second.size())
+		{
+			delete (*(mit->second.begin()));
+			mit->second.erase(mit->second.begin());
+		}
+	}
+
 //    Shutdown();
     CfgMgrBldr::Free(); // only terminate config at the very last moment
 //	FileManager::Free();