Index: src/plugins/contrib/BrowseTracker/BrowseTracker.cpp
===================================================================
--- src/plugins/contrib/BrowseTracker/BrowseTracker.cpp	(revision 5716)
+++ src/plugins/contrib/BrowseTracker/BrowseTracker.cpp	(working copy)
@@ -194,6 +194,10 @@
 // ----------------------------------------------------------------------------

 {

     //dtor
+    if (m_pCfgFile)
+    {
+        delete m_pCfgFile;
+    }
 }
 
 // ----------------------------------------------------------------------------
