Index: sdk/editormanager.cpp
===================================================================
--- sdk/editormanager.cpp       (revision 5865)
+++ sdk/editormanager.cpp       (working copy)
@@ -984,6 +984,7 @@
         if (opf)
         {
             wxString const &activeName = opf->file.GetName();
+            wxString const &activeDir = opf->file.GetPath(wxPATH_GET_VOLUME);

             // first try to find the file among the opened files
             for (int i = 0; i < GetEditorsCount(); ++i)
@@ -996,7 +997,7 @@
                 if (!pf)
                     continue;

-                if (pf->file.GetName() == activeName)
+                if ((pf->file.GetName() == activeName) && (pf->file.GetPath(wxPATH_GET_VOLUME) == activeDir))
                 {
                     wxFileName const & fname = pf->file;
                     FileType ft_other = FileTypeOf(fname.GetFullName());
@@ -1027,7 +1028,7 @@
                 if (dirs.Index(dir) == wxNOT_FOUND)
                     dirs.Add(dir);

-                if (pf->file.GetName() == activeName)
+                if ((pf->file.GetName() == activeName) && (pf->file.GetPath(wxPATH_GET_VOLUME) == activeDir))
                 {
                     wxFileName const & fname = pf->file;
                     FileType ft_other = FileTypeOf(fname.GetFullName());
