Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 5892)
+++ src/src/main.cpp	(working copy)
@@ -2572,16 +2572,14 @@
 void MainFrame::OnFileSaveProject(wxCommandEvent& event)
 {
     // no need to call SaveActiveProjectAs(), because this is handled in cbProject::Save()
-    if (Manager::Get()->GetProjectManager()->SaveActiveProject())
-        AddToRecentProjectsHistory(Manager::Get()->GetProjectManager()->GetActiveProject()->GetFilename());
+    Manager::Get()->GetProjectManager()->SaveActiveProject();
     DoUpdateStatusBar();
     DoUpdateAppTitle();
 }
 
 void MainFrame::OnFileSaveProjectAs(wxCommandEvent& event)
 {
-    if (Manager::Get()->GetProjectManager()->SaveActiveProjectAs())
-        AddToRecentProjectsHistory(Manager::Get()->GetProjectManager()->GetActiveProject()->GetFilename());
+    Manager::Get()->GetProjectManager()->SaveActiveProjectAs();
     DoUpdateStatusBar();
     DoUpdateAppTitle();
 }
