Index: src/sdk/multiselectdlg.cpp
===================================================================
--- src/sdk/multiselectdlg.cpp	(revision 5986)
+++ src/sdk/multiselectdlg.cpp	(working copy)
@@ -124,7 +124,7 @@
             bool MatchesWildCard = false;
 			for (unsigned int x = 0; x < wilds.GetCount(); ++x)
 			{
-				if (entry.Matches(wilds[x]))
+				if (entry.Matches(wilds[x].Lower()))
 				{
                     lst->Check(i, select);
                     MatchesWildCard = true;
