Index: src/src/startherepage.cpp
===================================================================
--- src/src/startherepage.cpp	(revision 5986)
+++ src/src/startherepage.cpp	(working copy)
@@ -179,13 +179,17 @@
     || href.StartsWith(_T("http://developer.berlios.de/bugs/")))
     {
         wxTextDataObject *data = new wxTextDataObject(revInfo);
+        wxTheClipboard->Open();
         wxTheClipboard->SetData(data);
+        wxTheClipboard->Close();
     }
 
     if(href.IsSameAs(_T("rev")))
     {
         wxTextDataObject *data = new wxTextDataObject(revInfo);
+        wxTheClipboard->Open();
         wxTheClipboard->SetData(data);
+        wxTheClipboard->Close();
         return true;
     }
 
