Index: src/src/app.cpp
===================================================================
--- src/src/app.cpp	(revision 5986)
+++ src/src/app.cpp	(working copy)
@@ -489,6 +489,9 @@
 
     wxTheClipboard->Flush();
 
+    wxCmdLineParser& parser = *Manager::GetCmdLineParser();
+    parser.SetDesc(cmdLineDesc);
+
     // NOTE: crash handler explicitly disabled because it causes problems
     //       with plugins loading/unloading...
     //
@@ -533,8 +536,6 @@
             if(connection)
             {
                 wxArrayString strFilesInCommandLine;
-                wxCmdLineParser& parser = *Manager::GetCmdLineParser();
-                parser.SetDesc(cmdLineDesc);
                 parser.SetCmdLine(argc, argv);
 
                 // search for valid filenames passed as argument on commandline
@@ -1036,7 +1037,6 @@
 
 #if wxUSE_CMDLINE_PARSER
     wxCmdLineParser& parser = *Manager::GetCmdLineParser();
-    parser.SetDesc(cmdLineDesc);
     parser.SetCmdLine(argc, argv);
     // wxApp::argc is a wxChar**
 
