Index: src/plugins/openfileslist/openfileslistplugin.cpp
===================================================================
--- src/plugins/openfileslist/openfileslistplugin.cpp	(revision 5986)
+++ src/plugins/openfileslist/openfileslistplugin.cpp	(working copy)
@@ -257,7 +257,6 @@
         if(mgr->GetActiveEditor() == ed)
             m_pTree->SelectItem(item);
         m_pTree->SortChildren(m_pTree->GetRootItem());
-        m_pTree->Expand(m_pTree->GetRootItem());
     }
 
     m_pTree->Thaw();
