Index: src/sdk/wxscintilla/src/ScintillaWX.cpp
===================================================================
--- src/sdk/wxscintilla/src/ScintillaWX.cpp	(revision 6015)
+++ src/sdk/wxscintilla/src/ScintillaWX.cpp	(working copy)
@@ -893,6 +893,7 @@
     SetFocusState(false);
     focusEvent = false;
     DestroySystemCaret();
+    SetMouseCapture(false);
 }
 
 void ScintillaWX::DoGainFocus(){
