Index: src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- src/plugins/compilergcc/compilergcc.cpp	(revision 6015)
+++ src/plugins/compilergcc/compilergcc.cpp	(working copy)
@@ -3440,7 +3440,7 @@
                     AskForActiveProject();
                     project = m_Project;
                 }
-                last_error_file.PrependDir(project->GetExecutionDir());
+                last_error_file = project->GetExecutionDir() + last_error_file.GetFullPath();
                 last_error_file.MakeRelativeTo(project->GetBasePath());
                 last_error_filename = last_error_file.GetFullPath();
             }
