Index: debian/changelog
===================================================================
--- debian/changelog	(revision 6089)
+++ debian/changelog	(working copy)
@@ -1,4 +1,4 @@
-codeblocks (8.02svn5933) unstable; urgency=low
+codeblocks (8.02svn6089) unstable; urgency=low
 
   * more fixes to debian build system, sources and packages are now
     lintian-clean, except for some words guessed as spelling-errors.
Index: src/tools/ConsoleRunner/main.cpp
===================================================================
--- src/tools/ConsoleRunner/main.cpp	(revision 6089)
+++ src/tools/ConsoleRunner/main.cpp	(working copy)
@@ -17,6 +17,9 @@
 #else
 	#define wait_key getchar
 #endif
+#if defined(__unix__) || defined(__unix)
+    #include <sys/wait.h>
+#endif
 #include <string.h>
 
 #ifdef __MINGW32__
