Index: src/i18n/extract_win.bat
===================================================================
--- src/i18n/extract_win.bat	(revision 6130)
+++ src/i18n/extract_win.bat	(working copy)
@@ -64,7 +64,7 @@
 goto TheEnd
 
 :Error
-echo Error: At least one requited tools was not found.
+echo Error: At least one required tool was not found.
 echo Make sure you have the following tools in the PATH:
 echo - grep, xargs, xgettext, msginit
 echo They are all part of the GnuWin32 tools (package gettext).
Index: src/plugins/contrib/wxSmith/wxsmith/wxsBitmapIconEditorDlg.wxs
===================================================================
--- src/plugins/contrib/wxSmith/wxsmith/wxsBitmapIconEditorDlg.wxs	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxsmith/wxsBitmapIconEditorDlg.wxs	(working copy)
@@ -1,187 +1,217 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <wxsmith>
-    <object class="wxScrollingDialog" name="wxsBitmapIconEditorDlg">
-        <title>Image editor</title>
-        <style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER</style>
-        <object class="wxBoxSizer" variable="BoxSizer1" member="yes">
-            <orient>wxVERTICAL</orient>
-            <object class="sizeritem">
-                <object class="wxBoxSizer" variable="BoxSizer5" member="yes">
-                    <object class="sizeritem">
-                        <object class="wxStaticBoxSizer" variable="StaticBoxSizer1" member="yes">
-                            <label>Image options</label>
-                            <object class="sizeritem">
-                                <object class="wxFlexGridSizer" variable="FlexGridSizer1" member="yes">
-                                    <cols>1</cols>
-                                    <growablecols>1</growablecols>
-                                    <object class="sizeritem">
-                                        <object class="wxRadioButton" name="ID_RADIOBUTTON1" variable="NoImage" member="yes">
-                                            <label>No image</label>
-                                            <handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
-                                        </object>
-                                        <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
-                                        <border>5</border>
-                                        <option>1</option>
-                                    </object>
-                                    <object class="spacer">
-                                        <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                        <border>5</border>
-                                        <option>1</option>
-                                        <size>1,1</size>
-                                    </object>
-                                    <object class="sizeritem">
-                                        <object class="wxRadioButton" name="ID_RADIOBUTTON2" variable="ImageFile" member="yes">
-                                            <label>Image From File:</label>
-                                            <handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
-                                        </object>
-                                        <flag>wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
-                                        <border>5</border>
-                                        <option>1</option>
-                                    </object>
-                                    <object class="sizeritem">
-                                        <object class="wxBoxSizer" variable="BoxSizer2" member="yes">
-                                            <object class="spacer">
-                                                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                <border>5</border>
-                                                <size>20,1</size>
-                                            </object>
-                                            <object class="sizeritem">
-                                                <object class="wxTextCtrl" name="ID_TEXTCTRL1" variable="FileName" member="yes">
-                                                    <handler function="OnFileNameText" entry="EVT_TEXT" />
-                                                </object>
-                                                <flag>wxTOP|wxBOTTOM|wxLEFT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                <border>5</border>
-                                                <option>1</option>
-                                            </object>
-                                            <object class="sizeritem">
-                                                <object class="wxButton" name="ID_BUTTON3" variable="Button3" member="yes">
-                                                    <label>...</label>
-                                                    <style>wxBU_EXACTFIT</style>
-                                                    <handler function="OnButton3Click" entry="EVT_BUTTON" />
-                                                </object>
-                                                <flag>wxTOP|wxBOTTOM|wxRIGHT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                <border>5</border>
-                                            </object>
-                                        </object>
-                                        <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                        <border>5</border>
-                                        <option>1</option>
-                                    </object>
-                                    <object class="sizeritem">
-                                        <object class="wxRadioButton" name="ID_RADIOBUTTON3" variable="ImageArt" member="yes">
-                                            <label>Image from wxArtProvider:</label>
-                                            <handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
-                                        </object>
-                                        <flag>wxBOTTOM|wxLEFT|wxRIGHT|wxALIGN_LEFT|wxALIGN_TOP</flag>
-                                        <border>5</border>
-                                        <option>1</option>
-                                    </object>
-                                    <object class="sizeritem">
-                                        <object class="wxBoxSizer" variable="BoxSizer4" member="yes">
-                                            <object class="spacer">
-                                                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                <border>5</border>
-                                                <size>20,1</size>
-                                            </object>
-                                            <object class="sizeritem">
-                                                <object class="wxFlexGridSizer" variable="FlexGridSizer2" member="yes">
-                                                    <cols>2</cols>
-                                                    <object class="sizeritem">
-                                                        <object class="wxStaticText" name="ID_STATICTEXT1" variable="StaticText1" member="yes">
-                                                            <label>Art Id:</label>
-                                                        </object>
-                                                        <flag>wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
-                                                        <border>5</border>
-                                                        <option>1</option>
-                                                    </object>
-                                                    <object class="sizeritem">
-                                                        <object class="wxComboBox" name="ID_COMBOBOX2" variable="ArtId" member="yes">
-                                                            <style>wxCB_SORT</style>
-                                                            <handler function="OnArtIdSelect" entry="EVT_COMBOBOX" />
-                                                        </object>
-                                                        <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                        <border>5</border>
-                                                        <option>1</option>
-                                                    </object>
-                                                    <object class="sizeritem">
-                                                        <object class="wxStaticText" name="ID_STATICTEXT2" variable="StaticText2" member="yes">
-                                                            <label>Art Client:</label>
-                                                        </object>
-                                                        <flag>wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
-                                                        <border>5</border>
-                                                        <option>1</option>
-                                                    </object>
-                                                    <object class="sizeritem">
-                                                        <object class="wxComboBox" name="ID_COMBOBOX1" variable="ArtClient" member="yes">
-                                                            <style>wxCB_SORT</style>
-                                                            <handler function="OnArtIdSelect" entry="EVT_COMBOBOX" />
-                                                        </object>
-                                                        <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                        <border>5</border>
-                                                        <option>1</option>
-                                                    </object>
-                                                </object>
-                                                <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                                <border>5</border>
-                                                <option>1</option>
-                                            </object>
-                                        </object>
-                                        <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                        <border>5</border>
-                                        <option>1</option>
-                                    </object>
-                                </object>
-                                <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                <border>5</border>
-                                <option>1</option>
-                            </object>
-                        </object>
-                        <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                        <border>5</border>
-                    </object>
-                    <object class="sizeritem">
-                        <object class="wxStaticBoxSizer" variable="StaticBoxSizer2" member="yes">
-                            <label>Preview</label>
-                            <object class="sizeritem">
-                                <object class="wxStaticBitmap" name="ID_STATICBITMAP1" variable="Preview" member="yes">
-                                    <size>200,200</size>
-                                </object>
-                                <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                                <border>5</border>
-                                <option>1</option>
-                            </object>
-                        </object>
-                        <flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                        <border>5</border>
-                        <option>1</option>
-                    </object>
-                </object>
-                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                <border>5</border>
-                <option>1</option>
-            </object>
-            <object class="sizeritem">
-                <object class="wxBoxSizer" variable="BoxSizer3" member="yes">
-                    <object class="sizeritem">
-                        <object class="wxButton" name="ID_BUTTON1" variable="Button1" member="yes">
-                            <label>OK</label>
-                            <default>1</default>
-                            <handler function="OnButton1Click" entry="EVT_BUTTON" />
-                        </object>
-                        <flag>wxTOP|wxBOTTOM|wxLEFT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                        <border>5</border>
-                    </object>
-                    <object class="sizeritem">
-                        <object class="wxButton" name="wxID_CANCEL" variable="Button2" member="yes">
-                            <label>Cancel</label>
-                        </object>
-                        <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                        <border>5</border>
-                    </object>
-                </object>
-                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
-                <border>5</border>
-            </object>
-        </object>
-    </object>
+	<object class="wxScrollingDialog" name="wxsBitmapIconEditorDlg">
+		<title>Image editor</title>
+		<id_arg>0</id_arg>
+		<style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER</style>
+		<object class="wxBoxSizer" variable="BoxSizer1" member="yes">
+			<orient>wxVERTICAL</orient>
+			<object class="sizeritem">
+				<object class="wxBoxSizer" variable="BoxSizer5" member="yes">
+					<object class="sizeritem">
+						<object class="wxStaticBoxSizer" variable="StaticBoxSizer1" member="yes">
+							<label>Image options</label>
+							<object class="sizeritem">
+								<object class="wxFlexGridSizer" variable="FlexGridSizer1" member="yes">
+									<cols>1</cols>
+									<growablecols>1</growablecols>
+									<object class="sizeritem">
+										<object class="wxRadioButton" name="ID_RADIOBUTTON1" variable="NoImage" member="yes">
+											<label>No image</label>
+											<handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
+										</object>
+										<flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+									<object class="spacer">
+										<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+										<border>5</border>
+										<option>1</option>
+										<size>1,1</size>
+									</object>
+									<object class="sizeritem">
+										<object class="wxRadioButton" name="ID_RADIOBUTTON2" variable="ImageFile" member="yes">
+											<label>Image From File:</label>
+											<handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
+										</object>
+										<flag>wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+									<object class="sizeritem">
+										<object class="wxBoxSizer" variable="BoxSizer2" member="yes">
+											<object class="spacer">
+												<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+												<border>5</border>
+												<size>20,1</size>
+											</object>
+											<object class="sizeritem">
+												<object class="wxTextCtrl" name="ID_TEXTCTRL1" variable="FileName" member="yes">
+													<handler function="OnFileNameText" entry="EVT_TEXT" />
+												</object>
+												<flag>wxTOP|wxBOTTOM|wxLEFT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+												<border>5</border>
+												<option>1</option>
+											</object>
+											<object class="sizeritem">
+												<object class="wxButton" name="ID_BUTTON3" variable="Button3" member="yes">
+													<label>...</label>
+													<style>wxBU_EXACTFIT</style>
+													<handler function="OnButton3Click" entry="EVT_BUTTON" />
+												</object>
+												<flag>wxTOP|wxBOTTOM|wxRIGHT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+												<border>5</border>
+											</object>
+										</object>
+										<flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+									<object class="sizeritem">
+										<object class="wxRadioButton" name="ID_RADIOBUTTON3" variable="ImageArt" member="yes">
+											<label>Image from wxArtProvider:</label>
+											<handler function="OnUpdatePreview" entry="EVT_RADIOBUTTON" />
+										</object>
+										<flag>wxBOTTOM|wxLEFT|wxRIGHT|wxALIGN_LEFT|wxALIGN_TOP</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+									<object class="sizeritem">
+										<object class="wxBoxSizer" variable="BoxSizer4" member="yes">
+											<object class="spacer">
+												<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+												<border>5</border>
+												<size>20,1</size>
+											</object>
+											<object class="sizeritem">
+												<object class="wxFlexGridSizer" variable="FlexGridSizer2" member="yes">
+													<cols>2</cols>
+													<object class="sizeritem">
+														<object class="wxStaticText" name="ID_STATICTEXT1" variable="StaticText1" member="yes">
+															<label>Art Id:</label>
+														</object>
+														<flag>wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+														<border>5</border>
+														<option>1</option>
+													</object>
+													<object class="sizeritem">
+														<object class="wxComboBox" name="ID_COMBOBOX2" variable="ArtId" member="yes">
+															<style>wxCB_SORT</style>
+															<handler function="OnArtIdSelect" entry="EVT_COMBOBOX" />
+														</object>
+														<flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+														<border>5</border>
+														<option>1</option>
+													</object>
+													<object class="sizeritem">
+														<object class="wxStaticText" name="ID_STATICTEXT2" variable="StaticText2" member="yes">
+															<label>Art Client:</label>
+														</object>
+														<flag>wxTOP|wxLEFT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+														<border>5</border>
+														<option>1</option>
+													</object>
+													<object class="sizeritem">
+														<object class="wxComboBox" name="ID_COMBOBOX1" variable="ArtClient" member="yes">
+															<style>wxCB_SORT</style>
+															<handler function="OnArtIdSelect" entry="EVT_COMBOBOX" />
+														</object>
+														<flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+														<border>5</border>
+														<option>1</option>
+													</object>
+												</object>
+												<flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+												<border>5</border>
+												<option>1</option>
+											</object>
+										</object>
+										<flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+									<object class="sizeritem">
+										<object class="wxRadioButton" name="ID_RADIOBUTTON4" variable="ImageCode" member="yes">
+											<label>Code</label>
+										</object>
+										<flag>wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+									<object class="sizeritem">
+										<object class="wxBoxSizer" variable="BoxSizer7" member="no">
+											<object class="spacer">
+												<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+												<border>5</border>
+												<option>1</option>
+												<size>2,1</size>
+											</object>
+											<object class="sizeritem">
+												<object class="wxTextCtrl" name="ID_TEXTCTRL2" variable="CodeText" member="yes">
+													<handler function="OnCodeTextText" entry="EVT_TEXT" />
+												</object>
+												<flag>wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM</flag>
+												<border>5</border>
+												<option>8</option>
+											</object>
+										</object>
+										<flag>wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM</flag>
+										<border>5</border>
+										<option>1</option>
+									</object>
+								</object>
+								<flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+								<border>5</border>
+								<option>1</option>
+							</object>
+						</object>
+						<flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+						<border>5</border>
+					</object>
+					<object class="sizeritem">
+						<object class="wxStaticBoxSizer" variable="StaticBoxSizer2" member="yes">
+							<label>Preview</label>
+							<object class="sizeritem">
+								<object class="wxStaticBitmap" name="ID_STATICBITMAP1" variable="Preview" member="yes">
+									<size>200,200</size>
+								</object>
+								<flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+								<border>5</border>
+								<option>1</option>
+							</object>
+						</object>
+						<flag>wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+						<border>5</border>
+						<option>1</option>
+					</object>
+				</object>
+				<flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+				<border>5</border>
+				<option>1</option>
+			</object>
+			<object class="sizeritem">
+				<object class="wxBoxSizer" variable="BoxSizer3" member="yes">
+					<object class="sizeritem">
+						<object class="wxButton" name="ID_BUTTON1" variable="Button1" member="yes">
+							<label>OK</label>
+							<default>1</default>
+							<handler function="OnButton1Click" entry="EVT_BUTTON" />
+						</object>
+						<flag>wxTOP|wxBOTTOM|wxLEFT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+						<border>5</border>
+					</object>
+					<object class="sizeritem">
+						<object class="wxButton" name="wxID_CANCEL" variable="Button2" member="yes">
+							<label>Cancel</label>
+						</object>
+						<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+						<border>5</border>
+					</object>
+				</object>
+				<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+				<border>5</border>
+			</object>
+		</object>
+	</object>
 </wxsmith>
Index: src/plugins/contrib/wxSmith/wxsmith/wxsDeleteItemRes.wxs
===================================================================
--- src/plugins/contrib/wxSmith/wxsmith/wxsDeleteItemRes.wxs	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxsmith/wxsDeleteItemRes.wxs	(working copy)
@@ -28,7 +28,7 @@
 					</object>
 					<object class="sizeritem">
 						<object class="wxCheckBox" name="ID_CHECKBOX3" variable="m_PhisDeleteSources" member="yes">
-							<label>Phisically delete source files</label>
+							<label>Physically delete source files</label>
 							<enabled>0</enabled>
 						</object>
 						<flag>wxBOTTOM|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.cpp	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.cpp	(working copy)
@@ -29,12 +29,8 @@
 //(*InternalHeaders(wxsToolBarEditor)
 #include <wx/artprov.h>
 #include <wx/bitmap.h>
-#include <wx/font.h>
-#include <wx/fontenum.h>
-#include <wx/fontmap.h>
-#include <wx/image.h>
 #include <wx/intl.h>
-#include <wx/settings.h>
+#include <wx/image.h>
 #include <wx/string.h>
 //*)
 
@@ -77,7 +73,7 @@
 {
     wxWindowID id = wxID_ANY;
 	//(*Initialize(wxsToolBarEditor)
-	Create(parent, id, wxDefaultPosition, wxDefaultSize, wxTAB_TRAVERSAL, _T("wxPanel"));
+	Create(parent, id, wxDefaultPosition, wxDefaultSize, wxTAB_TRAVERSAL, _T("id"));
 	BoxSizer1 = new wxBoxSizer(wxHORIZONTAL);
 	StaticBoxSizer1 = new wxStaticBoxSizer(wxHORIZONTAL, this, _("Content"));
 	m_Content = new wxListBox(this, ID_LISTBOX1, wxDefaultPosition, wxDefaultSize, 0, 0, 0, wxDefaultValidator, _T("ID_LISTBOX1"));
@@ -112,12 +108,12 @@
 	FlexGridSizer1->Add(m_Label, 1, wxTOP|wxLEFT|wxRIGHT|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 4);
 	StaticText4 = new wxStaticText(this, ID_STATICTEXT4, _("Bitmap:"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT4"));
 	FlexGridSizer1->Add(StaticText4, 1, wxTOP|wxLEFT|wxRIGHT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 4);
-	m_Bitmap = new wxBitmapButton(this, ID_BITMAPBUTTON1, wxArtProvider::GetBitmap(wxART_MAKE_ART_ID_FROM_STR(_("wxART_FILE_OPEN")),_T("wxART_TOOLBAR")), wxDefaultPosition, wxDefaultSize, wxBU_AUTODRAW, wxDefaultValidator, _T("ID_BITMAPBUTTON1"));
+	m_Bitmap = new wxBitmapButton(this, ID_BITMAPBUTTON1, wxArtProvider::GetBitmap(wxART_MAKE_ART_ID_FROM_STR(_T("wxART_FILE_OPEN")),wxART_TOOLBAR), wxDefaultPosition, wxDefaultSize, wxBU_AUTODRAW, wxDefaultValidator, _T("ID_BITMAPBUTTON1"));
 	m_Bitmap->SetDefault();
 	FlexGridSizer1->Add(m_Bitmap, 1, wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 4);
 	StaticText5 = new wxStaticText(this, ID_STATICTEXT5, _("Disabled bitmap:"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT5"));
 	FlexGridSizer1->Add(StaticText5, 1, wxTOP|wxLEFT|wxRIGHT|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 4);
-	m_Bitmap2 = new wxBitmapButton(this, ID_BITMAPBUTTON2, wxArtProvider::GetBitmap(wxART_MAKE_ART_ID_FROM_STR(_("wxART_FILE_OPEN")),_T("wxART_BUTTON_C")), wxDefaultPosition, wxDefaultSize, wxBU_AUTODRAW, wxDefaultValidator, _T("ID_BITMAPBUTTON2"));
+	m_Bitmap2 = new wxBitmapButton(this, ID_BITMAPBUTTON2, wxArtProvider::GetBitmap(wxART_MAKE_ART_ID_FROM_STR(_T("wxART_FILE_OPEN")),wxART_BUTTON), wxDefaultPosition, wxDefaultSize, wxBU_AUTODRAW, wxDefaultValidator, _T("ID_BITMAPBUTTON2"));
 	m_Bitmap2->SetDefault();
 	FlexGridSizer1->Add(m_Bitmap2, 1, wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 4);
 	StaticText2 = new wxStaticText(this, ID_STATICTEXT2, _("Tooltip:"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT2"));
@@ -147,6 +143,7 @@
 	SetSizer(BoxSizer1);
 	BoxSizer1->Fit(this);
 	BoxSizer1->SetSizeHints(this);
+
 	Connect(ID_LISTBOX1,wxEVT_COMMAND_LISTBOX_SELECTED,(wxObjectEventFunction)&wxsToolBarEditor::Onm_ContentSelect);
 	Connect(ID_RADIOBUTTON1,wxEVT_COMMAND_RADIOBUTTON_SELECTED,(wxObjectEventFunction)&wxsToolBarEditor::OnTypeChanged);
 	Connect(ID_RADIOBUTTON2,wxEVT_COMMAND_RADIOBUTTON_SELECTED,(wxObjectEventFunction)&wxsToolBarEditor::OnTypeChanged);
@@ -436,7 +433,8 @@
             m_Label->Enable(!IsSeparator);
             m_Label->SetValue(m_Selected->m_Label);
             m_Bitmap->Enable(!IsSeparator);
-            if ( m_Selected->m_Bitmap.IsEmpty() )
+			// A bitmap created from code can't be rendered easily here, so show the default for that case, too
+            if ( m_Selected->m_Bitmap.IsEmpty() || !m_Selected->m_Bitmap.CodeText.IsEmpty())
             {
                 m_Bitmap->SetBitmapLabel(wxArtProvider::GetBitmap(wxART_MAKE_ART_ID_FROM_STR(_T("wxART_FILE_OPEN")),_T("wxART_TOOLBAR")));
             }
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.h	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.h	(working copy)
@@ -28,15 +28,15 @@
 #include "wxstoolbar.h"
 
 //(*Headers(wxsToolBarEditor)
-#include <wx/bmpbuttn.h>
-#include <wx/button.h>
-#include <wx/listbox.h>
-#include <wx/panel.h>
-#include <wx/radiobut.h>
 #include <wx/sizer.h>
-#include <wx/statline.h>
 #include <wx/stattext.h>
 #include <wx/textctrl.h>
+#include <wx/listbox.h>
+#include <wx/statline.h>
+#include <wx/radiobut.h>
+#include <wx/panel.h>
+#include <wx/bmpbuttn.h>
+#include <wx/button.h>
 //*)
 
 class wxsToolBarEditor: public wxPanel
@@ -89,36 +89,36 @@
 		//*)
 
 		//(*Declarations(wxsToolBarEditor)
-		wxBoxSizer* BoxSizer1;
-		wxStaticBoxSizer* StaticBoxSizer1;
-		wxListBox* m_Content;
 		wxStaticBoxSizer* StaticBoxSizer2;
-		wxGridSizer* GridSizer1;
+		wxBitmapButton* m_Bitmap2;
+		wxButton* Button4;
+		wxStaticText* StaticText2;
+		wxTextCtrl* m_HelpText;
 		wxRadioButton* m_TypeNormal;
-		wxRadioButton* m_TypeSeparator;
+		wxButton* Button1;
+		wxStaticText* StaticText6;
 		wxRadioButton* m_TypeRadio;
-		wxRadioButton* m_TypeCheck;
-		wxStaticLine* StaticLine1;
-		wxFlexGridSizer* FlexGridSizer1;
-		wxStaticText* StaticText6;
-		wxTextCtrl* m_Id;
 		wxStaticText* StaticText1;
-		wxTextCtrl* m_Label;
-		wxStaticText* StaticText4;
+		wxBoxSizer* BoxSizer2;
+		wxStaticText* StaticText3;
+		wxButton* Button2;
 		wxBitmapButton* m_Bitmap;
+		wxStaticLine* StaticLine2;
+		wxRadioButton* m_TypeCheck;
+		wxRadioButton* m_TypeSeparator;
+		wxButton* Button3;
+		wxGridSizer* GridSizer1;
 		wxStaticText* StaticText5;
-		wxBitmapButton* m_Bitmap2;
-		wxStaticText* StaticText2;
+		wxStaticLine* StaticLine1;
+		wxBoxSizer* BoxSizer1;
 		wxTextCtrl* m_ToolTip;
-		wxStaticText* StaticText3;
-		wxTextCtrl* m_HelpText;
-		wxStaticLine* StaticLine2;
+		wxStaticBoxSizer* StaticBoxSizer1;
+		wxFlexGridSizer* FlexGridSizer1;
 		wxBoxSizer* BoxSizer3;
-		wxButton* Button3;
-		wxButton* Button4;
-		wxBoxSizer* BoxSizer2;
-		wxButton* Button1;
-		wxButton* Button2;
+		wxStaticText* StaticText4;
+		wxListBox* m_Content;
+		wxTextCtrl* m_Label;
+		wxTextCtrl* m_Id;
 		//*)
 
         enum ItemType
Index: src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.cpp	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.cpp	(working copy)
@@ -109,6 +109,8 @@
 const long wxsBitmapIconEditorDlg::ID_COMBOBOX2 = wxNewId();
 const long wxsBitmapIconEditorDlg::ID_STATICTEXT2 = wxNewId();
 const long wxsBitmapIconEditorDlg::ID_COMBOBOX1 = wxNewId();
+const long wxsBitmapIconEditorDlg::ID_RADIOBUTTON4 = wxNewId();
+const long wxsBitmapIconEditorDlg::ID_TEXTCTRL2 = wxNewId();
 const long wxsBitmapIconEditorDlg::ID_STATICBITMAP1 = wxNewId();
 const long wxsBitmapIconEditorDlg::ID_BUTTON1 = wxNewId();
 //*)
@@ -125,7 +127,9 @@
     Data(_Data)
 {
 	//(*Initialize(wxsBitmapIconEditorDlg)
-	Create(parent, id, _("Image editor"), wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER, _T("id"));
+	wxBoxSizer* BoxSizer7;
+
+	Create(parent, wxID_ANY, _("Image editor"), wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER, _T("wxID_ANY"));
 	BoxSizer1 = new wxBoxSizer(wxVERTICAL);
 	BoxSizer5 = new wxBoxSizer(wxHORIZONTAL);
 	StaticBoxSizer1 = new wxStaticBoxSizer(wxHORIZONTAL, this, _("Image options"));
@@ -158,6 +162,13 @@
 	FlexGridSizer2->Add(ArtClient, 1, wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	BoxSizer4->Add(FlexGridSizer2, 1, wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	FlexGridSizer1->Add(BoxSizer4, 1, wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	ImageCode = new wxRadioButton(this, ID_RADIOBUTTON4, _("Code"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_RADIOBUTTON4"));
+	FlexGridSizer1->Add(ImageCode, 1, wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 5);
+	BoxSizer7 = new wxBoxSizer(wxHORIZONTAL);
+	BoxSizer7->Add(2,1,1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	CodeText = new wxTextCtrl(this, ID_TEXTCTRL2, wxEmptyString, wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_TEXTCTRL2"));
+	BoxSizer7->Add(CodeText, 8, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 5);
+	FlexGridSizer1->Add(BoxSizer7, 1, wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 5);
 	StaticBoxSizer1->Add(FlexGridSizer1, 1, wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	BoxSizer5->Add(StaticBoxSizer1, 0, wxALL|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	StaticBoxSizer2 = new wxStaticBoxSizer(wxHORIZONTAL, this, _("Preview"));
@@ -183,6 +194,7 @@
 	Connect(ID_RADIOBUTTON3,wxEVT_COMMAND_RADIOBUTTON_SELECTED,(wxObjectEventFunction)&wxsBitmapIconEditorDlg::OnUpdatePreview);
 	Connect(ID_COMBOBOX2,wxEVT_COMMAND_COMBOBOX_SELECTED,(wxObjectEventFunction)&wxsBitmapIconEditorDlg::OnArtIdSelect);
 	Connect(ID_COMBOBOX1,wxEVT_COMMAND_COMBOBOX_SELECTED,(wxObjectEventFunction)&wxsBitmapIconEditorDlg::OnArtIdSelect);
+	Connect(ID_TEXTCTRL2,wxEVT_COMMAND_TEXT_UPDATED,(wxObjectEventFunction)&wxsBitmapIconEditorDlg::OnCodeTextText);
 	Connect(ID_BUTTON1,wxEVT_COMMAND_BUTTON_CLICKED,(wxObjectEventFunction)&wxsBitmapIconEditorDlg::OnButton1Click);
 	//*)
 
@@ -257,13 +269,22 @@
     {
         Data.Id.Clear();
         Data.Client.Clear();
+        Data.CodeText.Clear();
         Data.FileName = FileName->GetValue();
     }
+    else if ( ImageCode->GetValue() )
+    {
+        Data.Id.Clear();
+        Data.Client.Clear();
+        Data.FileName.Clear();
+        Data.CodeText = CodeText->GetValue();
+    }
     else
     {
         Data.Id = ArtId->GetValue();
         Data.Client = Data.Id.empty() ? _T("") : ArtClient->GetValue();
         Data.FileName.Clear();
+        Data.CodeText.Clear();
     }
 }
 
@@ -271,10 +292,15 @@
 {
     if ( Data.Id.empty() )
     {
-        if ( Data.FileName.empty() )
+        if ( Data.FileName.empty() && Data.CodeText.empty() )
         {
             NoImage->SetValue(true);
         }
+        else if( !Data.CodeText.empty() )
+        {
+            ImageCode->SetValue(true);
+            CodeText->SetValue(Data.CodeText);
+        }
         else
         {
             ImageFile->SetValue(true);
@@ -320,3 +346,8 @@
 {
     UpdatePreview();
 }
+
+void wxsBitmapIconEditorDlg::OnCodeTextText(wxCommandEvent& event)
+{
+    UpdatePreview();
+}
Index: src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.h	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconeditordlg.h	(working copy)
@@ -30,6 +30,7 @@
 #include <wx/radiobut.h>
 #include <wx/statbmp.h>
 #include <wx/button.h>
+#include "scrollingdialog.h"
 #include <wx/combobox.h>
 #include "scrollingdialog.h"
 //*)
@@ -54,6 +55,8 @@
 		static const long ID_COMBOBOX2;
 		static const long ID_STATICTEXT2;
 		static const long ID_COMBOBOX1;
+		static const long ID_RADIOBUTTON4;
+		static const long ID_TEXTCTRL2;
 		static const long ID_STATICBITMAP1;
 		static const long ID_BUTTON1;
 		//*)
@@ -67,6 +70,8 @@
 		void OnButton3Click(wxCommandEvent& event);
 		void OnArtIdSelect(wxCommandEvent& event);
 		void OnUpdatePreview(wxCommandEvent& event);
+		void OnOtherTextText(wxCommandEvent& event);
+		void OnCodeTextText(wxCommandEvent& event);
 		//*)
 
 		//(*Declarations(wxsBitmapIconEditorDlg)
@@ -74,8 +79,10 @@
 		wxStaticBoxSizer* StaticBoxSizer2;
 		wxStaticBitmap* Preview;
 		wxRadioButton* ImageArt;
+		wxRadioButton* ImageCode;
 		wxBoxSizer* BoxSizer5;
 		wxStaticText* StaticText2;
+		wxTextCtrl* CodeText;
 		wxButton* Button1;
 		wxTextCtrl* FileName;
 		wxFlexGridSizer* FlexGridSizer2;
Index: src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconproperty.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconproperty.cpp	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconproperty.cpp	(working copy)
@@ -64,17 +64,28 @@
             wxString Code;
             if ( Id.empty() )
             {
-                if ( FileName.empty() ) return wxEmptyString;
-
-                if ( NoResize )
+                if ( FileName.empty() )
                 {
-                    Code << _T("wxBitmap(wxImage(") << wxsCodeMarks::WxString(wxsCPP,FileName,false) << _T("))");
+					if ( CodeText.empty() )
+					{
+						return wxEmptyString;
+					}
+					else
+					{
+						Code << CodeText;
+					}
                 }
-                else
-                {
-                    Code << _T("wxBitmap(wxImage(") << wxsCodeMarks::WxString(wxsCPP,FileName,false) << _T(").Rescale(")
-                         << SizeCode << _T(".GetWidth(),") << SizeCode << _T(".GetHeight()))");
-                }
+				else{
+					if ( NoResize )
+					{
+						Code << _T("wxBitmap(wxImage(") << wxsCodeMarks::WxString(wxsCPP,FileName,false) << _T("))");
+					}
+					else
+					{
+						Code << _T("wxBitmap(wxImage(") << wxsCodeMarks::WxString(wxsCPP,FileName,false) << _T(").Rescale(")
+							 << SizeCode << _T(".GetWidth(),") << SizeCode << _T(".GetHeight()))");
+					}
+				}
             }
             else
             {
@@ -126,7 +137,7 @@
 
 bool wxsBitmapIconData::IsEmpty()
 {
-    return Id.IsEmpty() && FileName.IsEmpty();
+    return Id.IsEmpty() && FileName.IsEmpty() && CodeText.IsEmpty();
 }
 
 
@@ -169,10 +180,17 @@
 
     if ( VALUE.Id.IsEmpty() )
     {
-        // No wxART_PROVIDER Id, do it must be filename
+        // No wxART_PROVIDER Id, so it must be a filename or code
         VALUE.Id.Clear();
         VALUE.Client.Clear();
-        return XmlGetString(Element,VALUE.FileName);
+
+		VALUE.CodeText = cbC2U(Element->Attribute("code"));
+		if ( VALUE.CodeText.IsEmpty() )
+		{
+			// It's a filename
+			VALUE.CodeText.Clear();
+			return XmlGetString(Element,VALUE.FileName);
+		}
     }
 
     VALUE.FileName.Clear();
@@ -197,6 +215,12 @@
         return true;
     }
 
+    if ( !VALUE.CodeText.empty() )
+    {
+        Element->SetAttribute("code",cbU2C(VALUE.CodeText));
+        return true;
+    }
+
     return false;
 }
 
Index: src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconproperty.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconproperty.h	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/wxsbitmapiconproperty.h	(working copy)
@@ -39,6 +39,7 @@
     wxString Id;
     wxString Client;
     wxString FileName;
+    wxString CodeText;
 
     /** \brief Generating preview for this bitmap / icon */
     wxBitmap GetPreview(const wxSize& Size,const wxString& DefaultClient = wxART_OTHER);
Index: src/plugins/contrib/wxSmith/wxwidgets/wxsdeleteitemres.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/wxsdeleteitemres.cpp	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/wxsdeleteitemres.cpp	(working copy)
@@ -23,9 +23,9 @@
 #include "wxsdeleteitemres.h"
 
 //(*InternalHeaders(wxsDeleteItemRes)
+#include <wx/intl.h>
 #include <wx/button.h>
 #include <wx/string.h>
-#include <wx/intl.h>
 //*)
 
 //(*IdInit(wxsDeleteItemRes)
@@ -46,16 +46,16 @@
 	Create(0, wxID_ANY, _("Deleting resource"), wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE, _T("wxID_ANY"));
 	BoxSizer1 = new wxBoxSizer(wxVERTICAL);
 	StaticBoxSizer1 = new wxStaticBoxSizer(wxVERTICAL, this, _("Delete options"));
-	m_PhisDeleteWXS = new wxCheckBox(this, ID_CHECKBOX1, _("Physically delete WXS file"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_CHECKBOX1"));
-	m_PhisDeleteWXS->SetValue(true);
-	StaticBoxSizer1->Add(m_PhisDeleteWXS, 0, wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	m_PhysDeleteWXS = new wxCheckBox(this, ID_CHECKBOX1, _("Physically delete WXS file"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_CHECKBOX1"));
+	m_PhysDeleteWXS->SetValue(true);
+	StaticBoxSizer1->Add(m_PhysDeleteWXS, 0, wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	m_DeleteSources = new wxCheckBox(this, ID_CHECKBOX2, _("Delete source files from project"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_CHECKBOX2"));
 	m_DeleteSources->SetValue(false);
 	StaticBoxSizer1->Add(m_DeleteSources, 0, wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
-	m_PhisDeleteSources = new wxCheckBox(this, ID_CHECKBOX3, _("Phisically delete source files"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_CHECKBOX3"));
-	m_PhisDeleteSources->SetValue(false);
-	m_PhisDeleteSources->Disable();
-	StaticBoxSizer1->Add(m_PhisDeleteSources, 0, wxBOTTOM|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	m_PhysDeleteSources = new wxCheckBox(this, ID_CHECKBOX3, _("Physically delete source files"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_CHECKBOX3"));
+	m_PhysDeleteSources->SetValue(false);
+	m_PhysDeleteSources->Disable();
+	StaticBoxSizer1->Add(m_PhysDeleteSources, 0, wxBOTTOM|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	StaticText1 = new wxStaticText(this, ID_STATICTEXT1, _("Note that there\'s no undo\nafter deleting resource"), wxDefaultPosition, wxDefaultSize, wxALIGN_CENTRE, _T("ID_STATICTEXT1"));
 	StaticBoxSizer1->Add(StaticText1, 1, wxBOTTOM|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
 	BoxSizer1->Add(StaticBoxSizer1, 1, wxTOP|wxLEFT|wxRIGHT|wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
@@ -67,7 +67,7 @@
 	SetSizer(BoxSizer1);
 	BoxSizer1->Fit(this);
 	BoxSizer1->SetSizeHints(this);
-	
+
 	Connect(ID_CHECKBOX2,wxEVT_COMMAND_CHECKBOX_CLICKED,(wxObjectEventFunction)&wxsDeleteItemRes::Onm_DeleteSourcesClick);
 	//*)
 }
@@ -81,5 +81,5 @@
 
 void wxsDeleteItemRes::Onm_DeleteSourcesClick(wxCommandEvent& event)
 {
-    m_PhisDeleteSources->Enable(m_DeleteSources->GetValue());
+    m_PhysDeleteSources->Enable(m_DeleteSources->GetValue());
 }
Index: src/plugins/contrib/wxSmith/wxwidgets/wxsdeleteitemres.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/wxsdeleteitemres.h	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/wxsdeleteitemres.h	(working copy)
@@ -24,10 +24,10 @@
 #define WXSDELETEITEMRES_H
 
 //(*Headers(wxsDeleteItemRes)
+#include <wx/sizer.h>
+#include <wx/stattext.h>
 #include <wx/checkbox.h>
-#include <wx/sizer.h>
 #include "scrollingdialog.h"
-#include <wx/stattext.h>
 //*)
 
 class wxsDeleteItemRes: public wxScrollingDialog
@@ -49,13 +49,13 @@
 		//*)
 
 		//(*Declarations(wxsDeleteItemRes)
-		wxCheckBox* m_PhisDeleteWXS;
 		wxStaticText* StaticText1;
+		wxCheckBox* m_PhysDeleteSources;
 		wxBoxSizer* BoxSizer1;
-		wxCheckBox* m_PhisDeleteSources;
+		wxCheckBox* m_DeleteSources;
+		wxStaticBoxSizer* StaticBoxSizer1;
+		wxCheckBox* m_PhysDeleteWXS;
 		wxStdDialogButtonSizer* StdDialogButtonSizer1;
-		wxStaticBoxSizer* StaticBoxSizer1;
-		wxCheckBox* m_DeleteSources;
 		//*)
 
 	private:
Index: src/plugins/contrib/wxSmith/wxwidgets/wxsitemres.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/wxsitemres.cpp	(revision 6130)
+++ src/plugins/contrib/wxSmith/wxwidgets/wxsitemres.cpp	(working copy)
@@ -516,9 +516,9 @@
 
 bool wxsItemRes::OnDeleteCleanup(bool ShowDialog)
 {
-    bool PhisDeleteWXS = true;
+    bool PhysDeleteWXS = true;
     bool DeleteSources = false;
-    bool PhisDeleteSources = false;
+    bool PhysDeleteSources = false;
 
     if ( ShowDialog )
     {
@@ -527,9 +527,9 @@
         {
             return false;
         }
-        PhisDeleteWXS = Dlg.m_PhisDeleteWXS->GetValue();
+        PhysDeleteWXS = Dlg.m_PhysDeleteWXS->GetValue();
         DeleteSources = Dlg.m_DeleteSources->GetValue();
-        PhisDeleteSources = Dlg.m_PhisDeleteSources->GetValue();
+        PhysDeleteSources = Dlg.m_PhysDeleteSources->GetValue();
     }
 
     ProjectFile* Wxs = GetProject()->GetCBProject()->GetFileByFilename(m_WxsFileName,true);
@@ -538,7 +538,7 @@
         GetProject()->GetCBProject()->RemoveFile(Wxs);
     }
 
-    if ( PhisDeleteWXS )
+    if ( PhysDeleteWXS )
     {
         wxRemoveFile(GetProjectPath() + m_WxsFileName);
     }
@@ -556,7 +556,7 @@
             GetProject()->GetCBProject()->RemoveFile(Pf);
         }
 
-        if ( PhisDeleteSources )
+        if ( PhysDeleteSources )
         {
             wxRemoveFile(GetProjectPath() + m_SrcFileName);
             wxRemoveFile(GetProjectPath() + m_HdrFileName);
