Index: src/plugins/contrib/HexEditor/HexEditor.cpp
===================================================================
--- src/plugins/contrib/HexEditor/HexEditor.cpp	(revision 6195)
+++ src/plugins/contrib/HexEditor/HexEditor.cpp	(working copy)
@@ -137,14 +137,15 @@
         wxString label = item->GetLabel();
         #endif
         label.Replace( _T("_"), _T("") );
-        if ( label.Contains( _("&Open...")) )
+        // Cryogen 23/3/10 Fixed menu positioning. Label text shouldn't contain an ampersand. Adjusted help strings.
+        if ( label.Contains( _("Open...")) )
         {
-            fileMenu->Insert( pos+1, idOpenWithHE, _("Open with HexEditor"), _("Open file using HexEditor") );
+            fileMenu->Insert( pos+1, idOpenWithHE, _("Open with he&x editor"), _("Open file using hex editor") );
             return;
         }
     }
 
-    fileMenu->Append( idOpenWithHE, _("Open with HexEditor"), _("Open file using HexEditor") );
+    fileMenu->Append( idOpenWithHE, _("Open with he&x editor"), _("Open a file using hex editor") );
 }
 
 void HexEditor::OnOpenHexEditFileBrowser( wxCommandEvent& event )
