Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 7020)
+++ src/src/main.cpp	(working copy)
@@ -1711,6 +1711,8 @@
 bool MainFrame::DoOpenFile(const wxString& filename, bool addToHistory)
 {
     cbEditor* ed = Manager::Get()->GetEditorManager()->Open(filename);
+    // Cryogen 24/3/10 Activate the editor after opening. Partial fix for bug #14087.
+    ed->Activate();
     if (ed)
     {
         if (addToHistory)
