Index: src/plugins/debuggergdb/cdb_driver.cpp
===================================================================
--- src/plugins/debuggergdb/cdb_driver.cpp	(revision 6196)
+++ src/plugins/debuggergdb/cdb_driver.cpp	(working copy)
@@ -147,14 +147,15 @@
 void CDB_driver::StepIn()
 {
     ResetCursor();
-//    QueueCommand(new DebuggerCmd(this, _T("step")));
-    NOT_IMPLEMENTED();
+    QueueCommand(new DebuggerCmd(this, _T("t")));
+    Step();
 }
 
 void CDB_driver::StepOut()
 {
     ResetCursor();
-    NOT_IMPLEMENTED();
+    QueueCommand(new DebuggerCmd(this, _T("gu")));
+    Step();
 }
 
 void CDB_driver::Backtrace()
