Index: scriptedwizard/resources/plugins/wizard.script
===================================================================
--- scriptedwizard/resources/plugins/wizard.script	(revision 6202)
+++ scriptedwizard/resources/plugins/wizard.script	(working copy)
@@ -226,7 +226,7 @@
         target.AddLinkerOption(_T("-mthreads"));
 
         target.AddIncludeDir(_T("$(#cb)\\include"));
-        target.AddIncludeDir(_T("$(#cb)\\include\\wxscintilla\\include"));
+        target.AddIncludeDir(_T("$(#cb)\\sdk\\wxscintilla\\include"));
         target.AddIncludeDir(_T("$(#wx.include)"));
         target.AddIncludeDir(_T("$(#wx.lib)\\gcc_dll\\mswu"));
 
