Index: main.cpp
===================================================================
--- main.cpp	(revision 6281)
+++ main.cpp	(working copy)
@@ -776,7 +776,7 @@
 
 void MainFrame::PluginsUpdated(cbPlugin* plugin, int status)
 {
-    Freeze();
+    //Freeze();
 
     // menu
     RecreateMenuBar();
@@ -810,12 +810,12 @@
         }
     }
 
-    Thaw();
+    //Thaw();
 }
 
 void MainFrame::RecreateMenuBar()
 {
-    Freeze();
+    //Freeze();
     wxMenuBar* m = GetMenuBar();
     SetMenuBar(0); // unhook old menubar
     CreateMenubar(); // create new menubar
@@ -829,7 +829,7 @@
         SaveViewLayout(it->first, it->second, it->first == m_LastLayoutName);
     }
 
-    Thaw();
+    //Thaw();
 }
 
 void MainFrame::CreateMenubar()
@@ -2412,13 +2412,13 @@
     else
     {
         wxBusyCursor useless;
-        wxPaintEvent e;
-        ProcessEvent(e);
+        //wxPaintEvent e;
+        //ProcessEvent(e);
 
-        Freeze();
+        //Freeze();
         for (unsigned int i = 0; i < files.GetCount(); ++i)
           success &= OpenGeneric(files[i]);
-        Thaw();
+        //Thaw();
     }
     return success;
 }
