--- src/plugins/contrib/HexEditor/HexEditPanel.cpp	(revision 6374)
+++ src/plugins/contrib/HexEditor/HexEditPanel.cpp	(working copy)
@@ -771,7 +771,7 @@
     }
     else
     {
-        m_Status->SetLabel( F( _("Position: %llX / %llX ( %lld%% )"), m_Current, size-1, ( m_Current * 100 / (size-1) ) ) );
+        m_Status->SetLabel( F( _("Position: %llX / %llX ( %lld%% )"), m_Current, size-1, ( (size-1) == 0 ? 100 : m_Current * 100 / (size-1) ) ) );
     }
 
     FileContentBase::OffsetT left = size - m_Current;
