Index: sc_wxtypes.cpp
===================================================================
--- sc_wxtypes.cpp	(Revision 6378)
+++ sc_wxtypes.cpp	(Arbeitskopie)
@@ -236,6 +236,7 @@
 #else
         typedef bool(wxFileName::*WXFN_SETCWD)();
 #endif
+	typedef bool(wxFileName::*WXFN_ISFILEWRITEABLE)()const;
 
         SqPlus::SQClassDef<wxFileName>("wxFileName").
                 emptyCtor().
@@ -279,6 +280,7 @@
                 func(&wxFileName::SetFullName, "SetFullName").
                 func(&wxFileName::SetName, "SetName").
-                func(&wxFileName::SetVolume, "SetVolume");
+                func(&wxFileName::SetVolume, "SetVolume").
+                func<WXFN_ISFILEWRITEABLE>(&wxFileName::IsFileWritable,"IsFileWritable");
 
         /////////////
         // wxPoint //
