Index: src/sdk/filefilters.cpp
===================================================================
--- src/sdk/filefilters.cpp	(revision 6522)
+++ src/sdk/filefilters.cpp	(working copy)
@@ -26,6 +26,7 @@
         return;
     Add(_("Code::Blocks workspace files"),         _T("*.workspace"));
     Add(_("Code::Blocks project files"),           _T("*.cbp"));
+    Add(_("Code::Blocks project/workspace files"), _T("*.workspace;*.cbp"));
     Add(_("Bloodshed Dev-C++ project files"),      _T("*.dev"));
     Add(_("MS Visual C++ 6.0 project files"),      _T("*.dsp"));
     Add(_("MS Visual Studio 7.0+ project files"),  _T("*.vcproj"));
Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 6522)
+++ src/src/main.cpp	(working copy)
@@ -2522,7 +2522,7 @@
         }
         else
         {
-            FileFilters::GetFilterIndexFromName(Filters, _("Code::Blocks project files"), StoredIndex);
+            FileFilters::GetFilterIndexFromName(Filters, _("Code::Blocks project/workspace files"), StoredIndex);
         }
     }
     wxFileDialog dlg(this,
