Index: classwizarddlg.cpp
===================================================================
--- classwizarddlg.cpp	(revision 6604)
+++ classwizarddlg.cpp	(working copy)
@@ -707,7 +707,7 @@
     wxFileName parentname(filename);
     parentname.RemoveLastDir();
 
-    if (filename != parentname)
+    if (filename != parentname&&parentname.GetDirCount()>=1)
         DoForceDirectory(parentname);
 
     if (!wxDirExists(filename.GetPath()))
