Index: src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- src/plugins/compilergcc/compilergcc.cpp	(revision 7282)
+++ src/plugins/compilergcc/compilergcc.cpp	(working copy)
@@ -2962,11 +2964,11 @@
     }
     else
     {
-        Compiler* compiler = CompilerFactory::GetCompiler(m_CompilerId);
+        Compiler* compiler = CompilerFactory::GetCompiler(bt->GetCompilerID());
         if (compiler)
             compiler->Init(m_Project);
 
-        DirectCommands dc(this, CompilerFactory::GetCompiler(bt->GetCompilerID()), m_Project, m_PageIndex);
+        DirectCommands dc(this, compiler, m_Project, m_PageIndex);
         wxArrayString compile = dc.CompileFile(bt, pf);
         AddToCommandQueue(compile);
     }
