--- codeblocks-trunk/src/plugins/contrib/CppCheck/CppCheck.cpp.orig	2011-09-10 08:54:59.496778372 +0200
+++ codeblocks-trunk/src/plugins/contrib/CppCheck/CppCheck.cpp	2011-09-10 09:00:13.384775381 +0200
@@ -249,7 +249,7 @@
         }
     }
 
-    wxString CommandLine = m_CppCheckApp + _T(" --verbose --all --style --xml --file-list=") + InputFileName;
+    wxString CommandLine = m_CppCheckApp + _T(" --verbose --force --all --style --xml --file-list=") + InputFileName;
     if(!IncludeList.IsEmpty())
     {
 		CommandLine += _T(" ") + IncludeList.Trim();
