Index: AutoVersioning.cpp
===================================================================
--- AutoVersioning.cpp	(revision 7386)
+++ AutoVersioning.cpp	(working copy)
@@ -809,8 +809,8 @@
 
     if (!changes.IsEmpty())
     {
-        changes.Prepend(_T("        -"));
-        changes.Replace(_T("\n"), _T("\n        -"));
+        changes.Prepend(_T("-"));
+        changes.Replace(_T("\n"), _T("\n-"));
 
         wxDateTime actualDate = wxDateTime::Now();
         wxString changesTitle = cbC2U(GetConfig().ChangesLog.AppTitle.c_str());
@@ -852,10 +852,9 @@
 
         wxString changesOutput = _T("");
 
-        changesOutput << actualDate.Format(_T("%d %B %Y\n"));
-        changesOutput << _T("   ") << changesTitle << _T("\n");
-        changesOutput << _T("\n     Change log:\n");
-        changesOutput << changes << _T("\n\n");
+        changesOutput << changesTitle << _T("\n");
+        changesOutput << _T("\n");
+        changesOutput << changes << _T("\n\n\n");
         changesOutput << changesCurrentContent;
 
         wxFile file;
