Index: src/plugins/contrib/SpellChecker/DictionariesNeededDialog.cpp
===================================================================
--- src/plugins/contrib/SpellChecker/DictionariesNeededDialog.cpp	(revision 7464)
+++ src/plugins/contrib/SpellChecker/DictionariesNeededDialog.cpp	(working copy)
@@ -15,7 +15,7 @@
         wxSizerItemList& childs = outerSizer->GetChildren();
         wxSizer *mainArea = childs[0]->GetSizer();
 
-        wxHyperlinkCtrl *hyperlink1 = new wxHyperlinkCtrl( this, wxID_ANY, wxT("How to configure SpellChecker?"), wxT("http://wiki.codeblocks.org/index.php?title=SpellChecker"), wxDefaultPosition, wxDefaultSize, wxHL_DEFAULT_STYLE );
+        wxHyperlinkCtrl *hyperlink1 = new wxHyperlinkCtrl( this, wxID_ANY, wxT("How to configure SpellChecker?"), wxT("http://wiki.codeblocks.org/index.php?title=SpellChecker_plugin"), wxDefaultPosition, wxDefaultSize, wxHL_DEFAULT_STYLE );
 
         mainArea->Add(hyperlink1, 0, wxALL, 5);
 
Index: src/plugins/contrib/SpellChecker/dictionaries/readme.txt
===================================================================
--- src/plugins/contrib/SpellChecker/dictionaries/readme.txt	(revision 7464)
+++ src/plugins/contrib/SpellChecker/dictionaries/readme.txt	(working copy)
@@ -3,5 +3,5 @@
 http://wiki.services.openoffice.org/wiki/Dictionaries
 OOo 2.x files. OOo 3.x Extensions (.oxt) are zip'ed archives containing these files.
 During the build process, the dictionaries will be copied to the right place.
-Notice tat we do not priovide any dictionaires due to limiting the bandwidth.
+Notice that we do not provide any dictionaries due to limiting the bandwidth.
 The Code::Blocks team.
Index: src/plugins/contrib/SpellChecker/SpellCheckSettingsPanel.xrc
===================================================================
--- src/plugins/contrib/SpellChecker/SpellCheckSettingsPanel.xrc	(revision 7464)
+++ src/plugins/contrib/SpellChecker/SpellCheckSettingsPanel.xrc	(working copy)
@@ -143,8 +143,8 @@
 			<object class="sizeritem">
 				<object class="wxHyperlinkCtrl" name="ID_HYPERLINKCTRL1">
 					<label>How to configure SpellChecker?</label>
-					<url>http://wiki.codeblocks.org/index.php?title=SpellChecker</url>
-					<tooltip>Get help from http://wiki.codeblocks.org/index.php?title=SpellChecker</tooltip>
+					<url>http://wiki.codeblocks.org/index.php?title=SpellChecker_plugin</url>
+					<tooltip>Get help from http://wiki.codeblocks.org/index.php?title=SpellChecker_plugin</tooltip>
 				</object>
 				<flag>wxALL|wxALIGN_LEFT|wxALIGN_TOP</flag>
 				<border>5</border>
