Index: src/sdk/wxscintilla/src/wxscintilla.cpp
===================================================================
--- src/sdk/wxscintilla/src/wxscintilla.cpp	(revision 7663)
+++ src/sdk/wxscintilla/src/wxscintilla.cpp	(working copy)
@@ -4793,6 +4794,7 @@
     if (m_lastKeyDownConsumed && evt.GetUnicodeKey() > 255)
         m_lastKeyDownConsumed = false;
 #endif
+    m_lastKeyDownConsumed=false;
 
     if (!m_lastKeyDownConsumed && !skip) {
 #if wxUSE_UNICODE
