Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 7934)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -412,6 +412,10 @@
             int lineEnd = control->GetLineEndPosition(line);
             int i = lineEnd-1;
             wxChar ch = (wxChar)(control->GetCharAt(i));
+            if (control->GetLexer() == wxSCI_LEX_DIFF)
+            {
+                lineStart++;
+            }
             while ((i >= lineStart) && ((ch == _T(' ')) || (ch == _T('\t'))))
             {
                 i--;
