Index: src/sdk/editorcolourset.cpp
===================================================================
--- src/sdk/editorcolourset.cpp	(revision 7945)
+++ src/sdk/editorcolourset.cpp	(working copy)
@@ -381,7 +381,7 @@
 {
     // convert filename to lowercase first (m_FileMasks already contains
     // lowercase-only strings)
-    wxString lfname = filename.Lower();
+    wxString lfname = wxFileName(filename.Lower()).GetFullName();
 
     // first search in filemasks
     for (OptionSetsMap::iterator it = m_Sets.begin(); it != m_Sets.end(); ++it)
