Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 7946)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -3220,7 +3220,7 @@
     } // SelectionBraceCompletion
 
     // indent
-    if (ch == _T('\n'))
+    if (ch == _T('\n') || (control->GetEOLMode() == wxSCI_EOL_CR && ch == _T('\r')))
     {
         control->BeginUndoAction();
         // new-line: adjust indentation
