Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 8065)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -2824,16 +2824,16 @@
         pop->AppendSeparator();
         if (GetEditor(event.GetSelection())->GetModified())
             pop->Append(idNBTabSave, _("Save"));
-        if (any_modified > 1)
+        if (any_modified > 1 || !GetEditor(event.GetSelection())->GetModified())
             pop->Append(idNBTabSaveAll, _("Save all"));
     }
 
     pop->AppendSeparator();
-    pop->Append(idNBSwapHeaderSource, _("Swap header/source"));
 
     cbEditor* ed = GetBuiltinEditor(event.GetSelection());
     if (ed)
     {
+        pop->Append(idNBSwapHeaderSource, _("Swap header/source"));
         pop->Append(idNBTabOpenContainingFolder, _("Open containing folder"));
         pop->Append(idNBProperties, _("Properties..."));
         pop->AppendSeparator();
