Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 8096)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -1729,6 +1729,7 @@
             case crAll:
                 replace = true;
                 confirm = false;
+                control->Freeze();
                 break;
             case crCancel:
                 stop = true;
@@ -1790,6 +1791,8 @@
             }
         }
     }
+    if (control->IsFrozen())
+        control->Thaw();
     control->EndUndoAction();
     wxString msg;
     if (foundcount == 0)
