Index: src/plugins/compilergcc/compilerGDC.cpp
===================================================================
--- src/plugins/compilergcc/compilerGDC.cpp	(revision 8237)
+++ src/plugins/compilergcc/compilerGDC.cpp	(working copy)
@@ -110,7 +110,7 @@
     category = _("D");
     m_Options.AddOption(_("generate documentation"), _T("-fdoc"), category);
     m_Options.AddOption(_("allow deprecated features"), _T("-fdeprecated"), category);
-    m_Options.AddOption(_("compile in debug code"), _T("-debug"), category);
+    m_Options.AddOption(_("compile in debug code"), _T("-fdebug"), category);
     m_Options.AddOption(_("inline expand functions"), _T("-finline-functions"), category);
     m_Options.AddOption(_("compile release version, which means not generating code for contracts and asserts"), _T("-frelease"), category);
     m_Options.AddOption(_("compile in unittest code, also turns on asserts"), _T("-funittest"), category);
