Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 8250)
+++ src/src/main.cpp	(working copy)
@@ -2051,7 +2051,6 @@
 
 void MainFrame::ShowHideStartPage(bool forceHasProject, int forceState)
 {
-    Manager::Get()->GetLogManager()->LogWarning(_("Toggling start page"));
     if (Manager::IsBatchBuild())
         return;
 
@@ -2080,7 +2079,6 @@
     if (show && !sh)
     {
         sh = new StartHerePage(this, Manager::Get()->GetEditorManager()->GetNotebook());
-
     }
     else if (!show && sh)
     {
