Index: src/plugins/abbreviations/abbreviations.cpp
===================================================================
--- src/plugins/abbreviations/abbreviations.cpp	(revision 8429)
+++ src/plugins/abbreviations/abbreviations.cpp	(working copy)
@@ -244,6 +244,11 @@
 
             // replace any other macros in the generated code
             Manager::Get()->GetMacrosManager()->ReplaceMacros(code);
+            // match current EOL mode
+            if (control->GetEOLMode() == wxSCI_EOL_CRLF)
+                code.Replace(wxT("\n"), wxT("\r\n"));
+            else if (control->GetEOLMode() == wxSCI_EOL_CR)
+                code.Replace(wxT("\n"), wxT("\r"));
             // add the text
             control->InsertText(curPos, code);
 
