Index: src/plugins/contrib/SmartIndent/SmartIndentCpp.cpp
===================================================================
--- src/plugins/contrib/SmartIndent/SmartIndentCpp.cpp	(revision 8551)
+++ src/plugins/contrib/SmartIndent/SmartIndentCpp.cpp	(working copy)
@@ -42,7 +42,7 @@
         return;
 
     wxString langname = Manager::Get()->GetEditorManager()->GetColourSet()->GetLanguageName(ed->GetLanguage());
-    if ( langname != _T("C/C++") && langname != _T("D") && langname != _T("Java") ) return;
+    if ( langname != wxT("D") && (stc->GetLexer() != wxSCI_LEX_CPP || langname == wxT("Hitachi asm"))) return;
 
     ed->AutoIndentDone(); // we are responsible.
 
