--- C:/Program Files (x86)/_/program/codeblocks/share/CodeBlocks/templates/wizard/common_functions.script	Thu Nov 01 21:12:47 2012
+++ D:/bat/data/program/gcc/codeblocks/src/plugins/scriptedwizard/resources/common_functions.script	Sun Nov 11 22:47:55 2012
@@ -49,10 +49,6 @@
     {
         // with SDCC all warnings are enabled by default. You can only ask for less by using --less-pedantic or --disable-warning.
     }
-    else if( GetCompilerFactory().CompilerInheritsFrom(compilerID, _T("lcc")) )
-    {
-        base.AddCompilerOption(_T("-A"));
-    }
     else
     {
         ShowWarning(_T("This wizard doesn't know how to setup warning flags for this compiler.\n"));
@@ -106,10 +102,6 @@
     {
         base.AddCompilerOption(_T("--debug"));
     }
-    else if (GetCompilerFactory().CompilerInheritsFrom(compilerID, _T("lcc")))
-    {
-        base.AddCompilerOption(_T("-g2"));
-    }
     else
     {
         ShowWarning(_T("This wizard doesn't know how to setup debug flags for this compiler.\n"));
@@ -183,10 +175,6 @@
     else if (GetCompilerFactory().CompilerInheritsFrom(compilerID, _T("sdcc")))
     {
         base.AddCompilerOption(_T("--opt-code-size"));
-    }
-    else if (GetCompilerFactory().CompilerInheritsFrom(compilerID, _T("lcc")))
-    {
-        base.AddCompilerOption(_T("-O"));
     }
     else
     {
