Index: cdb_driver.cpp
===================================================================
--- cdb_driver.cpp	(revision 8484)
+++ cdb_driver.cpp	(working copy)
@@ -86,6 +86,12 @@
     else
         cmd << m_Target->GetParentProject()->GetBasePath() << wxT("/") << debuggee;
 
+    if (!userArguments.IsEmpty())
+    {
+        cmd << _T(' ');
+        cmd << userArguments;
+    }
+
     return cmd;
 }
 
