Index: src/plugins/codecompletion/classbrowserbuilderthread.cpp
===================================================================
--- src/plugins/codecompletion/classbrowserbuilderthread.cpp	(wersja 8640)
+++ src/plugins/codecompletion/classbrowserbuilderthread.cpp	(kopia robocza)
@@ -1039,8 +1039,8 @@
             // modification suggested by ollydbg in http://forums.codeblocks.org/index.php/topic,10242.msg70865.html#msg70865:
 //            if (!token->m_ActualType.IsEmpty())
 //                 str = str + _T(" : ") + token->m_ActualType;
-            if (!token->m_BaseType.IsEmpty())
-                str = str + _T(" : ") + token->m_BaseType + token->m_TemplateArgument;
+            if (!token->m_FullType.IsEmpty())
+                str = str + _T(" : ") + token->m_FullType + token->m_TemplateArgument;
 
             wxTreeItemId child = tree->AppendItem(parent, str, img, img, new CCTreeCtrlData(sfToken, token, tokenKindMask));
 
