Index: src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- src/plugins/compilergcc/compilergcc.cpp	(wersja 8771)
+++ src/plugins/compilergcc/compilergcc.cpp	(kopia robocza)
@@ -103,6 +103,13 @@
         style[caption].SetFont(style[error].GetFont());
         style[warning].SetTextColour(BlendTextColour(*wxBLUE));
         style[error].SetFont(style[info].GetFont());
+        
+        ConfigManager *mcfg = Manager::Get()->GetConfigManager(_T("message_manager"));
+        if(mcfg)
+        {
+            style[warning].SetTextColour(mcfg->ReadColour(_T("/log_warning_text_colour"), *wxBLUE));
+            style[error].SetTextColour(mcfg->ReadColour(_T("/log_error_text_colour"), *wxRED));
+        }
     }
 
     virtual wxWindow* CreateControl(wxWindow* parent)
