Index: src/plugins/codecompletion/parser/parserthread.cpp
===================================================================
--- src/plugins/codecompletion/parser/parserthread.cpp (wersja 8771)
+++ src/plugins/codecompletion/parser/parserthread.cpp (kopia robocza)
@@ -1077,7 +1079,9 @@
else if ( (peek == ParserConsts::semicolon)
|| ( ( m_Options.useBuffer
&& (peek.GetChar(0) == ParserConsts::opbracket_chr) )
- && (!m_Str.Contains(ParserConsts::dcolon)) ) )
+ && (!m_Str.Contains(ParserConsts::dcolon))
+ && (!m_Str.Contains(ParserConsts::opbracket_chr))
+ && (!m_Str.Contains(ParserConsts::clbracket_chr)) ) )
{
if ( !m_Str.IsEmpty()
&& ( wxIsalpha(token.GetChar(0)) 