Index: src/include/prep.h
===================================================================
--- src/include/prep.h	(revision 8813)
+++ src/include/prep.h	(working copy)
@@ -9,8 +9,12 @@
 #ifndef wxMAJOR_VERSION
     #include <wx/version.h>
 #endif
+#ifndef __has_feature
+    #define __has_feature(x) 0
+#endif
 
-#if !(__GNUC__ == 4 && __GNUC_MINOR__ >= 6 && defined __GXX_EXPERIMENTAL_CXX0X__)
+#if !(__GNUC__ == 4 && __GNUC_MINOR__ >= 6 && defined __GXX_EXPERIMENTAL_CXX0X__) \
+    && !(__clang__ && __has_feature(cxx_nullptr))
     // it is a const object...
     const class nullptr_t
     {
