Index: src/plugins/codecompletion/nativeparser_base.cpp
===================================================================
--- src/plugins/codecompletion/nativeparser_base.cpp	(wersja 8909)
+++ src/plugins/codecompletion/nativeparser_base.cpp	(kopia robocza)
@@ -621,6 +621,8 @@
     while (   (nest > 0)
            && (startAt < line.Len()) )
     {
+        if (line.GetChar(startAt) == '(')
+            ++nest;
         if (line.GetChar(startAt) == ')')
             --nest;
         ++startAt;
