Index: src/plugins/codecompletion/classbrowser.cpp
===================================================================
--- src/plugins/codecompletion/classbrowser.cpp	(wersja 8909)
+++ src/plugins/codecompletion/classbrowser.cpp	(kopia robocza)
@@ -841,7 +841,9 @@
         // now the actual token
         m_CCTreeCtrl->Expand(start);
         m_CCTreeCtrl->SelectItem(start);
-        wxTreeItemId res = FindChild(token->m_Name, m_CCTreeCtrl, start);
+        bool isBottomShown = m_CCTreeCtrlBottom->IsShown();
+        wxTreeItemId res = FindChild(token->m_Name, m_CCTreeCtrl, start, !isBottomShown, !isBottomShown);
+        
         if (res.IsOk())
             m_CCTreeCtrl->SelectItem(res);
         else
