Index: sdk/pluginsconfigurationdlg.cpp
===================================================================
--- sdk/pluginsconfigurationdlg.cpp	(revision 8929)
+++ sdk/pluginsconfigurationdlg.cpp	(working copy)
@@ -140,7 +140,7 @@
         long idx = list->InsertItem(i, elem->info.title);
         list->SetItem(idx, 1, elem->info.version);
         list->SetItem(idx, 2, elem->plugin->IsAttached() ? _("Yes") : _("No"));
-        list->SetItem(idx, 3, UnixFilename(elem->fileName));
+        list->SetItem(idx, 3, UnixFilename(elem->fileName).AfterLast(wxFileName::GetPathSeparator()));
         list->SetItemData(idx, (long)elem);
 
         if (!elem->plugin->IsAttached())

