Index: src/plugins/codecompletion/parser/tokentree.cpp
===================================================================
--- src/plugins/codecompletion/parser/tokentree.cpp	(revision 9000)
+++ src/plugins/codecompletion/parser/tokentree.cpp	(working copy)
@@ -907,6 +907,9 @@
 void TokenTree::AppendDocumentation(int tokenIdx, const wxString& doc)
 {
     wxString& newDoc = m_TokenDocumentationMap[tokenIdx];
+    // Don't add the same document!
+    if ((newDoc != _T("")) && (newDoc == doc))
+        return;
     newDoc += doc;
     newDoc.Shrink();
 }
