Index: src/sdk/compilercommandgenerator.cpp
===================================================================
--- src/sdk/compilercommandgenerator.cpp	(revision 9123)
+++ src/sdk/compilercommandgenerator.cpp	(working copy)
@@ -373,7 +373,9 @@
         wxArrayString aCflags = GetArrayFromString(cFlags, wxT(" "));
         for (size_t i = 0; i < remFlags.GetCount(); ++i)
         {
-            aCflags.Remove(remFlags[i]);
+            int index = aCflags.Index(remFlags[i]);
+            if (index != wxNOT_FOUND)
+                aCflags.RemoveAt(index);
         }
         cFlags = GetStringFromArray(aCflags, wxT(" "), false);
     }
