Index: src/sdk/globals.cpp
===================================================================
--- src/sdk/globals.cpp	(revision 9123)
+++ src/sdk/globals.cpp	(working copy)
@@ -1019,8 +1019,8 @@
     long flags = lc->GetWindowStyleFlag();
     switch (style)
     {
-        case sisNoIcons: flags = (flags & ~wxLC_ICON) | wxLC_SMALL_ICON; break;
-        default: flags = (flags & ~wxLC_SMALL_ICON) | wxLC_ICON; break;
+        case sisNoIcons: flags = (flags & ~wxLC_ICON & ~wxLC_LIST & ~wxLC_REPORT) | wxLC_SMALL_ICON; break;
+        default: flags = (flags & ~wxLC_SMALL_ICON & ~wxLC_LIST & ~wxLC_REPORT) | wxLC_ICON; break;
     }
     lc->SetWindowStyleFlag(flags);
 #endif
