Index: src/plugins/contrib/wxContribItems/wxTreeList/src/treelistctrl.cpp
===================================================================
--- src/plugins/contrib/wxContribItems/wxTreeList/src/treelistctrl.cpp	(revision 9378)
+++ src/plugins/contrib/wxContribItems/wxTreeList/src/treelistctrl.cpp	(working copy)
@@ -2021,14 +2021,8 @@
                                    const wxString& name) {
 
 #ifdef __WXMAC__
-    if (style & wxTR_HAS_BUTTONS) style |= wxTR_MAC_BUTTONS;
-    if (style & wxTR_HAS_BUTTONS) style &= ~wxTR_HAS_BUTTONS;
     style &= ~wxTR_LINES_AT_ROOT;
     style |= wxTR_NO_LINES;
-
-    int major,minor;
-    wxGetOsVersion( &major, &minor );
-    if (major < 10) style |= wxTR_ROW_LINES;
 #endif
 
     wxScrolledWindow::Create (parent, id, pos, size, style|wxHSCROLL|wxVSCROLL, name);
