Index: projectmanager.cpp
===================================================================
--- projectmanager.cpp	(revision 9447)
+++ projectmanager.cpp	(working copy)
@@ -579,7 +579,8 @@
             m_IsClosingWorkspace = false;
             return false;
         }
-        if (!CloseAllProjects(false))
+        // m_ui->QueryCloseWorkspace asked for saving workspace AND projects, no need to do again
+        if (!CloseAllProjects(true))
         {
             m_IsClosingWorkspace = false;
             return false;
