Index: plugins/scriptedwizard/resources/sfml/wizard.script
===================================================================
--- plugins/scriptedwizard/resources/sfml/wizard.script	(revision 9677)
+++ plugins/scriptedwizard/resources/sfml/wizard.script	(working copy)
@@ -5,7 +5,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 // globals
-SFMLVersion <- 1	// 0 - SFML 1.6, 1 - SFML 2.0
+SFMLVersion <- 1	// 0 - SFML 1.6, 1 - SFML 2.1
 SFMLLinkType <- 1	// 0 - static library, 1 - dynamic link library
 SFMLDebugLib <- 0   // 0 - debug libraries are not available, 1 - debug libraries are available
 // windows globals
@@ -27,7 +27,7 @@
                              "\"include\" and \"lib\".");
 
     Wizard.AddInfoPage(_T("SFMLIntro"), intro_msg);
-    Wizard.AddGenericSingleChoiceListPage(_T("SFMLVersionPage"), _T("Please select the SFML version you want to use."), _T("SFML 1.6;SFML 2.0"), SFMLVersion);
+    Wizard.AddGenericSingleChoiceListPage(_T("SFMLVersionPage"), _T("Please select the SFML version you want to use."), _T("SFML 1.6;SFML 2.1"), SFMLVersion);
     if (PLATFORM == PLATFORM_MSW)
         Wizard.AddGenericSingleChoiceListPage(_T("SFMLLinkPage"), _T("Please choose the kind of library you want to link to."), _T("Static Library;Dynamic Link Library"), SFMLLinkType);
     Wizard.AddProjectPathPage();
