Index: sdl/wizard.script
===================================================================
--- sdl/wizard.script	(revision 9677)
+++ sdl/wizard.script	(working copy)
@@ -5,16 +5,16 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 // globals (used under windows only)
-SDLPathDefault    <- _T("$(#sdl)");
-SDLPathDefaultInc <- _T("$(#sdl.include)");
-SDLPathDefaultLib <- _T("$(#sdl.lib)");
+SDLPathDefault    <- _T("");
+SDLPathDefaultInc <- _T("");
+SDLPathDefaultLib <- _T("");
 SDLPath           <- _T("");
 
 function BeginWizard()
 {
     local intro_msg = _T("Welcome to the new SDL project wizard!\n\n" +
-                         "This wizard will guide you to create a new project\n" +
-                         "using the SDL graphics library.\n\n" +
+                         "This wizard will guide you to create a new\n" +
+                         "project using the SDL graphics library.\n\n" +
                          "When you 're ready to proceed, please click \"Next\"...");
 
     local sdlpath_descr = _T("Please select the location of SDL on your computer.\n" +
