Index: tokenizer.cpp
===================================================================
--- tokenizer.cpp	(revision 9745)
+++ tokenizer.cpp	(working copy)
@@ -228,7 +228,9 @@
         if (m_Buffer.Length() == 0)
         {
             // could not read as utf-8 encoding, try iso8859-1
-            m_Buffer = wxString(data, wxConvISO8859_1, m_BufferLen + 1); // + 1 => sentinel
+            // And what about other encodings (Cyrillic, Arabic, Greec, etc.)?
+            // Better use local encoding
+            m_Buffer = wxString(data, wxConvLocal, m_BufferLen + 1); // + 1 => sentinel
         }
 #else
         m_Buffer = wxString(data, m_BufferLen + 1); // + 1 => sentinel
