Index: C:/Programme/DANAE/CodeBlocks/CodeBlocks/src/sdk/pipedprocess.cpp
===================================================================
--- C:/Programme/DANAE/CodeBlocks/CodeBlocks/src/sdk/pipedprocess.cpp	(revision 1984)
+++ C:/Programme/DANAE/CodeBlocks/CodeBlocks/src/sdk/pipedprocess.cpp	(working copy)
@@ -57,7 +57,7 @@
         // The only change, is the removal of the MB2WC function
         // With PipedProcess we work with compilers/debuggers which (usually) don't
         // send us unicode (at least GDB).
-        wxChar cbTextInputStream::NextChar()
+        wxChar /*cbTextInputStream::*/NextChar()
         {
         #if wxUSE_UNICODE
             wxChar wbuf[2];
