Index: src/src/appglobals.cpp
===================================================================
--- src/src/appglobals.cpp	(revision 2040)
+++ src/src/appglobals.cpp	(arbetskopia)
@@ -27,7 +27,7 @@
 #elif defined(__WXOS2__)
   const wxString g_AppPlatform = _T("OS/2");
 #elif defined(__WXMAC__)
-  const wxString g_AppPlatform = _T("Mac OS/X");
+  const wxString g_AppPlatform = _T("Mac OS X");
 #elif defined(__UNIX__)
   const wxString g_AppPlatform = _T("Linux");
 #else
