Index: C:/CodeBlocks/src/sdk/cbeditor.h
===================================================================
--- C:/CodeBlocks/src/sdk/cbeditor.h	(revision 2068)
+++ C:/CodeBlocks/src/sdk/cbeditor.h	(working copy)
@@ -18,7 +18,6 @@
 class cbEditor;
 class ProjectFile;
 class EditorColorSet;
-class wxNotebook;
 
 class cbStyledTextCtrl : public wxScintilla
 {
Index: C:/CodeBlocks/src/sdk/manager.h
===================================================================
--- C:/CodeBlocks/src/sdk/manager.h	(revision 2068)
+++ C:/CodeBlocks/src/sdk/manager.h	(working copy)
@@ -15,7 +15,6 @@
 
 // forward decls
 class wxFrame;
-class wxNotebook;
 class ProjectManager;
 class EditorManager;
 class MessageManager;
