Index: C:/CodeBlocks/src/plugins/xpmanifest/windowsxplooknfeel.cpp
===================================================================
--- C:/CodeBlocks/src/plugins/xpmanifest/windowsxplooknfeel.cpp	(revision 2071)
+++ C:/CodeBlocks/src/plugins/xpmanifest/windowsxplooknfeel.cpp	(working copy)
@@ -83,10 +83,6 @@
 	if (!m_IsAttached)
 		return -1;
 
-#ifndef __WXMSW__
-	#error This plugin is for use under Win32 only!
-#endif // __WXMSW__
-
 	cbProject* project = Manager::Get()->GetProjectManager()->GetActiveProject();
 	if (!project)
 	{
@@ -172,7 +168,7 @@
 		wxFile file(filename, wxFile::write);
 		cbWrite(file,buffer);
 
-		cbMessageBox(_("Manifest file created"));
+		cbMessageBox(_("Manifest file created"), _("Information"), wxICON_INFORMATION | wxOK);
 	}
 
 	return 0;
Index: C:/CodeBlocks/src/plugins/xpmanifest/windowsxplooknfeel.h
===================================================================
--- C:/CodeBlocks/src/plugins/xpmanifest/windowsxplooknfeel.h	(revision 2071)
+++ C:/CodeBlocks/src/plugins/xpmanifest/windowsxplooknfeel.h	(working copy)
@@ -17,8 +17,6 @@
 		int Execute();
 		void OnAttach(); // fires when the plugin is attached to the application
 		void OnRelease(bool appShutDown); // fires when the plugin is released from the application
-	protected:
-	private:
 };
 
 CB_DECLARE_PLUGIN();
