Index: C:/Documents and Settings/sjackson/My Documents/web/features.php
===================================================================
--- C:/Documents and Settings/sjackson/My Documents/web/features.php	(revision 2071)
+++ C:/Documents and Settings/sjackson/My Documents/web/features.php	(working copy)
@@ -8,7 +8,7 @@
 <ul class="bullet">
 <li><b>Open Source</b>! GPL2, no hidden costs.
 </li><li> <b>Cross-platform</b>. Runs on Linux or Windows (uses wxWidgets).
-</li><li> <b>Made in GNU C++</b>. No interpreted languages or proprietary libs needed.
+</li><li> <b>Made in C++</b>. No interpreted languages or proprietary libs needed.
 </li><li> <b>Comes in two presentations</b>: Standalone, and MinGW bundle
 </li><li> <b>Devpack support</b> (optional)
 </li><li> <b>Extensible through plugins</b> (SDK available in the downloads section)
@@ -33,7 +33,7 @@
 
 <h3>Interface Features:</h3>
 <ul class="bullet"><li> <b>Syntax highlighting</b>, customizable and extensible
-</li><li> <b>Code folding</b> for C++ and XML files.
+</li><li> <b>Code folding</b>
 </li><li> <b>Tabbed interface</b>
 </li><li> <b>Code completion plugin</b>
 </li><li> <b>Class Browser</b>
@@ -46,4 +46,4 @@
 
 <p><b>And more!</b> (See the forums for plugins currently being worked on)</p>
 
-<p class="caption"><b>[ <a href="/">Home</a> | <a title="Testimonials" href="/testimonials.shtml">Next</a> ]</b></p>
\ No newline at end of file
+<p class="caption"><b>[ <a href="/">Home</a> | <a title="Testimonials" href="/testimonials.shtml">Next</a> ]</b></p>
