Index: src/sdk/messagelog.cpp
===================================================================
--- src/sdk/messagelog.cpp	(revision 2071)
+++ src/sdk/messagelog.cpp	(working copy)
@@ -30,7 +30,7 @@
     #include "messagelog.h" // class's header file
     #include "manager.h"
     #include "messagemanager.h"
-
+    #include "configmanager.h"
     #include <wx/intl.h>
 #endif
 
Index: src/sdk/workspaceloader.cpp
===================================================================
--- src/sdk/workspaceloader.cpp	(revision 2071)
+++ src/sdk/workspaceloader.cpp	(working copy)
@@ -22,7 +22,7 @@
     #include "projectmanager.h"
     #include "messagemanager.h"
     #include "cbproject.h"
-    #inclued "globals.h"
+    #include "globals.h"
     #include "workspaceloader.h"
 #endif
 
