Index: C:/CodeBlocks/src/sdk/templatemanager.cpp
===================================================================
--- C:/CodeBlocks/src/sdk/templatemanager.cpp	(revision 2086)
+++ C:/CodeBlocks/src/sdk/templatemanager.cpp	(working copy)
@@ -313,7 +313,12 @@
 
     // select directory to copy user template files
     wxString sep = wxFileName::GetPathSeparator();
-    wxString path = ChooseDirectory(0, _("Choose a directory to create the new project"));
+    wxString path = ChooseDirectory(0,
+                                    _("Choose a directory to create the new project"),
+                                    _T(""),
+                                    _T(""),
+                                    false,
+                                    true);
     if (path.IsEmpty())
     {
         return NULL;
