Index: src/sdk/resources/new_from_template.xrc
===================================================================
--- src/sdk/resources/new_from_template.xrc	(revision 2090)
+++ src/sdk/resources/new_from_template.xrc	(working copy)
@@ -193,15 +193,15 @@
                 <object class="sizeritem">
                   <object class="wxStaticText">
                     <label>* To delete a user template, delete the similarly named folder created
-   in %USER_HOME%\.CodeBlocks (for Windows)
-   or $$(HOME)\.CodeBlocks (for other platforms).</label>
+   in %APPDATA%\codeblocks\UserTemplates (for Windows)
+   or ~/.codeblocks (for other platforms).</label>
                   </object>
                   <flag>wxLEFT|wxRIGHT</flag>
                   <border>8</border>
                 </object>
                 <object class="sizeritem">
                   <object class="wxStaticText">
-                    <label>   Please note the dot (.) in front of &quot;CodeBlocks&quot;!</label>
+                    <label>   Please note the dot (.) in front of &quot;codeblocks&quot;!</label>
                     <fg>#800000</fg>
                   </object>
                   <flag>wxBOTTOM|wxLEFT|wxRIGHT</flag>
@@ -236,4 +236,4 @@
       </object>
     </object>
   </object>
-</resource>
\ No newline at end of file
+</resource>
