Index: src/sdk/messagemanager.cpp
===================================================================
--- src/sdk/messagemanager.cpp	(revision 2092)
+++ src/sdk/messagemanager.cpp	(working copy)
@@ -343,6 +343,8 @@
             m_pNotebook->RemovePage(id);
         ls->visible = false;
 
+        log->Show(false);
+
         if (id == m_DebugLog)
             cfg->Write(_T("/has_debug_log"), (bool)false);
     }
