Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp	(revision 2141)
+++ src/sdk/editormanager.cpp	(working copy)
@@ -208,11 +208,6 @@
 
     CreateSearchLog();
     LoadAutoComplete();
-
-#if !wxCHECK_VERSION(2, 5, 0)
-    /*wxNotebookSizer* nbs =*/
-    new wxNotebookSizer(m_pNotebook);
-#endif
 }
 
 // class destructor
