Index: src/sdk/editorlexerloader.cpp
===================================================================
--- src/sdk/editorlexerloader.cpp	(revision 2142)
+++ src/sdk/editorlexerloader.cpp	(working copy)
@@ -17,6 +17,7 @@
     #include "manager.h"
     #include "messagemanager.h"
     #include <wx/dynarray.h>
+    #include <wx/wxscintilla.h>
 #endif
 
 #include "editorcolorset.h"
Index: src/sdk/editkeywordsdlg.cpp
===================================================================
--- src/sdk/editkeywordsdlg.cpp	(revision 2142)
+++ src/sdk/editkeywordsdlg.cpp	(working copy)
@@ -17,6 +17,7 @@
     #include <wx/xrc/xmlres.h>
     #include <wx/spinctrl.h>
     #include <wx/textctrl.h>
+    #include <wx/wxscintilla.h>
 #endif
 
 #include "editkeywordsdlg.h"
