Index: src/sdk/editormanager.cpp
===================================================================
--- src/sdk/editormanager.cpp   (revision 2137)
+++ src/sdk/editormanager.cpp   (working copy)
@@ -1154,6 +1154,14 @@
         return -2;
     }
 
+    // Don't look for empty strings:
+       if (dlg->GetFindString().empty())
+       {
+               delete dlg;
+               cbMessageBox(_("Can't look for an empty search criterion!"), _("Error"), wxOK | wxICON_EXCLAMATION, Manager::Get()->GetAppWindow());
+               return -2;
+       }
+
     if (!m_LastFindReplaceData)
         m_LastFindReplaceData = new cbFindReplaceData;
