Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 2156)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -1759,7 +1759,7 @@
     if (!prj)
         return;
 
-    wxFileDialog dlg(m_pTree,
+    wxFileDialog dlg(Manager::Get()->GetAppWindow(),
                     _("Add files to project..."),
                     prj->GetBasePath(),
                     wxEmptyString,
