Index: src/plugins/compilergcc/resources/compiler_options.xrc
===================================================================
--- src/plugins/compilergcc/resources/compiler_options.xrc	(revision 2159)
+++ src/plugins/compilergcc/resources/compiler_options.xrc	(working copy)
@@ -158,6 +158,7 @@
                                     <border>8</border>
                                     <object class="wxCheckList" name="lstCompilerOptions">
                                       <content/>
+                                      <style>wxLB_HSCROLL</style>
                                     </object>
                                   </object>
                                 </object>
