Index: src/sdk/cbworkspace.cpp
===================================================================
--- src/sdk/cbworkspace.cpp	(revision 2171)
+++ src/sdk/cbworkspace.cpp	(working copy)
@@ -27,8 +27,8 @@
 #include "sdk_precomp.h"
 
 #ifndef CB_PRECOMP
+    #include "globals.h"
     #include "cbworkspace.h"
-    #include "globals.h"
     #include "manager.h"
     #include "configmanager.h"
     #include "messagemanager.h"
Index: src/sdk/filefilters.cpp
===================================================================
--- src/sdk/filefilters.cpp	(revision 2171)
+++ src/sdk/filefilters.cpp	(working copy)
@@ -1,5 +1,6 @@
 #include <map>
 #include <sdk_precomp.h>
+#include <globals.h>
 #include "filefilters.h"
 
 typedef std::map<wxString, wxString> FileFiltersMap;
