Index: src/plugins/todo/todolist.cpp
===================================================================
--- src/plugins/todo/todolist.cpp	(revision 2173)
+++ src/plugins/todo/todolist.cpp	(working copy)
@@ -291,8 +291,10 @@
     // continue with the type
 	buffer << dlg.GetType() << _T(" ");
 
+    wxString priority = wxString::Format(_T("%d"), dlg.GetPriority());
+
 	// now do the () part
-	buffer << _T("(") << dlg.GetUser() << _T("#") << dlg.GetPriority() << _T("#): ");
+	buffer << _T("(") << dlg.GetUser() << _T("#") << priority << _T("#): ");
 
     wxString text = dlg.GetText();
     if (CmtType != tdctC)
