Index: D:/EigeneDateien/Code/CodeBlocks/svn/src/plugins/contrib/profiler/cbprofiler.cpp
===================================================================
--- D:/EigeneDateien/Code/CodeBlocks/svn/src/plugins/contrib/profiler/cbprofiler.cpp	(revision 2188)
+++ D:/EigeneDateien/Code/CodeBlocks/svn/src/plugins/contrib/profiler/cbprofiler.cpp	(working copy)
@@ -117,6 +117,7 @@
 	if (target)
 	{
         exename = target->GetOutputFilename();
+        Manager::Get()->GetMacrosManager()->ReplaceEnvVars(exename);
         wxFileName ename(exename);
         ename.Normalize(wxPATH_NORM_ALL, project->GetBasePath());
 		  exename = ename.GetFullPath();
