Index: src/sdk/messagemanager.cpp
===================================================================
--- src/sdk/messagemanager.cpp  (revision 2199)
+++ src/sdk/messagemanager.cpp  (working copy)
@@ -513,6 +513,7 @@
     {
         m_pNotebook->GetImageList()->push_back(bitmap);
         m_pNotebook->SetPageImageIndex(index, m_pNotebook->GetImageList()->size() - 1);
+        m_Logs[id]->bitmap = bitmap;
     }
 }
