Index: D:/codeblocks/src/copymo.bat
===================================================================
--- D:/codeblocks/src/copymo.bat	(revision 0)
+++ D:/codeblocks/src/copymo.bat	(revision 0)
@@ -0,0 +1,13 @@
+echo Copying language files...
+
+if not exist %LOCALEDIR%\zh_CN md %LOCALEDIR%\zh_CN
+copy /y po\zh_CN.mo    %LOCALEDIR%\zh_CN\codeblocks.mo 
+
+rem if not exist %LOCALEDIR%\zh_TW md %LOCALEDIR%\zh_TW
+rem copy /y po\zh_TW.mo    %LOCALEDIR%\zh_TW\codeblocks.mo
+
+rem if not exist %LOCALEDIR%\fr md %LOCALEDIR%\fr
+rem copy /y po\fr.mo    %LOCALEDIR%\fr\codeblocks.mo
+
+rem if not exist %LOCALEDIR%\ru md %LOCALEDIR%\ru
+rem copy /y po\ru.mo    %LOCALEDIR%\ru\codeblocks.mo
Index: D:/codeblocks/src/intl.bat
===================================================================
--- D:/codeblocks/src/intl.bat	(revision 0)
+++ D:/codeblocks/src/intl.bat	(revision 0)
@@ -0,0 +1,59 @@
+@echo off
+wxrc src\resources\*.xrc -g -o  src\src-xrc.cpp
+wxrc sdk\resources\*.xrc -g -o  sdk\sdk-xrc.cpp
+wxrc plugins\astyle\resources\*.xrc -g -o  plugins\astyle\astyle-xrc.cpp
+wxrc plugins\classwizard\resources\*.xrc -g -o  plugins\classwizard\classwizard-xrc.cpp
+wxrc plugins\codecompletion\resources\*.xrc -g -o  plugins\codecompletion\codecompletion-xrc.cpp
+wxrc plugins\compilergcc\resources\*.xrc -g -o  plugins\compilergcc\compilergcc-xrc.cpp
+wxrc plugins\debuggergdb\resources\*.xrc -g -o  plugins\debuggergdb\debuggergdb-xrc.cpp
+wxrc plugins\defaultmimehandler\resources\*.xrc -g -o  plugins\defaultmimehandler\defaultmimehandler-xrc.cpp
+wxrc plugins\pluginwizard\resources\*.xrc -g -o  plugins\pluginwizard\pluginwizard-xrc.cpp
+wxrc plugins\todo\resources\*.xrc -g -o  plugins\todo\todo-xrc.cpp
+
+if not exist po md po
+
+xgettext -C -n -k_ -o po\codeblocks.pot src\*.cpp src\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot src\wxAUI\*.cpp src\wxAUI\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\*.cpp sdk\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\as\bindings\*.cpp sdk\as\bindings\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\propgrid\src\propgrid\*.cpp sdk\propgrid\include\wx\propgrid\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\wxFlatNotebook\*.cpp
+rem xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\as\include\*.h
+rem xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\as\source\*.cpp sdk\as\source\*.h
+rem xgettext -C -n -k_ -j -o po\codeblocks.pot sdk\wxscintilla\*.cpp sdk\wxscintilla\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\codecompletion\*.cpp plugins\codecompletion\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\codecompletion\parser\*.cpp plugins\codecompletion\parser\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\astyle\*.cpp plugins\astyle\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\astyle\astyle\*.cpp plugins\astyle\astyle\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\classwizard\*.cpp plugins\classwizard\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\codecompletion\*.cpp plugins\codecompletion\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\compilergcc\*.cpp plugins\compilergcc\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\debuggergdb\*.cpp plugins\debuggergdb\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\defaultmimehandler\*.cpp plugins\defaultmimehandler\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\pluginwizard\*.cpp plugins\pluginwizard\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\todo\*.cpp plugins\todo\*.h
+xgettext -C -n -k_ -j -o po\codeblocks.pot plugins\xpmanifest\*.cpp plugins\xpmanifest\*.h
+
+cd po
+if exist temp.po delete temp.po
+
+rename zh_CN.po temp.po
+msgmerge -o zh_CN.po temp.po codeblocks.pot
+del temp.po
+msgfmt -o zh_CN.mo zh_CN.po
+
+rem rename en.po temp.po
+rem msgmerge -o en.po temp.po codeblocks.pot
+rem del temp.po
+rem msgfmt -o en.mo en.po
+
+cd ..
+
+if not exist devel md devel
+if not exist devel\share\codeblocks\locale md devel\share\codeblocks\locale
+
+set LOCALEDIR=devel\share\codeblocks\locale
+
+call copymo
+
+pause
\ No newline at end of file
Index: D:/codeblocks/src/update.bat
===================================================================
--- D:/codeblocks/src/update.bat	(revision 2285)
+++ D:/codeblocks/src/update.bat	(working copy)
@@ -4,6 +4,7 @@
 if not exist output md output\
 if not exist output\share md output\share\
 if not exist output\share\CodeBlocks md output\share\CodeBlocks\
+if not exist output\share\CodeBlocks\locale md output\share\CodeBlocks\locale\
 if not exist output\share\CodeBlocks\lexers md output\share\CodeBlocks\lexers\
 if not exist output\share\CodeBlocks\images md output\share\CodeBlocks\images\
 if not exist output\share\CodeBlocks\images\settings md output\share\CodeBlocks\images\settings\
@@ -25,8 +26,13 @@
 if not exist devel\share\CodeBlocks\scripts md devel\share\CodeBlocks\scripts\
 
 set ZIPCMD=zip
+set TZ=MET-1MEST
 set RESDIR=devel\share\CodeBlocks
 
+set LOCALEDIR=output\share\codeblocks\locale
+
+call copymo
+
 echo Packing core UI resources
 %ZIPCMD% -j9 %RESDIR%\resources.zip src\resources\*.xrc > nul
 %ZIPCMD% -j9 %RESDIR%\manager_resources.zip sdk\resources\*.xrc sdk\resources\images\*.png > nul
@@ -90,3 +96,5 @@
 
 set ZIPCMD=
 set RESDIR=
+set LOCALEDIR=
+pause
