Index: src/plugins/contrib/codestat/codestatexec.cpp
===================================================================
--- src/plugins/contrib/codestat/codestatexec.cpp       (revision 2234)
+++ src/plugins/contrib/codestat/codestatexec.cpp       (working copy)
@@ -7,6 +7,7 @@
  * License:   GPL
  **************************************************************/
 
+#include <editormanager.h>
 #include "codestatexec.h"
 
 /** Count the lines on all project's files and display the results.
