Index: src/sdk/projectmanager.cpp
===================================================================
--- src/sdk/projectmanager.cpp	(revision 2244)
+++ src/sdk/projectmanager.cpp	(working copy)
@@ -631,6 +631,10 @@
 cbProject* ProjectManager::LoadProject(const wxString& filename, bool activateIt)
 {
     SANITY_CHECK(0L);
+
+	if (m_IsLoadingProject)
+        return 0L;
+
     cbProject* result = 0;
 
     // disallow application shutdown while opening files
