Index: src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- src/plugins/compilergcc/compilergcc.cpp     (revision 2258)
+++ src/plugins/compilergcc/compilergcc.cpp     (working copy)
@@ -2191,7 +2191,7 @@
     m_RunAfterCompile = false;
     if (!IsProcessRunning())
         return 0;
-    wxKillError ret;
+    wxKillError ret = wxKILL_OK;
 
     m_CommandQueue.Clear();
