Index: src/sdk/cbthreadpool.cpp
===================================================================
--- src/sdk/cbthreadpool.cpp    (revision 2261)
+++ src/sdk/cbthreadpool.cpp    (working copy)
@@ -136,6 +136,7 @@
 cbThreadPool::cbThreadPool(wxEvtHandler* owner, int id, int concurrentThreads)
     : m_pOwner(owner),
     m_ID(id),
+    m_ConcurrentThreads(0),
     m_Done(true),
     m_Batching(false),
     m_Counter(0),
