Index: src/src/splashscreen.h
===================================================================
--- src/src/splashscreen.h	(revision 2274)
+++ src/src/splashscreen.h	(working copy)
@@ -25,7 +25,7 @@
 
   public:
     // A value of -1 for timeout makes it stay forever (you need to close it manually)
-    cbSplashScreen(wxBitmap &label, long timeout, wxWindow *parent, wxWindowID id, long style = wxSTAY_ON_TOP | wxNO_BORDER | wxFRAME_NO_TASKBAR);
+    cbSplashScreen(wxBitmap &label, long timeout, wxWindow *parent, wxWindowID id, long style = wxNO_BORDER | wxFRAME_NO_TASKBAR);
     ~cbSplashScreen();
 
   private:
@@ -41,4 +41,3 @@
 };
 
 #endif // CBSPLASH_SCREEN_H
-
