Index: src/src/environmentsettingsdlg.cpp
===================================================================
--- src/src/environmentsettingsdlg.cpp  (revision 2274)
+++ src/src/environmentsettingsdlg.cpp  (working copy)
@@ -177,7 +177,6 @@
 
     // disable some windows-only settings, in other platforms
 #ifndef __WXMSW__
-    XRCCTRL(*this, "chkDDE", wxCheckBox)->Enable(false);
     XRCCTRL(*this, "chkAssociations", wxCheckBox)->Enable(false);
     XRCCTRL(*this, "btnSetAssocs", wxButton)->Enable(false);
     XRCCTRL(*this, "btnManageAssocs", wxButton)->Enable(false);
