Index: src/plugins/compilergcc/resources/compiler_options.xrc
===================================================================
--- src/plugins/compilergcc/resources/compiler_options.xrc	(revision 2285)
+++ src/plugins/compilergcc/resources/compiler_options.xrc	(working copy)
@@ -635,20 +635,20 @@
                                 </object>
                               </object>
                               <flag>wxTOP|wxLEFT|wxRIGHT|wxGROW</flag>
-                              <border>8</border>
+                              <border>4</border>
                             </object>
                             <object class="sizeritem">
                               <object class="wxStaticText">
-                                <label>NOTE: All programs below, must exist either in the &quot;bin&quot; sub-directory of this path
-or in any of the &quot;Additional paths&quot;...</label>
+                                <label>NOTE: All programs below, must exist either in the &quot;bin&quot; sub-directory of this
+path or in any of the&quot;Additional paths&quot;...</label>
                                 <fg>#800000</fg>
                               </object>
                               <flag>wxBOTTOM|wxLEFT|wxRIGHT|wxGROW</flag>
-                              <border>8</border>
+                              <border>1</border>
                             </object>
                           </object>
-                          <flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND</flag>
-                          <border>8</border>
+                          <!--flag>wxTOP|wxLEFT|wxRIGHT|wxEXPAND</flag-->
+                          <border>1</border>
                         </object>
                         <object class="sizeritem">
                           <object class="wxBoxSizer">
@@ -664,8 +664,8 @@
                                         <object class="wxFlexGridSizer">
                                           <cols>2</cols>
                                           <rows>8</rows>
-                                          <vgap>8</vgap>
-                                          <hgap>8</hgap>
+                                          <vgap>1</vgap>
+                                          <hgap>1</hgap>
                                           <growablecols>1</growablecols>
                                           <object class="sizeritem">
                                             <object class="wxStaticText">
@@ -679,7 +679,7 @@
                                                 <object class="wxTextCtrl" name="txtCcompiler"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnCcompiler">
@@ -702,7 +702,7 @@
                                                 <object class="wxTextCtrl" name="txtCPPcompiler"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnCPPcompiler">
@@ -725,7 +725,7 @@
                                                 <object class="wxTextCtrl" name="txtLinker"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnLinker">
@@ -748,7 +748,7 @@
                                                 <object class="wxTextCtrl" name="txtLibLinker"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnLibLinker">
@@ -771,7 +771,7 @@
                                                 <object class="wxTextCtrl" name="txtDebugger"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnDebugger">
@@ -794,7 +794,7 @@
                                                 <object class="wxTextCtrl" name="txtResComp"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnResComp">
@@ -817,7 +817,7 @@
                                                 <object class="wxTextCtrl" name="txtMake"/>
                                                 <option>1</option>
                                                 <flag>wxEXPAND</flag>
-                                                <border>8</border>
+                                                <border>1</border>
                                               </object>
                                               <object class="sizeritem">
                                                 <object class="wxButton" name="btnMake">
@@ -831,7 +831,7 @@
                                         </object>
                                         <option>1</option>
                                         <flag>wxALL|wxEXPAND</flag>
-                                        <border>8</border>
+                                        <border>1</border>
                                       </object>
                                     </object>
                                   </object>
@@ -847,7 +847,7 @@
                                         </object>
                                         <option>1</option>
                                         <flag>wxALL|wxGROW</flag>
-                                        <border>8</border>
+                                        <border>1</border>
                                       </object>
                                       <object class="sizeritem">
                                         <object class="wxBoxSizer">
@@ -857,7 +857,7 @@
                                               <label>Add</label>
                                             </object>
                                             <flag>wxRIGHT</flag>
-                                            <border>8</border>
+                                            <border>1</border>
                                           </object>
                                           <object class="sizeritem">
                                             <object class="wxButton" name="btnExtraEdit">
@@ -1189,4 +1189,4 @@
       </object>
     </object>
   </object>
-</resource>
\ No newline at end of file
+</resource>
