Index: main.cpp
===================================================================
--- main.cpp	(revision 2337)
+++ main.cpp	(working copy)
@@ -697,9 +697,9 @@
     m_pToolbar = new wxToolBar(this, -1, wxDefaultPosition, size, wxTB_FLAT | wxTB_NODIVIDER);
     m_pToolbar->SetToolBitmapSize(size);
     Manager::Get()->AddonToolBar(m_pToolbar,xrcToolbarName);
-    m_pToolbar->SetBestFittingSize();
 
 	m_pToolbar->Realize();
+    m_pToolbar->SetBestFittingSize();
 
     // add toolbars in docking system
     m_LayoutManager.AddPane(m_pToolbar, wxPaneInfo().

