Index: src/sdk/editorcolorset.cpp
===================================================================
--- src/sdk/editorcolorset.cpp	(revision 2337)
+++ src/sdk/editorcolorset.cpp	(working copy)
@@ -466,7 +466,8 @@
 {
 	wxString key;
 	ConfigManager* cfg = Manager::Get()->GetConfigManager(_T("editor"));
-	cfg->DeleteSubPath(_T("/color_sets/") + m_Name);
+	// The next stmt causes all previous user settings to be deleted        //pecan 2006/4/11
+	//-cfg->DeleteSubPath(_T("/color_sets/") + m_Name);                     //pecan 2006/4/11
 
 	// write the theme name
 	cfg->Write(_T("/color_sets/") + m_Name + _T("/name"), m_Name);